{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "58e411d1-d48b-4705-9d78-6cfaa9190fc6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# this book will start from query sequence\n",
    "# generate folding with > plddt 70 in a folder for all the \n",
    "# clustered a3m > 5 Kb.\n",
    "# use myenv . does not need to run colabfold in colabfold env since all modules are installed in base. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "6e1dad3b-9f62-44a7-9a4e-9edd05d88322",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: pickleshare in /home/yuhong/colabfold/localcolabfold/colabfold-conda/lib/python3.10/site-packages (0.7.5)\n",
      "/home/yuhong/colabfold\n",
      "/home/yuhong/colabfold\n"
     ]
    }
   ],
   "source": [
    "!pip install pickleshare\n",
    "# Change to the notebook directory\n",
    "%cd /home/yuhong/colabfold/\n",
    "# Verify the directory\n",
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "da5e1a47-33bb-4500-ad7c-8b756c70a3f3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2024-10-31 03:21:32,517 Running colabfold 1.5.5 (c21e1768d18e3608e6e6d99c97134317e7e41c75)\n",
      "2024-10-31 03:21:32,548 non-fasta/a3m file in input directory: SEQ2/log.txt\n",
      "2024-10-31 03:21:32,910 Running on GPU\n",
      "2024-10-31 03:21:33,138 Found 5 citations for tools or databases\n",
      "2024-10-31 03:21:33,138 Query 1/1: WT_Tu (length 394)\n",
      "2024-10-31 03:21:44,425 Setting max_seq=1024, max_extra_seq=10\n",
      "WARNING: All log messages before absl::InitializeLog() is called are written to STDERR\n",
      "I0000 00:00:1730362970.624975    5022 mlir_graph_optimization_pass.cc:401] MLIR V1 optimization pass is not enabled\n",
      "2024-10-31 03:59:42,889 alphafold2_ptm_model_1_seed_000 recycle=0 pLDDT=83.1 pTM=0.665\n"
     ]
    }
   ],
   "source": [
    "# run fasta to get initial a3m.\n",
    "!colabfold_batch --num-recycle 3 --num-models 1 --max-seq 1024 --max-extra-seq 10 SEQ2 SEQ2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d0f1ab75-5c8b-4a5e-84c2-5fdac092a061",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EX\n",
      "0 seqs removed for containing more than 25% gaps, 190 remaining\n",
      "Testing for min_eps=5, max_eps=20.0\n",
      "5.00\t1\t48\n",
      "6.50\t1\t48\n",
      "8.00\t1\t48\n",
      "9.50\t1\t48\n",
      "11.00\t1\t48\n",
      "12.50\t1\t48\n",
      "14.00\t1\t48\n",
      "15.50\t1\t48\n",
      "17.00\t2\t45\n",
      "18.50\t2\t44\n",
      "20.00\t2\t45\n",
      "Testing for min_eps=12.5, max_eps=27.5\n",
      "12.50\t1\t48\n",
      "14.00\t1\t48\n",
      "15.50\t1\t48\n",
      "17.00\t1\t48\n",
      "18.50\t1\t48\n",
      "20.00\t4\t39\n",
      "21.50\t3\t41\n",
      "23.00\t5\t35\n",
      "24.50\t8\t25\n",
      "26.00\t8\t16\n",
      "27.50\t10\t12\n",
      "Testing for min_eps=20.0, max_eps=35.0\n",
      "20.00\t2\t45\n",
      "21.50\t4\t39\n",
      "23.00\t4\t34\n",
      "24.50\t7\t26\n",
      "26.00\t9\t21\n",
      "27.50\t7\t21\n",
      "29.00\t10\t12\n",
      "30.50\t4\t4\n",
      "32.00\t1\t0\n",
      "33.50\t1\t0\n",
      "35.00\t1\t0\n",
      "Selected optimal min_eps=12.5 with 10 clusters\n",
      "Selected eps=23.00\n",
      "190 total seqs\n",
      "30 clusters, 29 of 190 not clustered (0.15)\n",
      "avg identity to query of unclustered: 0.35\n",
      "avg identity to query of clustered: 0.35\n",
      "wrote clustering data to /home/yuhong/colabfold/SEQ/Cluster/EX_clustering_assignments.tsv\n",
      "wrote cluster metadata to /home/yuhong/colabfold/SEQ/Cluster/EX_cluster_metadata.tsv\n",
      "Saved this output to EX.log\n"
     ]
    }
   ],
   "source": [
    "# cluster .a3m in SEQ/. Then move them into Cluster\n",
    "# need ClusterMSA_notebook.py and utils.py\n",
    "default_path='/home/yuhong/colabfold'\n",
    "!python $default_path/ClusterMSA_notebook.py EX -i $default_path/SEQ/combine.a3m -o $default_path/SEQ/Cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "31506d99-55bf-4f0c-b97b-8a6eb9743eda",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING: All log messages before absl::InitializeLog() is called are written to STDERR\n",
      "E0000 00:00:1733468711.269398    2110 cuda_dnn.cc:8310] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered\n",
      "E0000 00:00:1733468711.272193    2110 cuda_blas.cc:1418] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered\n",
      "2024-12-06 01:05:12,441 Running colabfold 1.5.5 (c21e1768d18e3608e6e6d99c97134317e7e41c75)\n",
      "2024-12-06 01:05:12,442 non-fasta/a3m file in input directory: SEQ/cite.bibtex\n",
      "2024-12-06 01:05:12,442 non-fasta/a3m file in input directory: SEQ/config.json\n",
      "2024-12-06 01:05:12,442 non-fasta/a3m file in input directory: SEQ/log.txt\n",
      "2024-12-06 01:05:12,766 Running on GPU\n",
      "W0000 00:00:1733468712.800088    2110 gpu_device.cc:2344] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.\n",
      "Skipping registering GPU devices...\n",
      "2024-12-06 01:05:12,987 Found 5 citations for tools or databases\n",
      "2024-12-06 01:05:12,987 Done\n"
     ]
    }
   ],
   "source": [
    "## just to fold\n",
    "!colabfold_batch SEQ SEQ --num-models 1 --max-seq 1024"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "3eb33a1d-81cb-492d-b4d1-8f153a930b0d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removing EX_006.a3m (Size: 2891 bytes)\n",
      "Removing EX_025.a3m (Size: 4391 bytes)\n",
      "Removing EX_009.a3m (Size: 6555 bytes)\n",
      "Removing EX_015.a3m (Size: 4350 bytes)\n",
      "Removing EX_016.a3m (Size: 5082 bytes)\n",
      "Removing EX_029.a3m (Size: 5084 bytes)\n",
      "Removing EX_004.a3m (Size: 2908 bytes)\n",
      "Removing EX_021.a3m (Size: 3620 bytes)\n",
      "Removing EX_018.a3m (Size: 2885 bytes)\n",
      "Removing EX_026.a3m (Size: 3668 bytes)\n",
      "Removing EX_019.a3m (Size: 2899 bytes)\n",
      "Removing EX_023.a3m (Size: 4414 bytes)\n",
      "Removing EX_011.a3m (Size: 6580 bytes)\n",
      "Removing EX_027.a3m (Size: 5825 bytes)\n",
      "Removing EX_014.a3m (Size: 2889 bytes)\n",
      "Removing EX_022.a3m (Size: 2952 bytes)\n",
      "Removing EX_007.a3m (Size: 5784 bytes)\n",
      "Removing EX_001.a3m (Size: 2891 bytes)\n",
      "Removing EX_002.a3m (Size: 3619 bytes)\n",
      "Removing EX_008.a3m (Size: 3613 bytes)\n",
      "Removing EX_013.a3m (Size: 5148 bytes)\n",
      "Removing EX_003.a3m (Size: 3699 bytes)\n",
      "Removing EX_010.a3m (Size: 4393 bytes)\n",
      "Removing EX_000.a3m (Size: 5054 bytes)\n",
      "Removing EX_028.a3m (Size: 3620 bytes)\n",
      "Removing EX_005.a3m (Size: 4337 bytes)\n",
      "Cleanup complete!\n",
      "WARNING: All log messages before absl::InitializeLog() is called are written to STDERR\n",
      "E0000 00:00:1730784244.976990    5841 cuda_dnn.cc:8310] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered\n",
      "E0000 00:00:1730784244.990051    5841 cuda_blas.cc:1418] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered\n",
      "2024-11-04 23:24:06,919 Running colabfold 1.5.5 (c21e1768d18e3608e6e6d99c97134317e7e41c75)\n",
      "2024-11-04 23:24:06,919 non-fasta/a3m file in input directory: /home/yuhong/colabfold/SEQ/Cluster/EX_cluster_metadata.tsv\n",
      "2024-11-04 23:24:06,919 non-fasta/a3m file in input directory: /home/yuhong/colabfold/SEQ/Cluster/EX_clustering_assignments.tsv\n",
      "2024-11-04 23:24:07,317 Running on GPU\n",
      "W0000 00:00:1730784247.441560    5841 gpu_device.cc:2344] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.\n",
      "Skipping registering GPU devices...\n",
      "2024-11-04 23:24:07,691 Found 5 citations for tools or databases\n",
      "2024-11-04 23:24:07,691 Query 1/4: EX_012 (length 706)\n",
      "WARNING: All log messages before absl::InitializeLog() is called are written to STDERR\n",
      "I0000 00:00:1730784252.817125    5841 mlir_graph_optimization_pass.cc:401] MLIR V1 optimization pass is not enabled\n",
      "2024-11-04 23:25:08,474 alphafold2_ptm_model_1_seed_000 recycle=0 pLDDT=44.2 pTM=0.339\n",
      "2024-11-04 23:25:48,034 alphafold2_ptm_model_1_seed_000 recycle=1 pLDDT=52.8 pTM=0.394 tol=15.8\n",
      "2024-11-04 23:26:26,277 alphafold2_ptm_model_1_seed_000 recycle=2 pLDDT=56.7 pTM=0.475 tol=7.24\n",
      "2024-11-04 23:27:05,403 alphafold2_ptm_model_1_seed_000 recycle=3 pLDDT=64.2 pTM=0.596 tol=4.65\n",
      "2024-11-04 23:27:05,404 alphafold2_ptm_model_1_seed_000 took 172.3s (3 recycles)\n",
      "2024-11-04 23:27:05,631 reranking models by 'plddt' metric\n",
      "2024-11-04 23:27:05,631 rank_001_alphafold2_ptm_model_1_seed_000 pLDDT=64.2 pTM=0.596\n",
      "2024-11-04 23:27:06,070 Query 2/4: EX_017 (length 706)\n",
      "2024-11-04 23:27:45,111 alphafold2_ptm_model_1_seed_000 recycle=0 pLDDT=33.1 pTM=0.242\n",
      "2024-11-04 23:28:24,640 alphafold2_ptm_model_1_seed_000 recycle=1 pLDDT=40.3 pTM=0.292 tol=14.3\n",
      "2024-11-04 23:29:04,232 alphafold2_ptm_model_1_seed_000 recycle=2 pLDDT=48.9 pTM=0.377 tol=11.1\n",
      "2024-11-04 23:29:43,457 alphafold2_ptm_model_1_seed_000 recycle=3 pLDDT=52.6 pTM=0.4 tol=7.62\n",
      "2024-11-04 23:29:43,458 alphafold2_ptm_model_1_seed_000 took 157.7s (3 recycles)\n",
      "2024-11-04 23:29:43,683 reranking models by 'plddt' metric\n",
      "2024-11-04 23:29:43,683 rank_001_alphafold2_ptm_model_1_seed_000 pLDDT=52.6 pTM=0.4\n",
      "2024-11-04 23:29:43,966 Query 3/4: EX_020 (length 706)\n",
      "2024-11-04 23:30:21,892 alphafold2_ptm_model_1_seed_000 recycle=0 pLDDT=41.4 pTM=0.275\n",
      "2024-11-04 23:31:01,352 alphafold2_ptm_model_1_seed_000 recycle=1 pLDDT=44.2 pTM=0.339 tol=14\n",
      "2024-11-04 23:31:40,102 alphafold2_ptm_model_1_seed_000 recycle=2 pLDDT=50.4 pTM=0.411 tol=8.85\n",
      "2024-11-04 23:32:18,394 alphafold2_ptm_model_1_seed_000 recycle=3 pLDDT=52.1 pTM=0.435 tol=4.54\n",
      "2024-11-04 23:32:18,395 alphafold2_ptm_model_1_seed_000 took 153.8s (3 recycles)\n",
      "2024-11-04 23:32:18,653 reranking models by 'plddt' metric\n",
      "2024-11-04 23:32:18,654 rank_001_alphafold2_ptm_model_1_seed_000 pLDDT=52.1 pTM=0.435\n",
      "2024-11-04 23:32:18,998 Query 4/4: EX_024 (length 706)\n",
      "2024-11-04 23:32:57,762 alphafold2_ptm_model_1_seed_000 recycle=0 pLDDT=40.4 pTM=0.268\n",
      "2024-11-04 23:33:36,917 alphafold2_ptm_model_1_seed_000 recycle=1 pLDDT=48.5 pTM=0.351 tol=11.1\n",
      "2024-11-04 23:34:15,234 alphafold2_ptm_model_1_seed_000 recycle=2 pLDDT=52.2 pTM=0.378 tol=8.49\n",
      "2024-11-04 23:34:54,915 alphafold2_ptm_model_1_seed_000 recycle=3 pLDDT=53.5 pTM=0.388 tol=6.05\n",
      "2024-11-04 23:34:54,915 alphafold2_ptm_model_1_seed_000 took 155.3s (3 recycles)\n",
      "2024-11-04 23:34:55,128 reranking models by 'plddt' metric\n",
      "2024-11-04 23:34:55,128 rank_001_alphafold2_ptm_model_1_seed_000 pLDDT=53.5 pTM=0.388\n",
      "2024-11-04 23:34:55,449 Done\n"
     ]
    }
   ],
   "source": [
    "# remove a3m that are less than 5KB.then fold\n",
    "import os\n",
    "# Construct the full path to the \"SEQ/Cluster\" folder\n",
    "default_path='/home/yuhong/colabfold'\n",
    "folder_path = os.path.join(default_path, \"SEQ\", \"Cluster\")\n",
    "\n",
    "# Set the size threshold (5 KB = 5120 bytes)\n",
    "size_threshold = 7 * 1024  # 5 KB\n",
    "\n",
    "# Iterate through each file in the folder\n",
    "for filename in os.listdir(folder_path):\n",
    "    # Check if the file is a .a3m file\n",
    "    if filename.endswith(\".a3m\"):\n",
    "        file_path = os.path.join(folder_path, filename)\n",
    "        # Get the size of the file in bytes\n",
    "        file_size = os.path.getsize(file_path)\n",
    "        # If the file size is less than the threshold, remove the file\n",
    "        if file_size < size_threshold:\n",
    "            print(f\"Removing {filename} (Size: {file_size} bytes)\")\n",
    "            os.remove(file_path)\n",
    "\n",
    "print(\"Cleanup complete!\")\n",
    "\n",
    "# folding all these a3m files. \n",
    "!colabfold_batch {folder_path} {folder_path}/Fold/ --num-models 1 --max-seq 1024"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4a29a075-2f9d-4353-89f6-09f73ef313f0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# searching for best pLDDT > threshold, then move those into sel_fold folder. \n",
    "import os\n",
    "import shutil\n",
    "import csv\n",
    "import re\n",
    "\n",
    "# Define the log file path, CSV path, PDB directories, and threshold\n",
    "log_file_path = '/home/yuhong/localcolabfold/SEQ/Cluster/Fold/log.txt'  # Replace with the actual path\n",
    "csv_output_path = '/home/yuhong/localcolabfold/SEQ/Cluster/Fold/pLDDT.csv'  # Path for the output CSV file\n",
    "pLDDT_threshold = 70.0  # Set your threshold for pLDDT\n",
    "pdb_source_dir = '/home/yuhong/localcolabfold/SEQ/Cluster/Fold'  # Directory containing the PDB files\n",
    "pdb_target_dir = '/home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb'  # Directory to move selected PDB files\n",
    "\n",
    "# Ensure the target directory exists\n",
    "os.makedirs(pdb_target_dir, exist_ok=True)\n",
    "\n",
    "# Function to process the log file and extract pLDDT values and EX_xxx codes\n",
    "def process_log_file_and_generate_csv(log_file_path, csv_output_path, threshold):\n",
    "    with open(log_file_path, 'r') as log_file:\n",
    "        lines = log_file.readlines()\n",
    "\n",
    "    extracted_data = []\n",
    "\n",
    "    # Process the lines in the log file\n",
    "    for i, line in enumerate(lines):\n",
    "        # Look for lines containing \"rank_001\"\n",
    "        if \"rank_001\" in line:\n",
    "            # Extract the pLDDT value\n",
    "            pLDDT_match = re.search(r'pLDDT=([\\d.]+)', line)\n",
    "            if pLDDT_match:\n",
    "                pLDDT_value = float(pLDDT_match.group(1))\n",
    "                print(f\"Extracted pLDDT: {pLDDT_value}\")  # Debugging\n",
    "\n",
    "                # Search backward for the corresponding Query line\n",
    "                for j in range(i, -1, -1):\n",
    "                    if \"Query\" in lines[j]:\n",
    "                        # Extract the EX_xxx string with exactly 6 characters\n",
    "                        ex_match = re.search(r'(EX_\\w{3})', lines[j])\n",
    "                        if ex_match:\n",
    "                            ex_code = ex_match.group(0)\n",
    "                            print(f\"Found EX code: {ex_code}\")  # Debugging\n",
    "                            extracted_data.append([ex_code, pLDDT_value])\n",
    "                            break\n",
    "\n",
    "    # Write the extracted data to CSV\n",
    "    with open(csv_output_path, 'w', newline='') as csv_file:\n",
    "        writer = csv.writer(csv_file)\n",
    "        writer.writerow(['EX_Code', 'pLDDT_Value'])  # Header\n",
    "        writer.writerows(extracted_data)\n",
    "\n",
    "    return extracted_data\n",
    "\n",
    "# Function to move PDB files if pLDDT is above the threshold and rename them\n",
    "def move_and_rename_pdb_files(extracted_data, source_dir, target_dir, threshold):\n",
    "    for ex_code, pLDDT_value in extracted_data:\n",
    "        if pLDDT_value > threshold:\n",
    "            # Search for files that start with EX_xxx_unrelaxed_rank_001\n",
    "            for file_name in os.listdir(source_dir):\n",
    "                if file_name.startswith(f'{ex_code}_unrelaxed_rank_001'):\n",
    "                    pdb_source_path = os.path.join(source_dir, file_name)\n",
    "                    new_file_name = f'{ex_code}.pdb'  # Truncated name\n",
    "                    pdb_target_path = os.path.join(target_dir, new_file_name)\n",
    "\n",
    "                    # Move the file to the target directory and rename it\n",
    "                    shutil.move(pdb_source_path, pdb_target_path)\n",
    "                    print(f'Moved and renamed: {file_name} -> {new_file_name}')\n",
    "                    break\n",
    "            else:\n",
    "                print(f'File not found for EX code: {ex_code}')\n",
    "\n",
    "# Process the log file and generate the CSV\n",
    "extracted_data = process_log_file_and_generate_csv(log_file_path, csv_output_path, pLDDT_threshold)\n",
    "\n",
    "# Move and rename the PDB files\n",
    "move_and_rename_pdb_files(extracted_data, pdb_source_dir, pdb_target_dir, pLDDT_threshold)\n",
    "\n",
    "print(\"PDB file processing complete.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3639de8d-485b-4975-95d0-7b9b9de08454",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load all selected pdb into pymol, then align them to the highest confident structure.\n",
    "# Load all selected pdb into pymol, then align them to the highest confident structure.\n",
    "import os\n",
    "import shutil\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import pymol2\n",
    "\n",
    "# Define paths\n",
    "pdb_dir = '/home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/'        # Replace with your PDB files directory\n",
    "csv_file_path = '/home/yuhong/localcolabfold/SEQ/Cluster/Fold/pLDDT.csv'   # Replace with the path to your CSV file\n",
    "\n",
    "# Step 1: Move the CSV file into the PDB files directory\n",
    "csv_file_name = os.path.basename(csv_file_path)\n",
    "destination_csv_path = os.path.join(pdb_dir, csv_file_name)\n",
    "shutil.move(csv_file_path, destination_csv_path)\n",
    "\n",
    "print(f\"CSV file moved to: {destination_csv_path}\")\n",
    "\n",
    "# Step 2: Read and rearrange the CSV file\n",
    "# Read the CSV file into a DataFrame\n",
    "df = pd.read_csv(destination_csv_path)\n",
    "\n",
    "# Step 3: Sort the DataFrame based on the second column in descending order\n",
    "# Assuming the second column is at index 1 (zero-based indexing)\n",
    "second_column_name = df.columns[1]\n",
    "sorted_df = df.sort_values(by=second_column_name, ascending=False)\n",
    "\n",
    "# Step 4: Rewrite the CSV file with the sorted data\n",
    "sorted_df.to_csv(destination_csv_path, index=False)\n",
    "\n",
    "print(f\"CSV file sorted by '{second_column_name}' in descending order and saved.\")\n",
    "\n",
    "# Step 5: Proceed to align the PDB files\n",
    "# Get list of PDB files\n",
    "pdb_files = [f for f in os.listdir(pdb_dir) if f.endswith('.pdb')]\n",
    "\n",
    "# Use the first PDB file as the reference\n",
    "reference_pdb = pdb_files[0]\n",
    "reference_name = os.path.splitext(reference_pdb)[0]\n",
    "\n",
    "with pymol2.PyMOL() as pymol:\n",
    "    cmd = pymol.cmd\n",
    "\n",
    "    # Load the reference PDB\n",
    "    cmd.load(os.path.join(pdb_dir, reference_pdb), reference_name)\n",
    "\n",
    "    # List to store object names\n",
    "    object_names = [reference_name]\n",
    "\n",
    "    for pdb_file in pdb_files:\n",
    "        target_name = os.path.splitext(pdb_file)[0]\n",
    "\n",
    "        if pdb_file == reference_pdb:\n",
    "            continue  # Skip the reference file\n",
    "\n",
    "        # Load the target PDB\n",
    "        cmd.load(os.path.join(pdb_dir, pdb_file), target_name)\n",
    "\n",
    "        # Align the target to the reference\n",
    "        cmd.align(target_name, reference_name)\n",
    "\n",
    "        # Save the aligned structure, overwriting the original file\n",
    "        cmd.save(os.path.join(pdb_dir, pdb_file), target_name)\n",
    "\n",
    "        # Keep the target object in the session for RMSD calculation\n",
    "        object_names.append(target_name)\n",
    "\n",
    "    # Optional: Adjust visualization settings here if needed\n",
    "\n",
    "    # Save the session as 'Align_all.pse' in the PDB directory\n",
    "    session_path = os.path.join(pdb_dir, 'Align_all.pse')\n",
    "    cmd.save(session_path)\n",
    "\n",
    "    print(f\"Session saved as: {session_path}\")\n",
    "\n",
    "    # Step 6: Compute the RMSD matrix\n",
    "    num_objects = len(object_names)\n",
    "    rmsd_matrix = np.zeros((num_objects, num_objects))\n",
    "\n",
    "    for i in range(num_objects):\n",
    "        obj1 = object_names[i]\n",
    "        for j in range(i, num_objects):\n",
    "            obj2 = object_names[j]\n",
    "\n",
    "            # Since the structures are already aligned, use 'rms_cur' without further alignment\n",
    "            rmsd = cmd.rms_cur(obj1, obj2, quiet=1)\n",
    "            rmsd_matrix[i, j] = rmsd\n",
    "            rmsd_matrix[j, i] = rmsd  # Symmetric matrix\n",
    "\n",
    "    # Create DataFrame with headers\n",
    "    df_rmsd = pd.DataFrame(rmsd_matrix, index=object_names, columns=object_names)\n",
    "\n",
    "    # Save the RMSD matrix to CSV\n",
    "    rmsd_csv_path = os.path.join(pdb_dir, 'RMSD_matrix.csv')\n",
    "    df_rmsd.to_csv(rmsd_csv_path)\n",
    "\n",
    "    print(f\"RMSD matrix saved to: {rmsd_csv_path}\")\n",
    "\n",
    "print(\"All PDB files have been aligned, saved, and the session has been saved as 'Align_all.pse'.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8a38521f-0b34-42dc-b528-bba2fa10f034",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load all selected pdb into pymol, then align them to the highest confident structure.\n",
    "# make the RMSD matric and plit t-sne plot and DBscan. \n",
    "import os\n",
    "import shutil\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import pymol2\n",
    "from sklearn.manifold import TSNE\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "\n",
    "# Define paths\n",
    "pdb_dir = '/home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/'        # Replace with your PDB files directory\n",
    "csv_file_path = '/home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/pLDDT.csv'   # Replace with the path to your CSV file\n",
    "\n",
    "# Step 1: Move the CSV file into the PDB files directory\n",
    "csv_file_name = os.path.basename(csv_file_path)\n",
    "destination_csv_path = os.path.join(pdb_dir, csv_file_name)\n",
    "shutil.move(csv_file_path, destination_csv_path)\n",
    "\n",
    "print(f\"CSV file moved to: {destination_csv_path}\")\n",
    "\n",
    "# Step 2: Read and rearrange the CSV file\n",
    "df = pd.read_csv(destination_csv_path)\n",
    "\n",
    "# Step 3: Sort the DataFrame based on the second column in descending order\n",
    "second_column_name = df.columns[1]\n",
    "sorted_df = df.sort_values(by=second_column_name, ascending=False)\n",
    "\n",
    "# Step 4: Rewrite the CSV file with the sorted data\n",
    "sorted_df.to_csv(destination_csv_path, index=False)\n",
    "print(f\"CSV file sorted by '{second_column_name}' in descending order and saved.\")\n",
    "\n",
    "# Step 5: Proceed to align the PDB files\n",
    "pdb_files = [f for f in os.listdir(pdb_dir) if f.endswith('.pdb')]\n",
    "\n",
    "# Use the first PDB file as the reference\n",
    "reference_pdb = pdb_files[0]\n",
    "reference_name = os.path.splitext(reference_pdb)[0]\n",
    "\n",
    "with pymol2.PyMOL() as pymol:\n",
    "    cmd = pymol.cmd\n",
    "\n",
    "    # Load the reference PDB\n",
    "    cmd.load(os.path.join(pdb_dir, reference_pdb), reference_name)\n",
    "\n",
    "    # List to store object names\n",
    "    object_names = [reference_name]\n",
    "\n",
    "    for pdb_file in pdb_files:\n",
    "        target_name = os.path.splitext(pdb_file)[0]\n",
    "\n",
    "        if pdb_file == reference_pdb:\n",
    "            continue  # Skip the reference file\n",
    "\n",
    "        # Load the target PDB\n",
    "        cmd.load(os.path.join(pdb_dir, pdb_file), target_name)\n",
    "\n",
    "        # Align the target to the reference\n",
    "        cmd.align(target_name, reference_name)\n",
    "\n",
    "        # Save the aligned structure, overwriting the original file\n",
    "        cmd.save(os.path.join(pdb_dir, pdb_file), target_name)\n",
    "\n",
    "        # Keep the target object in the session for RMSD calculation\n",
    "        object_names.append(target_name)\n",
    "\n",
    "    # Save the session as 'Align_all.pse' in the PDB directory\n",
    "    session_path = os.path.join(pdb_dir, 'Align_all.pse')\n",
    "    cmd.save(session_path)\n",
    "    print(f\"Session saved as: {session_path}\")\n",
    "\n",
    "    # Step 6: Compute the RMSD matrix\n",
    "    num_objects = len(object_names)\n",
    "    rmsd_matrix = np.zeros((num_objects, num_objects))\n",
    "\n",
    "    for i in range(num_objects):\n",
    "        obj1 = object_names[i]\n",
    "        for j in range(i, num_objects):\n",
    "            obj2 = object_names[j]\n",
    "\n",
    "            # Since the structures are already aligned, use 'rms_cur' without further alignment\n",
    "            rmsd = cmd.rms_cur(obj1, obj2, quiet=1)\n",
    "            rmsd_matrix[i, j] = rmsd\n",
    "            rmsd_matrix[j, i] = rmsd  # Symmetric matrix\n",
    "\n",
    "    # Create DataFrame with headers\n",
    "    df_rmsd = pd.DataFrame(rmsd_matrix, index=object_names, columns=object_names)\n",
    "\n",
    "    # Save the RMSD matrix to CSV\n",
    "    rmsd_csv_path = os.path.join(pdb_dir, 'RMSD_matrix.csv')\n",
    "    df_rmsd.to_csv(rmsd_csv_path)\n",
    "    print(f\"RMSD matrix saved to: {rmsd_csv_path}\")\n",
    "\n",
    "print(\"All PDB files have been aligned, saved, and the session has been saved as 'Align_all.pse'.\")\n",
    "\n",
    "# Step 7: Perform t-SNE and plot the results\n",
    "\n",
    "# Read the RMSD matrix\n",
    "df_rmsd = pd.read_csv(rmsd_csv_path, index_col=0)\n",
    "object_names = df_rmsd.index.tolist()\n",
    "\n",
    "# Convert the RMSD matrix into a NumPy array for t-SNE\n",
    "rmsd_matrix = df_rmsd.values\n",
    "\n",
    "# Adjust perplexity to be less than the number of PDB files\n",
    "n_samples = rmsd_matrix.shape[0]  # Number of samples (PDB files)\n",
    "perplexity_value = min(30, n_samples - 1)  # Set perplexity to a smaller value\n",
    "\n",
    "# Initialize t-SNE with the adjusted perplexity and init as \"random\"\n",
    "tsne = TSNE(n_components=2, metric='precomputed', perplexity=perplexity_value, init=\"random\", random_state=42)\n",
    "\n",
    "# Perform t-SNE embedding\n",
    "tsne_results = tsne.fit_transform(rmsd_matrix)\n",
    "\n",
    "# Create a DataFrame to store t-SNE results and labels\n",
    "tsne_df = pd.DataFrame({\n",
    "    'tsne-2d-one': tsne_results[:, 0],\n",
    "    'tsne-2d-two': tsne_results[:, 1],\n",
    "    'label': object_names\n",
    "})\n",
    "\n",
    "# Plotting the t-SNE results with a reasonable figure size\n",
    "plt.figure(figsize=(10, 7))\n",
    "sns.scatterplot(\n",
    "    x='tsne-2d-one', y='tsne-2d-two',\n",
    "    hue='label',\n",
    "    palette=sns.color_palette(\"hsv\", len(object_names)),\n",
    "    data=tsne_df,\n",
    "    legend=\"full\",\n",
    "    alpha=0.8\n",
    ")\n",
    "plt.title(\"t-SNE projection of RMSD matrix\")\n",
    "plt.show()\n",
    "\n",
    "# Annotate each point with the label\n",
    "for i in range(tsne_df.shape[0]):\n",
    "    plt.text(\n",
    "        tsne_df['tsne-2d-one'][i],\n",
    "        tsne_df['tsne-2d-two'][i],\n",
    "        tsne_df['label'][i],\n",
    "        fontsize=9\n",
    "    )\n",
    "\n",
    "plt.title('t-SNE Visualization of PDB Structures Based on RMSD')\n",
    "plt.xlabel('t-SNE Dimension 1')\n",
    "plt.ylabel('t-SNE Dimension 2')\n",
    "plt.xlim(min(tsne_df['tsne-2d-one']) - 10, max(tsne_df['tsne-2d-one']) + 10)\n",
    "plt.ylim(min(tsne_df['tsne-2d-two']) - 10, max(tsne_df['tsne-2d-two']) + 10)\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f99efad9-667a-46e3-882e-380c628b0f07",
   "metadata": {},
   "outputs": [],
   "source": [
    "# reads the 2D RMSD matrix, sorts it to group together all PDB structures within an RMSD threshold of 3.1, \n",
    "#and extracts submatrices iteratively until no further grouping is possible.\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "# Load the 2D RMSD matrix\n",
    "rmsd_csv_path = '/home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/RMSD_matrix.csv'  # Replace with your actual path\n",
    "df_rmsd = pd.read_csv(rmsd_csv_path, index_col=0)\n",
    "rmsd_matrix = df_rmsd.values\n",
    "\n",
    "# Define the threshold for resemblance (RMSD < 3.1)\n",
    "threshold = 2.8\n",
    "\n",
    "# List to store clusters\n",
    "clusters = []\n",
    "\n",
    "# Function to pool all IDs within a given threshold from the first ID in a group\n",
    "def pool_ids(df_rmsd, starting_id, threshold):\n",
    "    pool = set([starting_id])  # Initialize the pool with the starting ID\n",
    "    to_check = [starting_id]  # List of IDs to check for further pooling\n",
    "\n",
    "    while to_check:\n",
    "        current_id = to_check.pop(0)\n",
    "        current_idx = df_rmsd.index.get_loc(current_id)\n",
    "        \n",
    "        # Iterate through remaining IDs to find those within the threshold\n",
    "        for other_id in df_rmsd.index:\n",
    "            if other_id not in pool and df_rmsd.loc[current_id, other_id] <= threshold:\n",
    "                pool.add(other_id)\n",
    "                to_check.append(other_id)\n",
    "    \n",
    "    return list(pool)\n",
    "\n",
    "# Process the RMSD matrix and group the IDs iteratively\n",
    "while len(df_rmsd) > 0:\n",
    "    first_id = df_rmsd.index[0]\n",
    "    \n",
    "    # Pool all IDs within the threshold from the first ID\n",
    "    current_cluster = pool_ids(df_rmsd, first_id, threshold)\n",
    "    clusters.append(current_cluster)\n",
    "    \n",
    "    # Remove the pooled IDs from the RMSD matrix\n",
    "    df_rmsd = df_rmsd.drop(current_cluster, axis=0).drop(current_cluster, axis=1)\n",
    "    \n",
    "    print(f\"Cluster created with IDs: {current_cluster}\")\n",
    "\n",
    "# Save the clusters to a text file\n",
    "cluster_output_path = '/home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/cluster_list.txt'\n",
    "with open(cluster_output_path, 'w') as f:\n",
    "    for idx, cluster in enumerate(clusters):\n",
    "        f.write(f\"Cluster {idx+1}: {', '.join(cluster)}\\n\")\n",
    "\n",
    "print(f\"Clusters saved to {cluster_output_path}\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "153ec268-7911-4538-803a-f7f78a5825e4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# combine a3m based on structure clustering\n",
    "import os\n",
    "import shutil\n",
    "\n",
    "# Define the paths\n",
    "cluster_txt_path = '/home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/cluster_list.txt'  # Path to cluster_list.txt\n",
    "a3m_dir = '/home/yuhong/localcolabfold/SEQ/Cluster/'  # Directory where all the .a3m files are located\n",
    "output_dir = '/home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/'  # Output directory where combined .a3m files will be saved\n",
    "\n",
    "# Ensure the output directory exists\n",
    "if not os.path.exists(output_dir):\n",
    "    os.makedirs(output_dir)\n",
    "\n",
    "# Function to combine .a3m files\n",
    "def combine_a3m_files(cluster_num, a3m_files, output_dir):\n",
    "    combined_content = []\n",
    "    first_file = a3m_files[0]\n",
    "    \n",
    "    # Read the first file and prefix C# to all headers\n",
    "    with open(first_file, 'r') as f:\n",
    "        lines = f.readlines()\n",
    "        for i in range(0, len(lines), 2):\n",
    "            header = lines[i].strip()\n",
    "            sequence = lines[i+1].strip()\n",
    "            \n",
    "            # Modify the header with cluster prefix\n",
    "            new_header = f\">C{cluster_num}_\" + header[1:]\n",
    "            combined_content.append(new_header + '\\n')\n",
    "            combined_content.append(sequence + '\\n')\n",
    "    \n",
    "    # For subsequent files, skip the first sequence and add the rest\n",
    "    for a3m_file in a3m_files[1:]:\n",
    "        with open(a3m_file, 'r') as f:\n",
    "            lines = f.readlines()\n",
    "        \n",
    "        # Skip the first sequence (>xxx and its sequence)\n",
    "        for i in range(2, len(lines), 2):\n",
    "            header = lines[i].strip()\n",
    "            sequence = lines[i+1].strip()\n",
    "            \n",
    "            # Modify the header with cluster prefix\n",
    "            new_header = f\">C{cluster_num}_\" + header[1:]\n",
    "            combined_content.append(new_header + '\\n')\n",
    "            combined_content.append(sequence + '\\n')\n",
    "    \n",
    "    # Write the combined file with the cluster name only\n",
    "    output_filename = f\"C{cluster_num}.a3m\"\n",
    "    output_filepath = os.path.join(output_dir, output_filename)\n",
    "    \n",
    "    with open(output_filepath, 'w') as out_f:\n",
    "        out_f.writelines(combined_content)\n",
    "    \n",
    "    print(f\"Combined file saved as: {output_filepath}\")\n",
    "\n",
    "# Read the cluster_list.txt file\n",
    "with open(cluster_txt_path, 'r') as f:\n",
    "    clusters = f.readlines()\n",
    "\n",
    "# Process each cluster\n",
    "for cluster in clusters:\n",
    "    # Extract cluster information\n",
    "    cluster_info = cluster.strip().split(':')\n",
    "    cluster_num = int(cluster_info[0].replace('Cluster ', '').strip())\n",
    "    ids = [id_.strip() for id_ in cluster_info[1].split(',')]\n",
    "    \n",
    "    # Find corresponding .a3m files in the a3m_dir\n",
    "    a3m_files = [os.path.join(a3m_dir, id_ + '.a3m') for id_ in ids if os.path.exists(os.path.join(a3m_dir, id_ + '.a3m'))]\n",
    "    \n",
    "    if len(a3m_files) > 1:\n",
    "        # If the cluster has more than 1 item, combine the .a3m files\n",
    "        combine_a3m_files(cluster_num, a3m_files, output_dir)\n",
    "    elif len(a3m_files) == 1:\n",
    "        # If the cluster has only 1 item, just move the .a3m file to the output directory\n",
    "        shutil.move(a3m_files[0], output_dir)\n",
    "        print(f\"Moved {os.path.basename(a3m_files[0])} to {output_dir}\")\n",
    "    else:\n",
    "        print(f\"No .a3m files found for Cluster {cluster_num}\")\n",
    "\n",
    "print(\"Processing complete.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "16a158ef-eb35-4046-8728-33e581b85022",
   "metadata": {},
   "outputs": [],
   "source": [
    "# this cell combines structure clustered a3m. and combine them with cluster tag. \n",
    "import os\n",
    "\n",
    "# Define the paths\n",
    "cluster_txt_path = '/home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/cluster_list.txt'  # Path to cluster_list.txt\n",
    "a3m_dir = '/home/yuhong/localcolabfold/SEQ/Cluster/'  # Directory where all the .a3m files are located\n",
    "output_dir = '/home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/'  # Output directory where combined .a3m files will be saved\n",
    "concatenated_fasta_path = os.path.join(output_dir, 'Concatenated_All.fasta')  # Save concatenated .a3m files as a single fasta\n",
    "\n",
    "# Ensure the output directory exists\n",
    "if not os.path.exists(output_dir):\n",
    "    os.makedirs(output_dir)\n",
    "\n",
    "# Dictionary to store line counts for each C#.a3m file\n",
    "c_file_line_counts = {}\n",
    "\n",
    "# Function to combine or modify .a3m files and append to the grand fasta\n",
    "def combine_a3m_files(cluster_num, a3m_files, output_dir):\n",
    "    combined_content = []\n",
    "    first_file = a3m_files[0]\n",
    "    \n",
    "    # Read the first file and prefix C# to all headers\n",
    "    with open(first_file, 'r') as f:\n",
    "        lines = f.readlines()\n",
    "        for i in range(0, len(lines), 2):\n",
    "            header = lines[i].strip()\n",
    "            sequence = lines[i+1].strip()\n",
    "            \n",
    "            # Modify the header with cluster prefix\n",
    "            new_header = f\">C{cluster_num}_\" + header[1:]\n",
    "            combined_content.append(new_header)\n",
    "            combined_content.append(sequence)\n",
    "    \n",
    "    # For subsequent files, skip the first sequence and add the rest\n",
    "    for a3m_file in a3m_files[1:]:\n",
    "        with open(a3m_file, 'r') as f:\n",
    "            lines = f.readlines()\n",
    "        \n",
    "        # Skip the first sequence (>xxx and its sequence)\n",
    "        for i in range(2, len(lines), 2):\n",
    "            header = lines[i].strip()\n",
    "            sequence = lines[i+1].strip()\n",
    "            \n",
    "            # Modify the header with cluster prefix\n",
    "            new_header = f\">C{cluster_num}_\" + header[1:]\n",
    "            combined_content.append(new_header)\n",
    "            combined_content.append(sequence)\n",
    "    \n",
    "    # Write the combined file with the cluster name only\n",
    "    output_filename = f\"C{cluster_num}.a3m\"\n",
    "    output_filepath = os.path.join(output_dir, output_filename)\n",
    "    \n",
    "    with open(output_filepath, 'w') as out_f:\n",
    "        out_f.writelines(f\"{line}\\n\" for line in combined_content)\n",
    "    \n",
    "    # Store the line count for this combined file\n",
    "    c_file_line_counts[output_filename] = len(combined_content)\n",
    "    \n",
    "    print(f\"Combined file saved as: {output_filepath} with {len(combined_content)} lines\")\n",
    "\n",
    "# Function to modify a single .a3m file by adding the prefix to the headers\n",
    "def modify_single_a3m(cluster_num, a3m_file, output_dir):\n",
    "    modified_content = []\n",
    "    \n",
    "    with open(a3m_file, 'r') as f:\n",
    "        lines = f.readlines()\n",
    "        for i in range(0, len(lines), 2):\n",
    "            header = lines[i].strip()\n",
    "            sequence = lines[i+1].strip()\n",
    "            \n",
    "            # Modify the header with cluster prefix\n",
    "            new_header = f\">C{cluster_num}_\" + header[1:]\n",
    "            modified_content.append(new_header)\n",
    "            modified_content.append(sequence)\n",
    "    \n",
    "    # Write the modified file with the cluster name\n",
    "    output_filename = f\"C{cluster_num}.a3m\"\n",
    "    output_filepath = os.path.join(output_dir, output_filename)\n",
    "    \n",
    "    with open(output_filepath, 'w') as out_f:\n",
    "        out_f.writelines(f\"{line}\\n\" for line in modified_content)\n",
    "    \n",
    "    # Store the line count for this single file\n",
    "    c_file_line_counts[output_filename] = len(modified_content)\n",
    "    \n",
    "    print(f\"Modified single file saved as: {output_filepath} with {len(modified_content)} lines\")\n",
    "\n",
    "# Read the cluster_list.txt file\n",
    "with open(cluster_txt_path, 'r') as f:\n",
    "    clusters = f.readlines()\n",
    "\n",
    "# Process each cluster\n",
    "for cluster in clusters:\n",
    "    # Extract cluster information\n",
    "    cluster_info = cluster.strip().split(':')\n",
    "    cluster_num = int(cluster_info[0].replace('Cluster ', '').strip())\n",
    "    ids = [id_.strip() for id_ in cluster_info[1].split(',')]\n",
    "    \n",
    "    # Find corresponding .a3m files in the a3m_dir\n",
    "    a3m_files = [os.path.join(a3m_dir, id_ + '.a3m') for id_ in ids if os.path.exists(os.path.join(a3m_dir, id_ + '.a3m'))]\n",
    "    \n",
    "    # Combine only Cluster 1 and Cluster 2\n",
    "    if cluster_num == 1 or cluster_num == 2:\n",
    "        if len(a3m_files) > 1:\n",
    "            combine_a3m_files(cluster_num, a3m_files, output_dir)\n",
    "        elif len(a3m_files) == 1:\n",
    "            modify_single_a3m(cluster_num, a3m_files[0], output_dir)\n",
    "    else:\n",
    "        # Modify the .a3m file by adding the prefix for all other clusters\n",
    "        if len(a3m_files) == 1:\n",
    "            modify_single_a3m(cluster_num, a3m_files[0], output_dir)\n",
    "\n",
    "# Function to filter out sequences with \"_101\" in the header, except for C1_101\n",
    "def filter_sequences(content):\n",
    "    filtered_content = []\n",
    "    first_c1_101_found = False\n",
    "    for i in range(0, len(content), 2):\n",
    "        header = content[i].strip()\n",
    "        if \"_101\" in header:\n",
    "            if \"C1_101\" in header and not first_c1_101_found:\n",
    "                first_c1_101_found = True\n",
    "            else:\n",
    "                continue  # Skip sequences with \"_101\" in the header, except \"C1_101\"\n",
    "        filtered_content.append(header)\n",
    "        filtered_content.append(content[i+1])  # Add the sequence\n",
    "    return filtered_content\n",
    "\n",
    "# Concatenate all individual .a3m files into one combined fasta file\n",
    "concatenated_content = []\n",
    "for c_file in sorted(c_file_line_counts.keys()):\n",
    "    with open(os.path.join(output_dir, c_file), 'r') as f:\n",
    "        concatenated_content.extend(f.readlines())\n",
    "\n",
    "# Filter the concatenated content\n",
    "filtered_concatenated_content = filter_sequences(concatenated_content)\n",
    "\n",
    "# Save the filtered concatenated content as a fasta file in the same directory as the .txt file\n",
    "with open(concatenated_fasta_path, 'w') as concatenated_fasta:\n",
    "    concatenated_fasta.writelines(f\"{line.strip()}\\n\" for line in filtered_concatenated_content)\n",
    "\n",
    "print(f\"Concatenated and filtered fasta file saved as: {concatenated_fasta_path}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "21d86d6a-2ca2-4c8a-8798-35899dc77971",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All.fasta...\n",
      "No UniRef100 ID found in C1_101, skipping.\n",
      "Fetched data for A0A8H8S4L3: Elongation factor Tu, Lachnellula occidentalis\n",
      "Fetched data for A0A3E2HCY8: Elongation factor Tu, Scytalidium lignicola (Hyphomycete)\n",
      "Fetched data for A0A8H4RH48: Elongation factor Tu, Cudoniella acicularis\n",
      "Fetched data for A0A2A9PMI8: Elongation factor Tu, Ophiocordyceps unilateralis (Zombie-ant fungus) (Torrubia unilateralis)\n",
      "No data found for UPI0020073F4F.\n",
      "Fetched data for A0A0D1XB65: Elongation factor Tu, Verruconis gallopava\n",
      "Fetched data for A0A5M9JYY1: Elongation factor Tu, Monilinia fructicola (Brown rot fungus) (Ciboria fructicola)\n",
      "Fetched data for A0A094CMQ6: Elongation factor Tu, Pseudogymnoascus sp. VKM F-3775\n",
      "Fetched data for A0A1D2JH36: Elongation factor Tu, Paracoccidioides brasiliensis\n",
      "Fetched data for A0A4Q4W8P2: Elongation factor Tu, Monosporascus sp. mg162\n",
      "Fetched data for A0A093XNZ1: Elongation factor Tu, Talaromyces marneffei PM1\n",
      "Fetched data for A0A6A6TY91: Elongation factor Tu, Microthyrium microscopicum\n",
      "Fetched data for E9CV19: Elongation factor Tu, Coccidioides posadasii (strain RMSCC 757 / Silveira) (Valley fever fungus)\n",
      "Fetched data for A0A0F4YMF9: Elongation factor Tu, Rasamsonia emersonii CBS 393.64\n",
      "Fetched data for A0A8H6FBP9: Elongation factor Tu, Letharia columbiana\n",
      "Fetched data for A0A420Y1Y2: Elongation factor Tu, Coniochaeta pulveracea\n",
      "Fetched data for A0A0G4MSC6: Elongation factor Tu, Verticillium longisporum\n",
      "Fetched data for A0A178B9H0: Elongation factor Tu, Stagonospora sp. SRC1lsM3a\n",
      "No data found for UPI001FDCBA40.\n",
      "Fetched data for A0A1L9SRY3: Elongation factor Tu, Penicilliopsis zonata CBS 506.65\n",
      "Fetched data for A0A0G2GRF6: Elongation factor Tu, Phaeomoniella chlamydospora\n",
      "No data found for UPI00216DCB76.\n",
      "No data found for Q5BEE6.\n",
      "Fetched data for A0A100II62: Elongation factor Tu, Aspergillus niger\n",
      "No UniRef100 ID found in C1_S3D6E9, skipping.\n",
      "No UniRef100 ID found in C1_S3CY31, skipping.\n",
      "No UniRef100 ID found in C1_A0A074X752, skipping.\n",
      "No UniRef100 ID found in C1_A0A094HQU2, skipping.\n",
      "No UniRef100 ID found in C1_A0A167CRK5, skipping.\n",
      "No UniRef100 ID found in C1_A0A179IE94, skipping.\n",
      "No UniRef100 ID found in C1_A0A1D2JH36, skipping.\n",
      "No UniRef100 ID found in C1_A0A2B7Y9E6, skipping.\n",
      "No UniRef100 ID found in C1_A0A0D9NR27, skipping.\n",
      "No UniRef100 ID found in C1_A0A0D9NR27_16494367|ERR873962_k119_83701|-|1153|0|3|71|3106|3106[3106]:2903[2903]:204[204]|2857[2860]:1730[1730]:1128[1125]|412[412]:71[71]:342[342], skipping.\n",
      "No UniRef100 ID found in C1_ERR1711939_1167197, skipping.\n",
      "Fetched data for A0A5N6TSQ9: Elongation factor Tu, Aspergillus avenaceus\n",
      "No UniRef100 ID found in C1_A0A1J9Q692, skipping.\n",
      "No UniRef100 ID found in C1_A0A100II62, skipping.\n",
      "No UniRef100 ID found in C1_A0A0G4NIJ4, skipping.\n",
      "Fetched data for A0A5C3NTY7: Elongation factor Tu, Heliocybe sulcata\n",
      "Fetched data for A0A0H2S8E1: Elongation factor Tu, Schizopora paradoxa\n",
      "Fetched data for A0A4Z0AB69: Elongation factor Tu, Hericium alpestre\n",
      "Fetched data for A0A2H3IWI7: Elongation factor Tu, Wolfiporia cocos (strain MD-104) (Brown rot fungus)\n",
      "Fetched data for A0A8H7FNM9: Elongation factor Tu, Athelia sp. TMB\n",
      "Fetched data for A0A409VBL4: Elongation factor Tu, Gymnopilus dilepis\n",
      "Fetched data for A0A0C3S7H8: Elongation factor Tu, Phlebiopsis gigantea (strain 11061_1 CR5-6) (White-rot fungus) (Peniophora gigantea)\n",
      "Fetched data for A0A4S8MR95: Elongation factor Tu, Dendrothele bispora (strain CBS 962.96)\n",
      "Fetched data for A0A067QJI0: Elongation factor Tu, Jaapia argillacea MUCL 33604\n",
      "Fetched data for A0A067MI84: Elongation factor Tu, Botryobasidium botryosum (strain FD-172 SS1)\n",
      "No data found for UPI001FA4C6CF.\n",
      "Fetched data for A0A8H8ME73: Elongation factor Tu, Ceratobasidium sp. AG-Ba\n",
      "Fetched data for A0A286URG5: Elongation factor Tu, Pyrrhoderma noxium\n",
      "Fetched data for K5WPK8: Elongation factor Tu, Phanerochaete carnosa (strain HHB-10118-sp) (White-rot fungus) (Peniophora carnosa)\n",
      "No data found for UPI0004409335.\n",
      "Fetched data for A0A0L6W793: Elongation factor Tu, Termitomyces sp. J132\n",
      "Fetched data for A0A0C2T4D2: Elongation factor Tu, Amanita muscaria (strain Koide BX008)\n",
      "Fetched data for A0A369K5C3: Elongation factor Tu, Hypsizygus marmoreus (White beech mushroom) (Agaricus marmoreus)\n",
      "Fetched data for A0A164UBM0: Elongation factor Tu, Sistotremastrum niveocremeum HHB9708\n",
      "Fetched data for A8N1J6: Elongation factor Tu, Coprinopsis cinerea (strain Okayama-7 / 130 / ATCC MYA-4618 / FGSC 9003) (Inky cap fungus) (Hormographiella aspergillata)\n",
      "Fetched data for A0A8H5MDS0: Elongation factor Tu, Collybiopsis confluens\n",
      "Fetched data for K5WAB1: Elongation factor Tu, Agaricus bisporus var. burnettii (strain JB137-S8 / ATCC MYA-4627 / FGSC 10392) (White button mushroom)\n",
      "Fetched data for A0A0D2LMF0: Elongation factor Tu, Hypholoma sublateritium (strain FD-334 SS-4)\n",
      "Fetched data for A0A8H6W5X9: Elongation factor Tu, Mycena indigotica\n",
      "Fetched data for A0A5C3QYW4: Elongation factor Tu, Pterula gracilis\n",
      "Fetched data for A0A401G6V3: Elongation factor Tu, Sparassis crispa\n",
      "Fetched data for A0A8H6XF81: Elongation factor Tu, Mycena sanguinolenta\n",
      "No data found for UPI0019062C6A.\n",
      "Fetched data for A0A8H5FZW9: Elongation factor Tu, Leucoagaricus leucothites\n",
      "No data found for UPI002007C6E4.\n",
      "Fetched data for A0A0C3BG74: Elongation factor Tu, Piloderma croceum (strain F 1598)\n",
      "Fetched data for A0A8H5AZZ4: Elongation factor Tu, Psilocybe cf. subviscida\n",
      "Fetched data for A0A8H6TK30: Elongation factor Tu, Mycena kentingensis (nom. inval.)\n",
      "Fetched data for A0A8H5H174: Elongation factor Tu, Tetrapyrgos nigripes\n",
      "No UniRef100 ID found in C1_ETNmetMinimDraft_3_1059899.scaffolds.fasta_scaffold221833_1, skipping.\n",
      "No UniRef100 ID found in C1_A0A151WDW1, skipping.\n",
      "No UniRef100 ID found in C1_A0A137QRP7, skipping.\n",
      "No UniRef100 ID found in C1_A0A1Q3ELD6, skipping.\n",
      "No UniRef100 ID found in C1_A0A165VZY5, skipping.\n",
      "No UniRef100 ID found in C1_A0A1J8R6A5, skipping.\n",
      "No UniRef100 ID found in C1_A0A1C7MQM6, skipping.\n",
      "Fetched data for A0A4Y9YD42: Elongation factor Tu, Rhodofomes roseus\n",
      "No UniRef100 ID found in C1_J4GUS3, skipping.\n",
      "Fetched data for A0A1M4SS91: Elongation factor Tu (EF-Tu), Ferrithrix thermotolerans DSM 19514\n",
      "Fetched data for A0A6I2YU27: Elongation factor Tu (EF-Tu), Actinomycetes bacterium\n",
      "Fetched data for A0A7V9BXY0: Elongation factor Tu (EF-Tu), Actinomycetes bacterium\n",
      "Fetched data for A0A073B1S9: Elongation factor Tu (EF-Tu), Saccharopolyspora rectivirgula\n",
      "Fetched data for A0A257RFX4: Elongation factor Tu (EF-Tu), Actinobacteria bacterium 21-73-9\n",
      "Fetched data for A0A931KVY1: Elongation factor Tu (EF-Tu), Frankiales bacterium\n",
      "No data found for UPI0021CB779D.\n",
      "Fetched data for A0A538JUN6: Elongation factor Tu (EF-Tu), Actinomycetes bacterium\n",
      "Fetched data for A0A6I3H0D8: Elongation factor Tu (EF-Tu), Actinomycetes bacterium\n",
      "Fetched data for A0A0D8HL47: Elongation factor Tu (EF-Tu), Acidithrix ferrooxidans\n",
      "Fetched data for A0A9D8H9Z6: Elongation factor Tu (EF-Tu), Actinomycetota bacterium\n",
      "Fetched data for A0A7W0HE09: Elongation factor Tu (EF-Tu), Acidimicrobiia bacterium\n",
      "Fetched data for A0A7Y3LMS3: Elongation factor Tu (EF-Tu), Acidimicrobiales bacterium\n",
      "Fetched data for D3CUS3: Elongation factor Tu (EF-Tu), Parafrankia sp. EUN1f\n",
      "Fetched data for A0A7W1E1A4: Elongation factor Tu (EF-Tu), Actinomycetes bacterium\n",
      "Fetched data for A0A2E5ZGU2: Elongation factor Tu (EF-Tu), Acidimicrobiaceae bacterium\n",
      "Fetched data for A0A2U1FDK7: Elongation factor Tu (EF-Tu), Actinomycetospora cinnamomea\n",
      "Fetched data for A0A7J9Z5K4: Elongation factor Tu (EF-Tu), Streptosporangiales bacterium\n",
      "Fetched data for A0A966B5E8: Elongation factor Tu (EF-Tu), Actinomycetota bacterium\n",
      "Fetched data for A0A6B1H0D9: Elongation factor Tu (EF-Tu), Acidimicrobiia bacterium\n",
      "Fetched data for A0A060JN72: Elongation factor Tu (EF-Tu), Rhodoluna lacicola\n",
      "Fetched data for A0A849BFE4: Elongation factor Tu (EF-Tu), Pseudokineococcus marinus\n",
      "No data found for UPI00036FBCF2.\n",
      "Fetched data for A0A853BP90: Elongation factor Tu (EF-Tu), Streptomonospora nanhaiensis\n",
      "Fetched data for A0A938J874: Elongation factor Tu (EF-Tu), Actinomycetota bacterium\n",
      "Fetched data for A0A6I2X6Y2: Elongation factor Tu (EF-Tu), Actinomycetes bacterium\n",
      "No data found for UPI00234DF3D3.\n",
      "Fetched data for A0A935AY77: Elongation factor Tu (EF-Tu), Actinomycetota bacterium\n",
      "Fetched data for A0A7W0A023: Elongation factor Tu (EF-Tu), Actinomycetes bacterium\n",
      "Fetched data for A0A7Y0GW13: Elongation factor Tu (EF-Tu), Phycicoccus sp\n",
      "Fetched data for A0A538FG31: Elongation factor Tu (EF-Tu), Actinomycetes bacterium\n",
      "Fetched data for A0A970YZR4: Elongation factor Tu (EF-Tu), Actinomycetales bacterium\n",
      "Fetched data for A0A1M6X779: Elongation factor Tu (EF-Tu), Pseudonocardia thermophila\n",
      "No data found for A0A2W4QMS7.\n",
      "Fetched data for A0A6N9EZ57: Elongation factor Tu (EF-Tu), Acidimicrobiaceae bacterium\n",
      "No data found for UPI001296AE8D.\n",
      "No data found for A0A934Z554.\n",
      "Fetched data for A0A852VNY4: Elongation factor Tu (EF-Tu), Janibacter cremeus\n",
      "Fetched data for A0A0K1JM72: Elongation factor Tu (EF-Tu), Luteipulveratus mongoliensis\n",
      "No data found for UPI0011CB7315.\n",
      "Fetched data for A0A6I2ZFX3: Elongation factor Tu (EF-Tu), Actinomycetes bacterium\n",
      "Fetched data for A0A2R2W5W5: Elongation factor Tu (EF-Tu), Frondihabitans sp. 762G35\n",
      "Fetched data for A0A6N7I5Y3: Elongation factor Tu (EF-Tu), Streptosporangiales bacterium\n",
      "Fetched data for A0A938JSC1: Elongation factor Tu (EF-Tu), Actinomycetota bacterium\n",
      "Fetched data for A0A7J9VS26: Elongation factor Tu (EF-Tu), Propionibacteriales bacterium\n",
      "Fetched data for A0A6I6RWI3: Elongation factor Tu (EF-Tu), Streptomyces sp. GS7\n",
      "Fetched data for A0A7J9XLA5: Elongation factor Tu (EF-Tu), Pseudonocardiaceae bacterium\n",
      "No data found for UPI0013CFA3FB.\n",
      "Fetched data for A0A7T3ZWG6: Elongation factor Tu (EF-Tu), Brevibacterium casei\n",
      "No data found for UPI000C81862C.\n",
      "Fetched data for A0A2W4JPN0: Elongation factor Tu (EF-Tu), Thermocrispum agreste\n",
      "Fetched data for A0A4R2K0Y4: Elongation factor Tu (EF-Tu), Actinocrispum wychmicini\n",
      "Fetched data for A0A938YCV9: Elongation factor Tu (EF-Tu), Nakamurella leprariae\n",
      "Fetched data for A0A7Y9DK38: Elongation factor Tu (EF-Tu), Kineococcus aurantiacus\n",
      "Fetched data for A0A917U0U1: Elongation factor Tu (EF-Tu), Micromonospora sonchi\n",
      "Fetched data for A0A8J7KNB2: Elongation factor Tu (EF-Tu), Longispora fulva\n",
      "Fetched data for A0A3Q9UCN8: Elongation factor Tu (EF-Tu), Acidipropionibacterium jensenii\n",
      "Fetched data for D3QB76: Elongation factor Tu (EF-Tu), Stackebrandtia nassauensis (strain DSM 44728 / CIP 108903 / NRRL B-16338 / NBRC 102104 / LLR-40K-21)\n",
      "Fetched data for A0A0W8IKR5: Elongation factor Tu (EF-Tu), Nesterenkonia jeotgali\n",
      "Fetched data for S2WJY6: Elongation factor Tu (EF-Tu), Propionimicrobium lymphophilum ACS-093-V-SCH5\n",
      "No data found for UPI000D747D7B.\n",
      "Fetched data for A0A1G9Z043: Elongation factor Tu (EF-Tu), Allokutzneria albata (Kibdelosporangium albatum)\n",
      "No data found for A0A1L7F1D2.\n",
      "Fetched data for A0A388NRG8: Elongation factor Tu (EF-Tu), Actinomycetes bacterium\n",
      "Fetched data for A0A810KYL2: Elongation factor Tu (EF-Tu), Actinocatenispora sera\n",
      "Fetched data for E5XR78: Elongation factor Tu (EF-Tu), Segniliparus rugosus (strain ATCC BAA-974 / DSM 45345 / CCUG 50838 / CIP 108380 / JCM 13579 / CDC 945)\n",
      "Fetched data for A0A2G1XI85: Elongation factor Tu (EF-Tu), Streptomyces cinnamoneus (Streptoverticillium cinnamoneum)\n",
      "Fetched data for A0A286EN96: Elongation factor Tu (EF-Tu), Jatrophihabitans sp. GAS493\n",
      "Fetched data for A0A0H5D5X6: Elongation factor Tu (EF-Tu), Alloactinosynnema sp. L-07\n",
      "Fetched data for A0A916UM33: Elongation factor Tu (EF-Tu), Hoyosella rhizosphaerae\n",
      "Fetched data for A0A4Q9KFG4: Elongation factor Tu (EF-Tu), Propioniciclava sinopodophylli\n",
      "Fetched data for A0A6P2C7A7: Elongation factor Tu (EF-Tu), Trebonia kvetii\n",
      "No data found for UPI001C639208.\n",
      "Fetched data for W5WL60: Elongation factor Tu (EF-Tu), Kutzneria albida DSM 43870\n",
      "Fetched data for A0A1B1MMS2: Elongation factor Tu (EF-Tu), Streptomyces lincolnensis\n",
      "Fetched data for A0A383S3W6: Elongation factor Tu (EF-Tu), Propionibacterium australiense\n",
      "Fetched data for Q8FS84: Elongation factor Tu (EF-Tu), Corynebacterium efficiens (strain DSM 44549 / YS-314 / AJ 12310 / JCM 11189 / NBRC 100395)\n",
      "No data found for UPI000407305C.\n",
      "No data found for UPI0020A3CF92.\n",
      "Fetched data for A0A9D2ZRP5: Elongation factor Tu (EF-Tu), Candidatus Corynebacterium intestinavium\n",
      "No data found for UPI00142D4413.\n",
      "No data found for A0A6I1PGF2.\n",
      "No data found for UPI0003495350.\n",
      "Fetched data for A0A2W5IE43: Elongation factor Tu (EF-Tu), Lawsonella clevelandensis\n",
      "Fetched data for A0A9D6XNY2: Elongation factor Tu (EF-Tu), Candidatus Nanopelagicales bacterium\n",
      "No data found for UPI0012664527.\n",
      "Fetched data for A0A563E4C0: Elongation factor Tu (EF-Tu), Leekyejoonella antrihumi\n",
      "Fetched data for A0A542DJE1: Elongation factor Tu (EF-Tu), Amycolatopsis cihanbeyliensis\n",
      "No data found for UPI00211BF448.\n",
      "Fetched data for A0A838A7W8: Elongation factor Tu (EF-Tu), Haloechinothrix aidingensis\n",
      "Fetched data for A0A1G6HWD1: Elongation factor Tu (EF-Tu), Raineyella antarctica\n",
      "Fetched data for A0A7W1R2Z1: Elongation factor Tu (EF-Tu), Geodermatophilaceae bacterium\n",
      "Fetched data for A0A8I1S6W6: Elongation factor Tu (EF-Tu), Actinomycetes bacterium\n",
      "Fetched data for A0A6G8KZR5: Elongation factor Tu (EF-Tu), Brevibacterium luteolum\n",
      "Fetched data for A0A542ZSG3: Elongation factor Tu (EF-Tu), Propioniferax innocua\n",
      "Fetched data for I7L9W8: Elongation factor Tu (EF-Tu), Corynebacterium otitidis ATCC 51513\n",
      "No data found for A0A8J5B1N4.\n",
      "No data found for UPI0004207FC6.\n",
      "Fetched data for A0A0T1W7T8: Elongation factor Tu (EF-Tu), Mycobacterium sp. Root135\n",
      "No data found for UPI0004075CF2.\n",
      "No data found for A0A2W5LFA0.\n",
      "Fetched data for A0A255EBU4: Elongation factor Tu (EF-Tu), Parenemella sanctibonifatiensis\n",
      "No UniRef100 ID found in C1_SRR5262245_10410087, skipping.\n",
      "No UniRef100 ID found in C1_SRR5579883_2426184, skipping.\n",
      "No UniRef100 ID found in C1_SRR5271163_2306742, skipping.\n",
      "No UniRef100 ID found in C1_SRR5664280_2437162, skipping.\n",
      "No UniRef100 ID found in C1_SRR6476620_9283916, skipping.\n",
      "No UniRef100 ID found in C1_SRR5215218_1033559, skipping.\n",
      "No UniRef100 ID found in C1_SRR5829696_1816057, skipping.\n",
      "No UniRef100 ID found in C1_SRR5690348_16757388, skipping.\n",
      "No UniRef100 ID found in C1_SRR5512144_2849365, skipping.\n",
      "No UniRef100 ID found in C1_SRR4051812_832122, skipping.\n",
      "No UniRef100 ID found in C1_SRR3954454_2738150, skipping.\n",
      "No UniRef100 ID found in C1_SRR5215204_1196073, skipping.\n",
      "No UniRef100 ID found in C1_ERR687885_231633, skipping.\n",
      "No UniRef100 ID found in C1_23288|Ga0306923_10062805_1|-15|00, skipping.\n",
      "No UniRef100 ID found in C1_SRR5664280_1806184, skipping.\n",
      "No UniRef100 ID found in C1_SRR5215472_10858022, skipping.\n",
      "No UniRef100 ID found in C1_SRR5713101_4478130, skipping.\n",
      "No UniRef100 ID found in C1_SRR5438034_4674079, skipping.\n",
      "No UniRef100 ID found in C1_ERR687890_68762, skipping.\n",
      "No UniRef100 ID found in C1_SRR6266508_4112046, skipping.\n",
      "No UniRef100 ID found in C1_MGYP001122905865, skipping.\n",
      "No UniRef100 ID found in C1_ERR671911_550539, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000929341134, skipping.\n",
      "No UniRef100 ID found in C1_SRR6516162_5505792, skipping.\n",
      "No UniRef100 ID found in C1_SRR6516165_87006, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000843708847, skipping.\n",
      "No UniRef100 ID found in C1_SRR5450756_34688, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000934313783, skipping.\n",
      "No UniRef100 ID found in C1_SRR6201994_1604418, skipping.\n",
      "No UniRef100 ID found in C1_AntAceMinimDraft_11_1070367.scaffolds.fasta_scaffold169276_1, skipping.\n",
      "No UniRef100 ID found in C1_ERR687890_1522030, skipping.\n",
      "No UniRef100 ID found in C1_ERR1019366_7963191, skipping.\n",
      "No UniRef100 ID found in C1_SRR6266568_2810472, skipping.\n",
      "No UniRef100 ID found in C1_SRR5690625_3886352, skipping.\n",
      "No UniRef100 ID found in C1_ERR1700691_2052401, skipping.\n",
      "No UniRef100 ID found in C1_SRR6478609_7944864, skipping.\n",
      "No UniRef100 ID found in C1_SRR5690242_969664, skipping.\n",
      "No UniRef100 ID found in C1_3300008089.a:Ga0104978_1000470_24, skipping.\n",
      "No UniRef100 ID found in C1_HubBroStandDraft_1064217.scaffolds.fasta_scaffold256331_1, skipping.\n",
      "No UniRef100 ID found in C1_OpeIllAssembly_1097287.scaffolds.fasta_scaffold116668_1, skipping.\n",
      "No UniRef100 ID found in C1_ERR1700759_4143, skipping.\n",
      "No UniRef100 ID found in C1_ERR1035437_2611161, skipping.\n",
      "No UniRef100 ID found in C1_SRR5665647_1250305, skipping.\n",
      "No UniRef100 ID found in C1_SRR5580693_8119003, skipping.\n",
      "Fetched data for A0A7W3LLP6: Elongation factor Tu (EF-Tu), Actinomadura namibiensis\n",
      "Fetched data for A0A543IP01: Elongation factor Tu (EF-Tu), Actinomadura hallensis\n",
      "No data found for UPI0008295250.\n",
      "No data found for UPI0004BEDA83.\n",
      "Fetched data for A0A6H9YFH4: Elongation factor Tu (EF-Tu), Actinomadura rudentiformis\n",
      "No data found for UPI0011206B1A.\n",
      "Fetched data for A0A4R4M586: Elongation factor Tu (EF-Tu), Actinomadura sp. KC216\n",
      "Fetched data for A0A3M2LJI2: Elongation factor Tu (EF-Tu), Actinomadura harenae\n",
      "No data found for UPI0008312D56.\n",
      "No data found for UPI00164F9883.\n",
      "No data found for UPI0010416A35.\n",
      "Fetched data for A0A5D0NZ14: Elongation factor Tu (EF-Tu), Actinomadura chibensis\n",
      "No data found for A0A6I4PKE2.\n",
      "Fetched data for A0A6I4M993: Elongation factor Tu (EF-Tu), Actinomadura physcomitrii\n",
      "No data found for UPI00037713D4.\n",
      "No data found for A0A1Q9UEX4.\n",
      "Fetched data for A0A7K0BVX6: Elongation factor Tu (EF-Tu), Actinomadura macrotermitis\n",
      "No data found for UPI00104145D5.\n",
      "Fetched data for A0A1H6CUV6: Elongation factor Tu (EF-Tu), Thermomonospora echinospora\n",
      "Fetched data for A0A7W3R6L5: Elongation factor Tu (EF-Tu), Thermomonospora cellulosilytica\n",
      "Fetched data for A0A3D9SNF9: Elongation factor Tu (EF-Tu), Thermomonospora umbrina\n",
      "No UniRef100 ID found in C1_JI102314A2RNA_FD_contig_21_1525277_length_806_multi_7_in_0_out_0_2, skipping.\n",
      "No UniRef100 ID found in C1_MGYP001294425853, skipping.\n",
      "No UniRef100 ID found in C1_A0A238W5S0, skipping.\n",
      "Fetched data for A0A4R9EXQ9: Elongation factor Tu (EF-Tu), Streptomyces sp. MZ04\n",
      "Fetched data for A0A1Q4UYS8: Elongation factor Tu (EF-Tu), Streptomyces uncialis\n",
      "No data found for UPI00110FC54B.\n",
      "Fetched data for A0A7X3ACK6: Elongation factor Tu (EF-Tu), Streptomyces sp. Z38\n",
      "No data found for UPI000FCC3A36.\n",
      "Fetched data for A0A944KRH2: Elongation factor Tu (EF-Tu), Streptomyces sp. ISL-98\n",
      "Fetched data for A0A919DET7: Elongation factor Tu (EF-Tu), Streptomyces capitiformicae\n",
      "No data found for UPI001667E8CA.\n",
      "No data found for A0A7H8J5P6.\n",
      "Fetched data for A0A6G4XFJ0: Elongation factor Tu (EF-Tu), Streptomyces mesophilus\n",
      "Fetched data for A0A5C6JZ56: Elongation factor Tu (EF-Tu), Streptomyces misionensis\n",
      "Fetched data for A0A2U3C0W4: Elongation factor Tu, Streptomyces sp. ICBB 8177\n",
      "No data found for UPI002251C001.\n",
      "Fetched data for A0A0C5GA75: Elongation factor Tu, Streptomyces cyaneogriseus subsp. noncyanogenus\n",
      "Fetched data for A0A124ED28: Elongation factor Tu (EF-Tu), Streptomyces kanasensis\n",
      "Fetched data for V6JMA1: Elongation factor Tu (EF-Tu), Streptomyces roseochromogenus subsp. oscitans DS 12.976\n",
      "No data found for UPI00207C63B4.\n",
      "Fetched data for A0A918GNF4: Elongation factor Tu, Streptomyces niveoruber\n",
      "Fetched data for A0A7K0CB26: Elongation factor Tu, Streptomyces smaragdinus\n",
      "Fetched data for A0A7H0IMG2: Elongation factor Tu (EF-Tu), Streptomyces roseirectus\n",
      "No data found for A0A5C8GV14.\n",
      "No data found for UPI001C2DF3D4.\n",
      "No data found for UPI0022455C9C.\n",
      "No data found for UPI001F5CEBDD.\n",
      "No data found for UPI00225502EF.\n",
      "No data found for UPI001D07C8EB.\n",
      "No data found for UPI0008404D4E.\n",
      "No data found for UPI0020C176D6.\n",
      "Fetched data for A0A6G3RX88: Elongation factor Tu (EF-Tu), Streptomyces sp. SID14478\n",
      "Fetched data for A0A2S4XVP4: Elongation factor Tu (EF-Tu), Streptomyces sp. Ru73\n",
      "No data found for UPI00190D2F46.\n",
      "No data found for UPI001295EE6F.\n",
      "Fetched data for A0A6G4TSH5: Elongation factor Tu (EF-Tu), Streptomyces coryli\n",
      "Fetched data for A0A8T4IWN8: Elongation factor Tu (EF-Tu), Streptomyces daliensis\n",
      "No data found for UPI00211A8697.\n",
      "Fetched data for A0A365Z5T0: Elongation factor Tu (EF-Tu), Streptomyces sp. PT12\n",
      "No data found for UPI00078594B8.\n",
      "Fetched data for A0A1I2AXZ0: Elongation factor Tu, Actinacidiphila alni\n",
      "Fetched data for A0A505CYL6: Elongation factor Tu (EF-Tu), Streptomyces sporangiiformans\n",
      "Fetched data for A0A3D0R5B3: Elongation factor Tu (EF-Tu), Streptomyces sp\n",
      "No data found for UPI000A3900F5.\n",
      "No data found for UPI0022571BB3.\n",
      "No data found for UPI00056AD874.\n",
      "Fetched data for A0A370B0A1: Elongation factor Tu (EF-Tu), Streptomyces corynorhini\n",
      "Fetched data for A0A221NTP9: Elongation factor Tu (EF-Tu), Streptomyces pluripotens\n",
      "No data found for UPI001B37C0ED.\n",
      "Fetched data for A0A2P2GEE7: Elongation factor Tu, Streptomyces showdoensis\n",
      "Fetched data for A0A1H8NGK5: Elongation factor Tu, Actinacidiphila rubida\n",
      "No data found for UPI00041E10DB.\n",
      "No data found for UPI00124B4E65.\n",
      "No data found for UPI00089748CE.\n",
      "Fetched data for A0A6B3HEY9: Elongation factor Tu (EF-Tu), Streptomyces sp. SID8455\n",
      "No data found for UPI002034912F.\n",
      "Fetched data for A0A2M9KMZ3: Elongation factor Tu (EF-Tu), Streptomyces sp. CB02120-2\n",
      "No data found for UPI001E46A4F6.\n",
      "No data found for UPI0021769C4F.\n",
      "No data found for UPI00210B8B01.\n",
      "Fetched data for A0A939F4V4: Elongation factor Tu (EF-Tu), Streptomyces beijiangensis\n",
      "Fetched data for A0A1Y2NTG1: Elongation factor Tu (EF-Tu), Streptomyces fradiae ATCC 10745 = DSM 40063\n",
      "Fetched data for A0A918NHC8: Elongation factor Tu (EF-Tu), Streptomyces fructofermentans\n",
      "No data found for UPI00210BA91F.\n",
      "No data found for UPI0005BA3EF8.\n",
      "No data found for UPI001C8611CA.\n",
      "No data found for UPI0022B6F8E6.\n",
      "Fetched data for A0A372M4V6: Elongation factor Tu, Streptomyces triticagri\n",
      "No data found for A0A1A9QTC1.\n",
      "Fetched data for A0A918U6A5: Elongation factor Tu, Streptomyces minutiscleroticus\n",
      "Fetched data for A0A1V2RTZ0: Elongation factor Tu (EF-Tu), Streptomyces sp. MP131-18\n",
      "No data found for UPI0020203336.\n",
      "Fetched data for A0A3M0ICR4: Elongation factor Tu (EF-Tu), Streptomyces shenzhenensis\n",
      "Fetched data for A0A919GYQ9: Elongation factor Tu (EF-Tu), Streptomyces xanthophaeus\n",
      "Fetched data for S5UYY7: Elongation factor Tu (EF-Tu), Streptomyces collinus (strain DSM 40733 / Tue 365)\n",
      "Fetched data for A0A930ZX77: Elongation factor Tu (EF-Tu), Streptomyces olivaceus\n",
      "Fetched data for A0A8G1ZKR6: Elongation factor Tu (EF-Tu), Streptomyces albidoflavus\n",
      "No data found for UPI001F5DEF2D.\n",
      "No data found for UPI00226FC289.\n",
      "Fetched data for A0A1H0N0L2: Elongation factor Tu, Actinacidiphila guanduensis\n",
      "No data found for UPI000787D52E.\n",
      "Fetched data for A0A918MFV0: Elongation factor Tu (EF-Tu), Streptomyces filipinensis\n",
      "Fetched data for A0A5P2CC72: Elongation factor Tu (EF-Tu), Streptomyces venezuelae\n",
      "No data found for UPI0016794C66.\n",
      "No data found for UPI0019663E57.\n",
      "No data found for UPI00224EAE94.\n",
      "Fetched data for A0A5N8WYA0: Elongation factor Tu (EF-Tu), Streptomyces acidicola\n",
      "Fetched data for A0A7K3BKF0: Elongation factor Tu (EF-Tu), Streptomyces sp. SID8377\n",
      "Fetched data for A0A2R4SY40: Elongation factor Tu (EF-Tu), Streptomyces lunaelactis\n",
      "No data found for UPI0016792CFB.\n",
      "Fetched data for A0A345XQ45: Elongation factor Tu, Streptomyces armeniacus\n",
      "No data found for UPI001F142BC2.\n",
      "Fetched data for A0A2B8B284: Elongation factor Tu, Streptomyces sp. Ru87\n",
      "Fetched data for A0A4R4TEU6: Elongation factor Tu (EF-Tu), Streptomyces hainanensis\n",
      "No data found for UPI00191D979D.\n",
      "Fetched data for A0A7M3LV78: Elongation factor Tu (EF-Tu), Streptomyces sp. SAJ15\n",
      "No data found for UPI001CA713A4.\n",
      "No data found for UPI00224CF0A8.\n",
      "No data found for UPI001CFCFF83.\n",
      "Fetched data for A0A7K2WM35: Elongation factor Tu (EF-Tu), Streptomyces sp. SID3212\n",
      "Fetched data for A0A101UD17: Elongation factor Tu, Streptomyces sp. DSM 15324\n",
      "Fetched data for A0A7J0C1M1: Elongation factor Tu, Streptomyces fulvorobeus\n",
      "No data found for UPI001CEC1BD7.\n",
      "No data found for UPI002264C0FA.\n",
      "Fetched data for D7BY48: Elongation factor Tu (EF-Tu), Streptomyces bingchenggensis (strain BCW-1)\n",
      "No data found for UPI000FEBBAF1.\n",
      "No data found for UPI0004C73BBA.\n",
      "Fetched data for A0A0X3W0C3: Elongation factor Tu, Streptomyces sp. NRRL F-4489\n",
      "Fetched data for A0A0N1NII9: Elongation factor Tu (EF-Tu), Actinobacteria bacterium OK074\n",
      "Fetched data for A0A1H9BGA1: Elongation factor Tu, Streptomyces sp. yr375\n",
      "Fetched data for A0A1M7Q0J6: Elongation factor Tu, Actinacidiphila paucisporea\n",
      "No data found for UPI0022FEB82C.\n",
      "No data found for UPI0015EEF4A3.\n",
      "No data found for UPI002257756E.\n",
      "No data found for A0A0N0MU14.\n",
      "Fetched data for A0A7K3DNB6: Elongation factor Tu, Streptomyces sp. SID5785\n",
      "Fetched data for A3KKP5: Elongation factor Tu (EF-Tu), Streptomyces ambofaciens (strain ATCC 23877 / 3486 / DSM 40053 / JCM 4204 / NBRC 12836 / NRRL B-2516)\n",
      "Fetched data for A0A0F7G286: Elongation factor Tu, Streptomyces xiamenensis\n",
      "No data found for UPI001C3890DA.\n",
      "No data found for UPI001E38ED79.\n",
      "Fetched data for A0A1H6CDW4: Elongation factor Tu, Actinacidiphila yanglinensis\n",
      "Fetched data for A0A387HNB4: Elongation factor Tu (EF-Tu), Streptomyces hundungensis\n",
      "Fetched data for A0A919BZ14: Elongation factor Tu (EF-Tu), Streptomyces capoamus\n",
      "Fetched data for A0A6B2RYS8: Elongation factor Tu (EF-Tu), Streptomyces sp. SID10853\n",
      "No data found for UPI0022AFDDC8.\n",
      "Fetched data for A0A7T1TBV1: Elongation factor Tu (EF-Tu), Streptomyces bathyalis\n",
      "No data found for UPI001F21F80B.\n",
      "No data found for UPI0021134B37.\n",
      "No data found for UPI002026C8E8.\n",
      "No data found for UPI00225B2F22.\n",
      "No data found for UPI001B362A83.\n",
      "Fetched data for A0A5B8IC35: Elongation factor Tu, Streptomyces qinzhouensis\n",
      "Fetched data for A0A3R9XQG8: Elongation factor Tu, Streptomyces sp. WAC05292\n",
      "Fetched data for A0A919L5M4: Elongation factor Tu, Streptomyces sulfonofaciens\n",
      "No data found for UPI001BEBD335.\n",
      "Fetched data for A0A344U3D7: Elongation factor Tu, Streptomyces globosus\n",
      "No data found for UPI00210CA4C4.\n",
      "Fetched data for A0A919HD98: Elongation factor Tu, Streptomyces sp. TS71-3\n",
      "No data found for UPI00125029F7.\n",
      "Fetched data for A0A974AYX1: Elongation factor Tu, Streptomyces sp. PKU-EA00015\n",
      "No data found for UPI001009E365.\n",
      "No data found for UPI001FF88ACF.\n",
      "Fetched data for A0A4Z0HB80: Elongation factor Tu, Streptomyces palmae\n",
      "No data found for UPI001905947A.\n",
      "Fetched data for F3NTJ2: Elongation factor Tu, Streptomyces griseoaurantiacus M045\n",
      "No data found for A0A917SNY4.\n",
      "No data found for UPI001F039099.\n",
      "No UniRef100 ID found in C1_SRR3954447_22124948, skipping.\n",
      "No UniRef100 ID found in C1_A0A1H6CDW4, skipping.\n",
      "No UniRef100 ID found in C1_SRR5690349_12166511, skipping.\n",
      "Fetched data for A0A4R3YAC7: Elongation factor Tu (EF-Tu), Testudinibacter aquarius\n",
      "No data found for UPI001CE2CE66.\n",
      "Fetched data for A0A143WPR6: Elongation factor Tu (EF-Tu), Candidatus Gullanella endobia\n",
      "Fetched data for A0A209AQB3: Elongation factor Tu (EF-Tu), Yersinia frederiksenii\n",
      "Fetched data for U3TY99: Elongation factor Tu (EF-Tu), Plautia stali symbiont\n",
      "Fetched data for A0A1N7CGP3: Elongation factor Tu (EF-Tu), Aeromonas sp. RU39B\n",
      "Fetched data for A0A5P2UM61: Elongation factor Tu (EF-Tu), Streptomyces subrutilus\n",
      "Fetched data for A0A1V8NCP4: Elongation factor Tu (EF-Tu), bacterium endosymbiont of Pedicinus badii\n",
      "Fetched data for A0A4D6YLG4: Elongation factor Tu (EF-Tu), Buchnera aphidicola (Stegophylla sp.)\n",
      "Fetched data for A0A420HRM0: Elongation factor G, mitochondrial (EF-Gmt) (Elongation factor G 1, mitochondrial) (mEF-G 1) (Elongation factor G1), Golovinomyces cichoracearum\n",
      "Fetched data for F7XY84: Elongation factor Tu (EF-Tu), Moranella endobia (strain PCIT)\n",
      "No data found for UPI00141E91C6.\n",
      "No data found for UPI001C7DFAB7.\n",
      "Fetched data for A0A2C8CVX6: Elongation factor Tu (EF-Tu), Sodalis endosymbiont of Henestaris halophilus\n",
      "Fetched data for A0A4D6YN62: Elongation factor Tu (EF-Tu), Buchnera aphidicola (Therioaphis trifolii)\n",
      "Fetched data for Q1LSY4: Elongation factor Tu (EF-Tu), Baumannia cicadellinicola subsp. Homalodisca coagulata\n",
      "Fetched data for Q7VRP0: Elongation factor Tu (EF-Tu), Blochmanniella floridana\n",
      "Fetched data for A0A0E3U3M2: Elongation factor Tu (EF-Tu), Blochmannia endosymbiont of Polyrhachis (Hedomyrma) turneri\n",
      "Fetched data for D5BVA9: Elongation factor Tu (EF-Tu), Nitrosococcus halophilus (strain Nc4)\n",
      "Fetched data for A0A1C7AF74: Elongation factor Tu (EF-Tu), Sulfurifustis variabilis\n",
      "Fetched data for A0A9C9VER6: Elongation factor Tu (EF-Tu), Gammaproteobacteria bacterium\n",
      "Fetched data for L0DYY6: Elongation factor Tu (EF-Tu), Thioalkalivibrio nitratireducens (strain DSM 14787 / UNIQEM 213 / ALEN2)\n",
      "Fetched data for A0A1Y0NA39: Elongation factor Tu (EF-Tu), Curvibacter sp. AEP1-3\n",
      "No data found for UPI000F843A59.\n",
      "Fetched data for A0A1C3L4A1: Elongation factor Tu (EF-Tu), secondary endosymbiont of Trabutina mannipara\n",
      "No data found for UPI00054EF50E.\n",
      "Fetched data for A0A3S8T3C9: Elongation factor Tu (EF-Tu), Rickettsiales endosymbiont of Stachyamoeba lipophora\n",
      "Fetched data for A0A450TB23: Elongation factor Tu, Candidatus Kentron sp. FM\n",
      "No data found for UPI0003F5B7E5.\n",
      "Fetched data for A0A6S6UKH2: Elongation factor Tu (EF-Tu), uncultured Thiotrichaceae bacterium\n",
      "Fetched data for A0A949D297: Elongation factor Tu (EF-Tu), Lysobacteraceae bacterium\n",
      "Fetched data for A0A9D0P1A1: Elongation factor Tu (EF-Tu), Gammaproteobacteria bacterium\n",
      "Fetched data for A0A963D6S8: Elongation factor Tu (EF-Tu), Rhodocyclaceae bacterium\n",
      "Fetched data for A0A2A4UKP8: Elongation factor Tu (EF-Tu), Moraxellaceae bacterium\n",
      "Fetched data for A0A2E8KCX4: Elongation factor Tu (EF-Tu), Gammaproteobacteria bacterium\n",
      "No data found for UPI0012BD7EF3.\n",
      "No data found for UPI002227934A.\n",
      "Fetched data for A0A1U9JI97: Elongation factor Tu (EF-Tu), Aquaspirillum sp. LM1\n",
      "Fetched data for A0P4F9: Elongation factor Tu (EF-Tu), Methylophilales bacterium HTCC2181\n",
      "No data found for UPI0008D9A67A.\n",
      "Fetched data for D9SCC2: Elongation factor Tu (EF-Tu), Gallionella capsiferriformans (strain ES-2) (Gallionella ferruginea capsiferriformans (strain ES-2))\n",
      "No data found for A0A8T6JK40.\n",
      "Fetched data for A0A1G0K9T7: Elongation factor Tu, Gammaproteobacteria bacterium RIFCSPLOWO2_12_FULL_52_10\n",
      "Fetched data for A0A086WEB3: Elongation factor Tu (EF-Tu), Massilia sp. BSC265\n",
      "Fetched data for A0A7X1P490: Elongation factor Tu (EF-Tu), Betaproteobacteria bacterium\n",
      "No data found for UPI0022648D29.\n",
      "Fetched data for A0A2A2EQH0: Elongation factor Tu (EF-Tu), Halovibrio salipaludis\n",
      "Fetched data for A0A9D7KDJ5: Elongation factor Tu (EF-Tu), Nitrosomonadales bacterium\n",
      "Fetched data for A0A7D4SXK4: Elongation factor Tu (EF-Tu), Thiomicrorhabdus xiamenensis\n",
      "Fetched data for A0A661EVB5: Elongation factor Tu (EF-Tu), Gammaproteobacteria bacterium\n",
      "Fetched data for A0A536YJ92: Elongation factor Tu (EF-Tu), Betaproteobacteria bacterium\n",
      "Fetched data for A0A521XAZ9: Elongation factor Tu (EF-Tu), Betaproteobacteria bacterium\n",
      "No data found for UPI0005C24092.\n",
      "Fetched data for A0A212KMK6: Elongation factor Tu (EF-Tu), uncultured Alphaproteobacteria bacterium\n",
      "Fetched data for A0A836S6B4: Elongation factor Tu (EF-Tu), Thiotrichales bacterium\n",
      "No data found for UPI0018DCDA65.\n",
      "No data found for UPI001E421E4F.\n",
      "Fetched data for A0A6N6VIM4: Elongation factor Tu (EF-Tu), Parvibaculum sedimenti\n",
      "Fetched data for A0A9D8CB88: Elongation factor Tu (EF-Tu), Alphaproteobacteria bacterium\n",
      "No data found for A0A6N3S231.\n",
      "Fetched data for A0A2E4GJ30: Elongation factor Tu (EF-Tu), Gammaproteobacteria bacterium\n",
      "Fetched data for A0A1H5XFN9: Elongation factor Tu (EF-Tu), Jhaorihella thermophila\n",
      "Fetched data for Q9P9Q9: Elongation factor Tu (EF-Tu), Xylella fastidiosa (strain 9a5c)\n",
      "Fetched data for A0A1Q6UCA4: Elongation factor Tu (EF-Tu), Azospirillum sp. 51_20\n",
      "Fetched data for A0A061QEV0: Elongation factor Tu (EF-Tu), alpha proteobacterium Q-1\n",
      "Fetched data for A0A7W5ELG9: Elongation factor Tu (EF-Tu), Luteibacter sp. Sphag1AF\n",
      "Fetched data for A0A7R6W0L4: Elongation factor Tu (EF-Tu), Candidatus Profftella armatura (Diaphorina cf. continua)\n",
      "Fetched data for A0A1N7K9D5: Elongation factor Tu (EF-Tu), Insolitispirillum peregrinum\n",
      "Fetched data for A0A3B0YSK9: Translation elongation factor Tu, hydrothermal vent metagenome\n",
      "Fetched data for U2E2F6: Elongation factor Tu (EF-Tu), Salinisphaera shabanensis E1L3A\n",
      "Fetched data for A0A7V8AKP9: Elongation factor Tu (EF-Tu), Rhodospirillaceae bacterium\n",
      "Fetched data for A0A2E7K4G9: Elongation factor Tu (EF-Tu), Rickettsiales bacterium\n",
      "Fetched data for A0A1Z8KXW1: Elongation factor Tu (EF-Tu), Micavibrio sp. TMED2\n",
      "Fetched data for A0A2D7AAY0: Elongation factor Tu (EF-Tu), Magnetovibrio sp\n",
      "Fetched data for A0A0Q4NEQ5: Elongation factor Tu (EF-Tu), Pseudomonas sp. Leaf58\n",
      "Fetched data for M4S4E3: Elongation factor Tu (EF-Tu), Sphingomonas sp. MM-1\n",
      "Fetched data for A0A3A1Y0W9: Elongation factor Tu, Psittacicella hinzii\n",
      "No data found for UPI000364F5A5.\n",
      "Fetched data for A0A2N6N749: Elongation factor Tu (EF-Tu), Coxiella-like endosymbiont\n",
      "Fetched data for A0A6L5G3T8: Elongation factor Tu, Actinomycetes bacterium\n",
      "Fetched data for A0A523FG19: Elongation factor Tu (EF-Tu), Alphaproteobacteria bacterium\n",
      "No data found for UPI0007BED29B.\n",
      "No data found for UPI001FCF9EFA.\n",
      "Fetched data for A0A6L7S3G9: Elongation factor Tu (EF-Tu), Boseongicola sp. SB0662_bin_57\n",
      "Fetched data for A0A523FVQ2: Elongation factor Tu (EF-Tu), Alphaproteobacteria bacterium\n",
      "No data found for UPI00037579BE.\n",
      "Fetched data for A0A2S6T5Q0: Elongation factor Tu (EF-Tu), Alphaproteobacteria bacterium MarineAlpha5_Bin2\n",
      "Fetched data for A0A9E1RA41: Elongation factor Tu (EF-Tu), Oceanospirillaceae bacterium\n",
      "No data found for A0A481QLI6.\n",
      "Fetched data for A0A9D8GMY6: Elongation factor Tu (EF-Tu), Alphaproteobacteria bacterium\n",
      "Fetched data for A0A2E5DE46: Elongation factor Tu (EF-Tu), Gammaproteobacteria bacterium\n",
      "Fetched data for A0A0Q6BEW9: Elongation factor Tu (EF-Tu), Methylophilus sp. Leaf416\n",
      "No data found for A0A812UR29.\n",
      "Fetched data for A0A2E5DWF1: Elongation factor Tu (EF-Tu), Alphaproteobacteria bacterium\n",
      "Fetched data for A0A5C7SRI2: Elongation factor Tu (EF-Tu), Thauera aminoaromatica\n",
      "No data found for UPI0019397348.\n",
      "Fetched data for A0A6B0XYM7: Elongation factor Tu (EF-Tu), Boseongicola sp. SB0664_bin_43\n",
      "No data found for UPI00082EB59B.\n",
      "Fetched data for A0A2E4GZN0: Elongation factor Tu (EF-Tu), Rhodospirillaceae bacterium\n",
      "Fetched data for M5E919: Elongation factor Tu (EF-Tu), Thalassolituus oleivorans MIL-1\n",
      "Fetched data for A0A1R4HHK1: Elongation factor Tu (EF-Tu), Crenothrix polyspora\n",
      "Fetched data for A0A6L8C4X0: Elongation factor Tu (EF-Tu), Paracoccaceae bacterium\n",
      "Fetched data for A0A4R5Q2H1: Elongation factor Tu (EF-Tu), Dankookia rubra\n",
      "Fetched data for A0A6N6JNP2: Elongation factor Tu (EF-Tu), Litoreibacter roseus\n",
      "Fetched data for A0A0S8BXZ4: Elongation factor Tu (EF-Tu), Nitrospira bacterium SG8_3\n",
      "Fetched data for I3TNZ7: Elongation factor Tu (EF-Tu), Tistrella mobilis (strain KA081020-065)\n",
      "No data found for UPI0019350FF0.\n",
      "Fetched data for A0A0Q4QUV2: Elongation factor Tu (EF-Tu), Stenotrophomonas sp. Leaf70\n",
      "Fetched data for A0A7V8W6U5: Elongation factor Tu (EF-Tu), Nitrospirales bacterium\n",
      "Fetched data for A0A6S6SHZ6: Elongation factor Tu, uncultured Thiotrichaceae bacterium\n",
      "Fetched data for A0A0Q5DC00: Elongation factor Tu (EF-Tu), Rhizobium sp. Leaf306\n",
      "No UniRef100 ID found in C1_SRR5467141_2292314, skipping.\n",
      "No UniRef100 ID found in C1_SRR5512137_1872836, skipping.\n",
      "No UniRef100 ID found in C1_SRR5580658_8520063, skipping.\n",
      "No UniRef100 ID found in C1_SRR5215470_10407445, skipping.\n",
      "No UniRef100 ID found in C1_MDTC01.3.fsa_nt_gb|MDTC01266932.1|_2, skipping.\n",
      "No UniRef100 ID found in C1_SRR5512147_2539663, skipping.\n",
      "No UniRef100 ID found in C1_SRR6478736_1044114, skipping.\n",
      "No UniRef100 ID found in C1_SRR6266568_2227211, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000748906589, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000403898526, skipping.\n",
      "No UniRef100 ID found in C1_SRR5674476_588408, skipping.\n",
      "No UniRef100 ID found in C1_SRR5882724_1869979, skipping.\n",
      "No UniRef100 ID found in C1_SRR3954469_24122091, skipping.\n",
      "No UniRef100 ID found in C1_MGYP001451236428, skipping.\n",
      "No UniRef100 ID found in C1_SRR5512144_2416837, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000995708373, skipping.\n",
      "No UniRef100 ID found in C1_SRR6478609_326210, skipping.\n",
      "No UniRef100 ID found in C1_SRR5450830_1738083, skipping.\n",
      "No UniRef100 ID found in C1_A6B9V5, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000619146406, skipping.\n",
      "No UniRef100 ID found in C1_SRR5579864_3114098, skipping.\n",
      "No UniRef100 ID found in C1_SRR5262245_50471884, skipping.\n",
      "No UniRef100 ID found in C1_ERR1051326_256254, skipping.\n",
      "No UniRef100 ID found in C1_SRR6202167_3212991, skipping.\n",
      "No UniRef100 ID found in C1_SRR6516165_8490383, skipping.\n",
      "No UniRef100 ID found in C1_SRR5690242_17847510, skipping.\n",
      "No UniRef100 ID found in C1_ERR1700755_3614217, skipping.\n",
      "No UniRef100 ID found in C1_SRR5213080_1272439, skipping.\n",
      "No UniRef100 ID found in C1_SRR5438132_868919, skipping.\n",
      "No UniRef100 ID found in C1_ERR671937_1717116, skipping.\n",
      "No UniRef100 ID found in C1_SRR6266571_3081274, skipping.\n",
      "No UniRef100 ID found in C1_SRR5690554_3999590, skipping.\n",
      "No UniRef100 ID found in C1_SRR5260364_156372, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000654562554, skipping.\n",
      "No UniRef100 ID found in C1_SRR3954447_7852418, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000737074068, skipping.\n",
      "No UniRef100 ID found in C1_SRR5574338_572072, skipping.\n",
      "No UniRef100 ID found in C1_ERR1700692_169334, skipping.\n",
      "No UniRef100 ID found in C1_SRR5580658_8102326, skipping.\n",
      "No UniRef100 ID found in C1_SRR5882757_159431, skipping.\n",
      "No UniRef100 ID found in C1_ERR1700712_4973307, skipping.\n",
      "No UniRef100 ID found in C1_MGYP001014076289, skipping.\n",
      "No UniRef100 ID found in C1_SRR6266568_3801555, skipping.\n",
      "No UniRef100 ID found in C1_MGYP001145577361, skipping.\n",
      "No UniRef100 ID found in C1_SRR5712671_5841784, skipping.\n",
      "No UniRef100 ID found in C1_SRR3954466_9420994, skipping.\n",
      "No UniRef100 ID found in C1_GraSoiStandDraft_37_1057305.scaffolds.fasta_scaffold1862948_1, skipping.\n",
      "No UniRef100 ID found in C1_SRR6516225_9612021, skipping.\n",
      "No UniRef100 ID found in C1_SRR5919205_1093717, skipping.\n",
      "No UniRef100 ID found in C1_ERR1051325_10649185, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000737109569, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000916167949, skipping.\n",
      "No UniRef100 ID found in C1_SRR6218665_3751821, skipping.\n",
      "No UniRef100 ID found in C1_MGYP001472276601, skipping.\n",
      "No UniRef100 ID found in C1_SRR6266481_4710259, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000647887413, skipping.\n",
      "No UniRef100 ID found in C1_ERR1700760_3355059, skipping.\n",
      "No UniRef100 ID found in C1_SRR6266513_2879973, skipping.\n",
      "No UniRef100 ID found in C1_SRR5437868_232562, skipping.\n",
      "No UniRef100 ID found in C1_1329|Ga0228676_1007809_2|+649|00, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000145849978, skipping.\n",
      "No UniRef100 ID found in C1_MGYP001340219270, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000102417299, skipping.\n",
      "No UniRef100 ID found in C1_MGYP001248235180, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000032030495, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000055235457, skipping.\n",
      "No UniRef100 ID found in C1_YNPBryBLVA2012_1023415.scaffolds.fasta_scaffold116735_1, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000082511344, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000909241566, skipping.\n",
      "No UniRef100 ID found in C1_MGYP001159490990, skipping.\n",
      "No UniRef100 ID found in C1_SRR5438445_3297928, skipping.\n",
      "No UniRef100 ID found in C1_SRR6218665_1992995, skipping.\n",
      "No UniRef100 ID found in C1_MGYP001362051158, skipping.\n",
      "No UniRef100 ID found in C1_ERR1700738_1333687, skipping.\n",
      "No UniRef100 ID found in C1_SRR5436853_1415635, skipping.\n",
      "No UniRef100 ID found in C1_MGYP001434031679, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000856856587, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000013152588, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000424538954, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000886781026, skipping.\n",
      "No UniRef100 ID found in C1_SRR5215813_949092, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000209313799, skipping.\n",
      "No UniRef100 ID found in C1_SRR5690242_8736817, skipping.\n",
      "No UniRef100 ID found in C1_MGYP001206855197, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000341272244, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000054351996, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000106083303, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000732881605, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000941522905, skipping.\n",
      "No UniRef100 ID found in C1_SRR5215472_3836791, skipping.\n",
      "No UniRef100 ID found in C1_SRR5680860_270960, skipping.\n",
      "No UniRef100 ID found in C1_SRR6266516_1246524, skipping.\n",
      "No UniRef100 ID found in C1_EndMetStandDraft_9_1072997.scaffolds.fasta_scaffold3821344_1, skipping.\n",
      "No UniRef100 ID found in C1_MGYP001013366152, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000906913082, skipping.\n",
      "No UniRef100 ID found in C1_SRR6266513_75382, skipping.\n",
      "No UniRef100 ID found in C1_MGYP001232857242, skipping.\n",
      "No UniRef100 ID found in C1_MGYP001211435984, skipping.\n",
      "No UniRef100 ID found in C1_GraSoiStandDraft_14_1057315.scaffolds.fasta_scaffold3062645_1, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000389787795, skipping.\n",
      "No UniRef100 ID found in C1_MGYP000572681120, skipping.\n",
      "No UniRef100 ID found in C1_ERR1700751_2774943, skipping.\n",
      "No UniRef100 ID found in C1_MGYP001218116943, skipping.\n",
      "No UniRef100 ID found in C1_MGYP001235772237, skipping.\n",
      "No data found for UPI0019450773.\n",
      "Fetched data for E9UZA8: Elongation factor Tu (EF-Tu), Nocardioidaceae bacterium Broad-1\n",
      "Fetched data for A0A810MQ54: Elongation factor Tu (EF-Tu), Polymorphospora rubra\n",
      "Fetched data for A0A7Y9ZI65: Elongation factor Tu (EF-Tu), Nocardioides aromaticivorans\n",
      "No data found for UPI00223716A1.\n",
      "No data found for UPI00196713DE.\n",
      "No data found for UPI001C69B59B.\n",
      "Fetched data for A0A853BXE2: Elongation factor Tu (EF-Tu), Nocardioides thalensis\n",
      "Fetched data for A0A8J4DQ16: Elongation factor Tu (EF-Tu), Virgisporangium aliadipatigenens\n",
      "Fetched data for A0A4V6PAK6: Elongation factor Tu (EF-Tu), Kribbella albertanoniae\n",
      "Fetched data for A0A561BKB4: Elongation factor Tu (EF-Tu), Kribbella amoyensis\n",
      "No data found for UPI001945AF66.\n",
      "No data found for UPI000526337F.\n",
      "Fetched data for A0A849DU62: Elongation factor Tu (EF-Tu), Dactylosporangium sp\n",
      "No data found for UPI001129561B.\n",
      "No data found for UPI000490B98C.\n",
      "No UniRef100 ID found in C2_5803|scaffold_9666_c1_1|+3|10, skipping.\n",
      "No UniRef100 ID found in C2_SRR6266508_3827042, skipping.\n",
      "No UniRef100 ID found in C2_SRR6266540_428292, skipping.\n",
      "No UniRef100 ID found in C2_MKWU01.1.fsa_nt_gb|MKWU01015139.1|_3, skipping.\n",
      "Fetched data for G7E513: Tr-type G domain-containing protein, Mixia osmundae (strain CBS 9802 / IAM 14324 / JCM 22182 / KY 12970)\n",
      "No UniRef100 ID found in C3_MGYP000691294331, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712230_259613, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712121_207945, skipping.\n",
      "No UniRef100 ID found in C3_T1ECZ4, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719277_2884238, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712035_50463, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719163_2223308, skipping.\n",
      "No data found for UPI000BFA19E4.\n",
      "No UniRef100 ID found in C3_ERR1719357_257174, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719259_907208, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712070_1202835, skipping.\n",
      "No UniRef100 ID found in C3_A0A0C5Q2A5, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712173_309211, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712212_51913, skipping.\n",
      "No UniRef100 ID found in C3_A0A0A1T913, skipping.\n",
      "No UniRef100 ID found in C3_SoiMethySBSTD1v2_1073268.scaffolds.fasta_scaffold2460889_2, skipping.\n",
      "No UniRef100 ID found in C3_ERR550539_2026888, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712170_300451, skipping.\n",
      "No UniRef100 ID found in C3_A0A2B7WKU2, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719363_51645, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711953_724444, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712137_253264, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719370_437258, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719328_174243, skipping.\n",
      "No UniRef100 ID found in C3_MGYP001078517979, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719214_184921, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719495_1355545, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719379_1942114, skipping.\n",
      "No UniRef100 ID found in C3_GraSoiStandDraft_48_1057284.scaffolds.fasta_scaffold4356905_1, skipping.\n",
      "No UniRef100 ID found in C3_ERR1740123_50045, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719328_358938, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719471_517893, skipping.\n",
      "No UniRef100 ID found in C3_GraSoiStandDraft_37_1057305.scaffolds.fasta_scaffold1657734_1, skipping.\n",
      "No UniRef100 ID found in C3_SRR3984957_13698453, skipping.\n",
      "No UniRef100 ID found in C3_Wag4MinimDraft_6_1082665.scaffolds.fasta_scaffold355259_1, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712018_575482, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712072_1314791, skipping.\n",
      "No data found for UPI000D181BB9.\n",
      "No UniRef100 ID found in C3_ERR1719328_43991, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712048_551657, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719284_552227, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712135_106854, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719204_2517984, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719411_1586778, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719471_384962, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719309_1020071, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711939_830286, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711976_786798, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719471_1612675, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719369_1790055, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711881_76877, skipping.\n",
      "No UniRef100 ID found in C3_SoiMetStandDraft_2_1073263.scaffolds.fasta_scaffold1964892_2, skipping.\n",
      "No UniRef100 ID found in C3_SaaInlStandDraft_3_1057020.scaffolds.fasta_scaffold552283_1, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719443_1678612, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712015_111814, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719411_1670428, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719282_1062852, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719251_52923, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712226_1493702, skipping.\n",
      "No UniRef100 ID found in C3_ERR1740123_2017266, skipping.\n",
      "No UniRef100 ID found in C3_MGYP001471933186, skipping.\n",
      "No UniRef100 ID found in C3_A0A286Q513, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712080_771092, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712071_16236, skipping.\n",
      "No UniRef100 ID found in C3_A0A2A6BV15, skipping.\n",
      "No UniRef100 ID found in C3_APIni6443716594_1056825.scaffolds.fasta_scaffold10188738_1, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719471_915536, skipping.\n",
      "No UniRef100 ID found in C3_JI71714B2RNA_FD_contig_21_4827968_length_379_multi_2_in_0_out_0_1, skipping.\n",
      "No UniRef100 ID found in C3_GraSoiStandDraft_17_1057272.scaffolds.fasta_scaffold1283440_1, skipping.\n",
      "No UniRef100 ID found in C3_N1J6M9, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711937_217669, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719276_716244, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719290_78806, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719447_1262202, skipping.\n",
      "No UniRef100 ID found in C3_J4H2Q8, skipping.\n",
      "No UniRef100 ID found in C3_N6TAF9, skipping.\n",
      "No UniRef100 ID found in C3_A0A286ZIH3, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719348_556853, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719447_2765197, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712156_194537, skipping.\n",
      "No UniRef100 ID found in C3_A0A162UQS2, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719193_202432, skipping.\n",
      "No UniRef100 ID found in C3_ERR1739838_534159, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711893_179513, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712099_48551, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719427_1749156, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712029_361439, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719427_1076034, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712141_848103, skipping.\n",
      "No UniRef100 ID found in C3_ERR550534_3423021, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719180_140660, skipping.\n",
      "No UniRef100 ID found in C3_A0A1S8VJ52, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712019_146163, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712231_3162, skipping.\n",
      "No UniRef100 ID found in C3_ERR550539_43419, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719369_1038439, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712183_111540, skipping.\n",
      "No UniRef100 ID found in C3_U5HJE9, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711920_341137, skipping.\n",
      "No UniRef100 ID found in C3_RhiMetStandDraft_8_1073273.scaffolds.fasta_scaffold1309123_1, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712238_522808, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711931_605665, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719400_1568999, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712156_519695, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719166_326363, skipping.\n",
      "No UniRef100 ID found in C3_ERR1739838_736772, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719429_350253, skipping.\n",
      "No UniRef100 ID found in C3_ERR550534_1076433, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712000_726985, skipping.\n",
      "No UniRef100 ID found in C3_R9AB31, skipping.\n",
      "No UniRef100 ID found in C3_ERR1739836_294433, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719238_1543150, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719376_1347782, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719336_651042, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712177_104516, skipping.\n",
      "No data found for UPI000239E012.\n",
      "No UniRef100 ID found in C3_A0A1I8GGZ6, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719315_352075, skipping.\n",
      "No UniRef100 ID found in C3_ERR1740128_227212, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719427_208140, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719334_195979, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712183_960718, skipping.\n",
      "No UniRef100 ID found in C3_APPan5920702856_1055754.scaffolds.fasta_scaffold1786076_1, skipping.\n",
      "No UniRef100 ID found in C3_A0A0M8P155, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712168_1360086, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712227_406349, skipping.\n",
      "No UniRef100 ID found in C3_A0A0F8UX52, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719461_2337610, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719427_2067619, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719283_505325, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712212_885977, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719383_817962, skipping.\n",
      "No UniRef100 ID found in C3_A0A1Y2FDW2, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712168_1559660, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712141_128387, skipping.\n",
      "No UniRef100 ID found in C3_ERR550534_645831, skipping.\n",
      "No UniRef100 ID found in C3_ERR550534_1661866, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712227_310035, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711970_771759, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712226_1606703, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712159_517586, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712213_154862, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719427_2676329, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712045_119342, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711997_1434713, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711937_16863, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719376_851866, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719219_353973, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712003_590569, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719163_627935, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719222_115112, skipping.\n",
      "No UniRef100 ID found in C3_ERR1739845_269634, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719397_1283017, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712202_40957, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719480_676858, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719515_253173, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719419_216919, skipping.\n",
      "No UniRef100 ID found in C3_ERR550519_608383, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712226_1633155, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711973_943402, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712168_469734, skipping.\n",
      "No UniRef100 ID found in C3_M7AUR2, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711976_1085826, skipping.\n",
      "No UniRef100 ID found in C3_W3VHW1, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719427_1370934, skipping.\n",
      "No UniRef100 ID found in C3_ERR550532_3975571, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719187_572309, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711929_48273, skipping.\n",
      "No UniRef100 ID found in C3_ERR550539_1631074, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719282_1494699, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711970_863997, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719468_364260, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711970_1261788, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712221_12702, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719447_2100285, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711909_231537, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712174_173896, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719188_2649105, skipping.\n",
      "No UniRef100 ID found in C3_ERR1739838_241387, skipping.\n",
      "No UniRef100 ID found in C3_ERR550534_1164636, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712036_108805, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712215_244582, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712080_453363, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719219_609767, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719481_1167124, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712150_315433, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711931_598023, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712121_564590, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711970_957136, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712183_379489, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719440_2715484, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712226_368928, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719227_66796, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712043_152637, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712142_118145, skipping.\n",
      "No UniRef100 ID found in C3_ERR550534_820089, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719273_1288303, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719457_204269, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719430_512481, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719428_2410604, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719361_3371696, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719348_1883682, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719419_314253, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712062_835967, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719244_1630364, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719433_2409644, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719318_2339218, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712121_621057, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719489_611031, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712212_496005, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719445_2489305, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711953_386903, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719480_262191, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712241_430117, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719411_602588, skipping.\n",
      "No UniRef100 ID found in C3_A0A0D9SBS7, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712076_210528, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711881_65876, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719321_888155, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712141_446735, skipping.\n",
      "No UniRef100 ID found in C3_ERR550532_2579591, skipping.\n",
      "No UniRef100 ID found in C3_ERR1740123_2931861, skipping.\n",
      "No UniRef100 ID found in C3_A0A0C4F6S8, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719453_2717845, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719334_338057, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719402_1187718, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719209_1790759, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719273_2888146, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712121_171322, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719355_37938, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712011_4351, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719186_503299, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711875_161092, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712157_648414, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711971_948551, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711920_549713, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719198_2461576, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719427_739474, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712012_1029583, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719192_2265, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719350_1046545, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719225_1041701, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719213_653626, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712055_1176595, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712064_164482, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719189_790188, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719215_1719675, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712064_260933, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712002_233148, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719228_2899127, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719429_886352, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712135_102897, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711939_768567, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711971_11396, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719297_663138, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719210_668310, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712223_2349095, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719197_1950871, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719333_1587774, skipping.\n",
      "No UniRef100 ID found in C3_ERR550519_3224996, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719167_661305, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719470_121289, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719244_900, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719362_2779486, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719412_1577193, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712042_315509, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719479_519282, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719471_1619543, skipping.\n",
      "No UniRef100 ID found in C3_ERR550532_2715930, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719170_151636, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719220_2732500, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712038_1963045, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719262_177083, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719177_5434, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719412_2060995, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719396_19203, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712038_1796313, skipping.\n",
      "No UniRef100 ID found in C3_MGYP000029337940, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711942_39351, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719216_679463, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712056_127613, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712168_697945, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719334_581672, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719495_340354, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712121_361805, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712179_256579, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719457_416990, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719376_1737011, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712025_447940, skipping.\n",
      "No UniRef100 ID found in C3_HubBroStandDraft_4_1064222.scaffolds.fasta_scaffold936775_1, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719394_1600951, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712137_434174, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719309_1421015, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712119_201478, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712015_341861, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719510_694947, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719422_672942, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712214_10345, skipping.\n",
      "No UniRef100 ID found in C3_ERR1739838_272075, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719508_664940, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719438_361697, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712033_116493, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719259_799252, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712027_274160, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719198_2446239, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719346_90410, skipping.\n",
      "No UniRef100 ID found in C3_A0A0P6B1Z7, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719354_1101172, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712223_1255485, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712039_416099, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712039_831891, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712206_33403, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712018_450526, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712002_164347, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719431_1945034, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719357_1321329, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712100_167117, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712079_970542, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719193_206112, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719210_116992, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711887_329313, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719456_1448500, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719483_1234535, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719330_929, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712241_1305797, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719435_734832, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719341_1944778, skipping.\n",
      "No UniRef100 ID found in C3_ERR1739837_19035, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719410_1239668, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719422_2343736, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712012_489136, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712013_89201, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719334_490498, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712042_310561, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712042_302627, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719215_142299, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712135_115603, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719347_1427426, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719495_3059240, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712025_701660, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719170_44571, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719378_1968798, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712055_456348, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719153_308279, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719495_2118785, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719222_479947, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719400_2036550, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719223_1114506, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719370_537406, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719226_468311, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712228_860415, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719270_3030461, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711977_269010, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711976_661250, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719209_240797, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712113_1030590, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711992_479241, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719289_98007, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719468_1164944, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719354_236648, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719444_105826, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712058_125783, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719415_299563, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719489_559404, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712135_281809, skipping.\n",
      "No UniRef100 ID found in C3_ERR550517_590104, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719482_2684813, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711977_455172, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711973_30742, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711887_54103, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719422_1034113, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719215_1309150, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719334_1482492, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711885_97417, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719323_91029, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719265_2905864, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719374_421445, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719300_575905, skipping.\n",
      "No UniRef100 ID found in C3_ERR1711892_47703, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712050_723761, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712121_66179, skipping.\n",
      "No UniRef100 ID found in C3_ERR1712121_547750, skipping.\n",
      "No UniRef100 ID found in C3_ERR1719265_1684429, skipping.\n",
      "No data found for UPI0020141CF0.\n",
      "Fetched data for A0A3B3XB95: Elongation factor Tu, Poecilia mexicana\n",
      "Fetched data for A0A9D3NN79: Elongation factor Tu, Hemibagrus wyckioides\n",
      "No data found for UPI001CF429E0.\n",
      "No data found for UPI0020B23DCC.\n",
      "Fetched data for A0A8C9VMI8: Elongation factor Tu, Scleropages formosus (Asian bonytongue) (Osteoglossum formosum)\n",
      "Fetched data for A0A8C4G2G9: Elongation factor Tu, Denticeps clupeoides (denticle herring)\n",
      "Fetched data for A0A8C5DN64: Elongation factor Tu, Gouania willdenowi (Blunt-snouted clingfish) (Lepadogaster willdenowi)\n",
      "No data found for UPI0015B114AE.\n",
      "Fetched data for A0A0F8CCZ2: Elongation factor Tu, Larimichthys crocea (Large yellow croaker) (Pseudosciaena crocea)\n",
      "No data found for UPI0014723A79.\n",
      "No data found for UPI0022342621.\n",
      "No data found for UPI001BF586A3.\n",
      "Fetched data for A0A8C5ABC5: Elongation factor Tu, Gadus morhua (Atlantic cod)\n",
      "No data found for UPI001C413D29.\n",
      "Fetched data for A0A3Q3KPB3: Elongation factor Tu, Mastacembelus armatus (zig-zag eel)\n",
      "No data found for UPI001592A892.\n",
      "Fetched data for A0A8C6UL35: Elongation factor Tu, Neogobius melanostomus (round goby)\n",
      "Fetched data for A0A3P8W7T3: Elongation factor Tu, Cynoglossus semilaevis (Tongue sole)\n",
      "Fetched data for A0A671VQ45: Elongation factor Tu, Sparus aurata (Gilthead sea bream)\n",
      "Fetched data for Q5BJ17: Elongation factor Tu, Danio rerio (Zebrafish) (Brachydanio rerio)\n",
      "Fetched data for H3AI71: Elongation factor Tu, Latimeria chalumnae (Coelacanth)\n",
      "Fetched data for A0A6P3VPT1: Elongation factor Tu, Clupea harengus (Atlantic herring)\n",
      "No data found for UPI001C4AD2E8.\n",
      "No data found for UPI001F07CEF3.\n",
      "No data found for UPI001B7F1D16.\n",
      "Fetched data for A0A8C6KEK9: Elongation factor Tu, Nothobranchius furzeri (Turquoise killifish)\n",
      "Fetched data for A0A3Q3R8Z5: Elongation factor Tu, Monopterus albus (Swamp eel)\n",
      "Fetched data for A0A3Q3A9B3: Elongation factor Tu, Kryptolebias marmoratus (Mangrove killifish) (Rivulus marmoratus)\n",
      "No data found for UPI0011273A6D.\n",
      "No data found for UPI00145B4F47.\n",
      "Fetched data for H9GA61: Elongation factor Tu, Anolis carolinensis (Green anole) (American chameleon)\n",
      "Fetched data for A0A3Q3GE42: Elongation factor Tu, Labrus bergylta (ballan wrasse)\n",
      "Fetched data for A0A9D3X2F3: Elongation factor Tu, Mauremys mutica (yellowpond turtle)\n",
      "Fetched data for A0A8C5R5N1: Elongation factor Tu, Leptobrachium leishanense (Leishan spiny toad)\n",
      "Fetched data for A0A3P8Y9G5: Elongation factor Tu, Esox lucius (Northern pike)\n",
      "No data found for UPI001C89C524.\n",
      "No data found for UPI00230303E9.\n",
      "No data found for UPI001CF9A600.\n",
      "No data found for UPI0015D08B8D.\n",
      "Fetched data for A0A6I9MXI0: Elongation factor Tu, Notothenia coriiceps (black rockcod)\n",
      "Fetched data for A0A6P7YKF5: Elongation factor Tu, Microcaecilia unicolor\n",
      "Fetched data for A0A6J0TJB1: Elongation factor Tu, Pogona vitticeps (central bearded dragon)\n",
      "No data found for UPI00214F37ED.\n",
      "Fetched data for A0A8J6EB46: Elongation factor Tu, Eleutherodactylus coqui (Puerto Rican coqui)\n",
      "Fetched data for A0A8C8VFQ3: Elongation factor Tu, Pelusios castaneus (West African mud turtle)\n",
      "Fetched data for V9KW97: Elongation factor Tu, Callorhinchus milii (Ghost shark)\n",
      "Fetched data for A0A6P8REA3: Elongation factor Tu, Geotrypetes seraphini (Gaboon caecilian) (Caecilia seraphini)\n",
      "Fetched data for A0A8D2KR75: Elongation factor Tu, Varanus komodoensis (Komodo dragon)\n",
      "Fetched data for A0A8D0BAB1: Elongation factor Tu, Salvator merianae (Argentine black and white tegu) (Tupinambis merianae)\n",
      "Fetched data for A0A674N0P1: Elongation factor Tu, Takifugu rubripes (Japanese pufferfish) (Fugu rubripes)\n",
      "No data found for UPI0010A82C70.\n",
      "Fetched data for A0A8B9QN43: Elongation factor Tu, Apteryx owenii (Little spotted kiwi)\n",
      "Fetched data for A0A5A9N9Y8: Elongation factor Tu, Triplophysa tibetana\n",
      "No data found for UPI002235AB73.\n",
      "No data found for UPI00202769E7.\n",
      "No data found for UPI00189FBACC.\n",
      "Fetched data for A0A6J2Q291: Elongation factor Tu, Cottoperca gobio (Frogmouth) (Aphritis gobio)\n",
      "Fetched data for A0A2K5W259: Elongation factor Tu, Macaca fascicularis (Crab-eating macaque) (Cynomolgus monkey)\n",
      "Fetched data for A0A674N8H6: Elongation factor Tu, mitochondrial, Takifugu rubripes (Japanese pufferfish) (Fugu rubripes)\n",
      "Fetched data for A0A3Q2TPY7: Elongation factor Tu, mitochondrial, Fundulus heteroclitus (Killifish) (Mummichog)\n",
      "Fetched data for A0A8D0TAV6: Elongation factor Tu, mitochondrial, Sus scrofa (Pig)\n",
      "No data found for UPI00083C0E5E.\n",
      "No UniRef100 ID found in C4_Q6P2W6, skipping.\n",
      "No UniRef100 ID found in C4_A0A2D0S4P3, skipping.\n",
      "No UniRef100 ID found in C4_I3KTG0, skipping.\n",
      "No UniRef100 ID found in C4_MGYP001316284555, skipping.\n",
      "No UniRef100 ID found in C4_H2T912, skipping.\n",
      "No UniRef100 ID found in C4_M7B6Y3, skipping.\n",
      "No UniRef100 ID found in C5_ERR1719163_1131554, skipping.\n",
      "No UniRef100 ID found in C5_ERR1719276_495020, skipping.\n",
      "No UniRef100 ID found in C5_ERR1719211_817415, skipping.\n",
      "No UniRef100 ID found in C5_ERR1719210_2639907, skipping.\n",
      "No UniRef100 ID found in C5_ERR1719444_53188, skipping.\n",
      "No UniRef100 ID found in C5_ERR1719270_1656757, skipping.\n",
      "No UniRef100 ID found in C5_ERR1711966_409292, skipping.\n",
      "No UniRef100 ID found in C5_ERR1711990_262663, skipping.\n",
      "No UniRef100 ID found in C5_ERR1719464_675995, skipping.\n",
      "No UniRef100 ID found in C5_ERR1719333_311695, skipping.\n",
      "No UniRef100 ID found in C5_ERR1712156_260910, skipping.\n",
      "No UniRef100 ID found in C5_ERR1711990_420498, skipping.\n",
      "No UniRef100 ID found in C5_ERR1719163_1849188, skipping.\n",
      "No UniRef100 ID found in C5_ERR1740139_194905, skipping.\n",
      "No UniRef100 ID found in C5_ERR1712021_254817, skipping.\n",
      "No UniRef100 ID found in C5_ERR1712086_1122919, skipping.\n",
      "No UniRef100 ID found in C5_ERR1719272_1174102, skipping.\n",
      "No UniRef100 ID found in C5_ERR1719418_97811, skipping.\n",
      "No UniRef100 ID found in C5_ERR1719491_72718, skipping.\n",
      "No UniRef100 ID found in C5_ERR1719361_305119, skipping.\n",
      "No UniRef100 ID found in C5_ERR1712166_59371, skipping.\n",
      "No UniRef100 ID found in C5_ERR1740139_646359, skipping.\n",
      "Fetched data for A0A653BRQ4: Elongation factor Tu, Callosobruchus maculatus (Southern cowpea weevil) (Pulse bruchid)\n",
      "Fetched data for A0A7R8UBP5: Elongation factor Tu, Hermetia illucens (Black soldier fly)\n",
      "Fetched data for A0A1A9ZPI0: Elongation factor Tu, Glossina pallidipes (Tsetse fly)\n",
      "Fetched data for A0A8J6LQI3: Elongation factor Tu, Tenebrio molitor (Yellow mealworm beetle)\n",
      "Fetched data for A0A182Q8D0: Elongation factor Tu, Anopheles farauti\n",
      "No data found for UPI001F5E5DC3.\n",
      "No data found for UPI0021662478.\n",
      "Fetched data for A0A2J7QJE0: Elongation factor Tu, Cryptotermes secundus\n",
      "Fetched data for A0A0K8TSH2: Elongation factor Tu, Tabanus bromius (Band-eyed brown horse fly)\n",
      "Fetched data for A0A1W4WQN0: Elongation factor Tu, Agrilus planipennis (Emerald ash borer) (Agrilus marcopoli)\n",
      "No data found for UPI00096AED34.\n",
      "No data found for UPI001D066681.\n",
      "Fetched data for D6X0G1: Elongation factor Tu, Tribolium castaneum (Red flour beetle)\n",
      "Fetched data for J3JU35: Elongation factor Tu, Dendroctonus ponderosae (Mountain pine beetle)\n",
      "Fetched data for A0A6B2E7A0: Elongation factor Tu, Phlebotomus kandelakii\n",
      "Fetched data for A0A0L0C7B8: Elongation factor Tu, Lucilia cuprina (Green bottle fly) (Australian sheep blowfly)\n",
      "No data found for UPI000C253FFA.\n",
      "Fetched data for A0A067R0M6: Elongation factor Tu, Zootermopsis nevadensis (Dampwood termite)\n",
      "Fetched data for U5ESU5: Elongation factor Tu, Corethrella appendiculata\n",
      "No data found for UPI00083BE24C.\n",
      "No data found for UPI000C20FB36.\n",
      "No data found for A0A8J5AVB7.\n",
      "Fetched data for A0A1Y1JRP4: Elongation factor Tu, Photinus pyralis (Common eastern firefly) (Lampyris pyralis)\n",
      "Fetched data for A0A6P7GQG5: Elongation factor Tu, Diabrotica virgifera virgifera (western corn rootworm)\n",
      "No data found for UPI001ED86E5B.\n",
      "No data found for A0A7J7BPZ2.\n",
      "Fetched data for A0A1L8DRI0: Elongation factor Tu, Nyssomyia neivai\n",
      "No data found for UPI001D06658B.\n",
      "Fetched data for A0A336LPD7: Elongation factor Tu, Culicoides sonorensis (Biting midge)\n",
      "Fetched data for A0A7R9AWA3: Elongation factor Tu, Timema shepardi (Walking stick)\n",
      "No data found for A0A7J6ZER7.\n",
      "Fetched data for A0A6M2DP98: Elongation factor Tu, Xenopsylla cheopis (oriental rat flea)\n",
      "Fetched data for A0A0T6B4E2: Elongation factor Tu, Oryctes borbonicus\n",
      "Fetched data for A0A8K0DJL1: Elongation factor Tu, Ignelater luminosus (Cucubano) (Pyrophorus luminosus)\n",
      "Fetched data for A0A1J1HQB0: Elongation factor Tu, Clunio marinus\n",
      "No data found for UPI0006251148.\n",
      "Fetched data for A0A6J2YUP2: Elongation factor Tu, Sitophilus oryzae (Rice weevil) (Curculio oryzae)\n",
      "Fetched data for A0A834MHC9: Elongation factor Tu, Rhynchophorus ferrugineus (Red palm weevil) (Curculio ferrugineus)\n",
      "No UniRef100 ID found in C6_A0A182Q8D0, skipping.\n",
      "No UniRef100 ID found in C6_A0A1B0GBE5, skipping.\n",
      "Fetched data for A0A1B0DQV9: Elongation factor Tu, Phlebotomus papatasi (Sandfly)\n",
      "No data found for UPI001E27E28C.\n",
      "No data found for A0A3S2LTQ5.\n",
      "No data found for UPI00067D2531.\n",
      "No data found for UPI0022B68196.\n",
      "Fetched data for A0A8S4R2T2: Jg20010 protein, Pararge aegeria aegeria\n",
      "No data found for UPI0006EADE5D.\n",
      "No data found for UPI0015698AD5.\n",
      "No data found for UPI00214E1616.\n",
      "No data found for UPI0023684DF4.\n",
      "Fetched data for A0A8J9Y142: Tr-type G domain-containing protein, Brenthis ino (lesser marbled fritillary)\n",
      "No data found for UPI00188FAACE.\n",
      "Fetched data for A0A8S3XLH0: (apollo) hypothetical protein, Parnassius apollo (Apollo butterfly) (Papilio apollo)\n",
      "No data found for UPI001E26EED5.\n",
      "Fetched data for A0A8S4F4S8: (diamondback moth) hypothetical protein, Plutella xylostella (Diamondback moth) (Plutella maculipennis)\n",
      "Fetched data for A0A6J1WDU5: Uncharacterized protein LOC113509378 isoform X1, Galleria mellonella (Greater wax moth)\n",
      "No data found for UPI00213179CF.\n",
      "Fetched data for A0A6J2KHG4: Uncharacterized protein LOC114250099, Bombyx mandarina (Wild silk moth) (Wild silkworm)\n",
      "No data found for UPI00103C886C.\n",
      "No data found for UPI000676AA68.\n",
      "Fetched data for A0A7E5W3U6: Uncharacterized protein LOC113499162, Trichoplusia ni (Cabbage looper)\n",
      "Fetched data for A0A821XFK6: Tr-type G domain-containing protein, Pieris macdunnoughi\n",
      "No data found for UPI000E6D9BA1.\n",
      "Fetched data for A0A8S0Z0F4: Tr-type G domain-containing protein, Arctia plantaginis (Wood tiger moth) (Phalaena plantaginis)\n",
      "No data found for UPI000C141A2E.\n",
      "No data found for UPI001F589B82.\n",
      "No data found for UPI00200FB8A3.\n",
      "Fetched data for A0A0N1IQ97: Elongation factor Tu, mitochondrial, Papilio machaon (Old World swallowtail butterfly)\n",
      "Fetched data for A0A922SEV8: Tr-type G domain-containing protein, Spodoptera exigua (Beet armyworm) (Noctua fulgens)\n",
      "No UniRef100 ID found in C7_A0A0N0P9P5, skipping.\n",
      "Fetched data for A0A1I8M4A3: Uncharacterized protein LOC101894922, Musca domestica (House fly)\n",
      "No data found for UPI0014329653.\n",
      "Fetched data for A0A6J1MLA1: Uncharacterized protein LOC111604136, Drosophila hydei (Fruit fly)\n",
      "Fetched data for B4LLC6: Tr-type G domain-containing protein, Drosophila virilis (Fruit fly)\n",
      "Fetched data for A0A1I8P1X1: Tr-type G domain-containing protein, Stomoxys calcitrans (Stable fly) (Conops calcitrans)\n",
      "No data found for UPI000A1CF5F2.\n",
      "Fetched data for B4J9Z8: GH19814, Drosophila grimshawi (Hawaiian fruit fly) (Idiomyia grimshawi)\n",
      "No data found for UPI001CF8EB21.\n",
      "Fetched data for A0A3B0JQR7: Blast:Elongation factor Tu, mitochondrial, Drosophila guanche (Fruit fly)\n",
      "No data found for UPI00148DB477.\n",
      "No data found for UPI001F065AD0.\n",
      "Fetched data for A0A6J2UIW1: Uncharacterized protein LOC115634501, Drosophila lebanonensis (Fruit fly) (Scaptodrosophila lebanonensis)\n",
      "Fetched data for B4MNZ3: Tr-type G domain-containing protein, Drosophila willistoni (Fruit fly)\n",
      "Fetched data for A0A034W4T8: Elongation factor Tu, mitochondrial (Uncharacterized protein LOC105228955), Bactrocera dorsalis (Oriental fruit fly) (Dacus dorsalis)\n",
      "Fetched data for W8BJV6: (Mediterranean fruit fly) hypothetical protein (Elongation factor Tu, mitochondrial), Ceratitis capitata (Mediterranean fruit fly) (Tephritis capitata)\n",
      "Fetched data for A0A1A9USP6: Uncharacterized protein, Glossina austeni (Savannah tsetse fly)\n",
      "No UniRef100 ID found in C8_A0A1I8M4A3, skipping.\n",
      "No UniRef100 ID found in C8_B4KQ95, skipping.\n",
      "No UniRef100 ID found in C8_B4MNZ3, skipping.\n",
      "No UniRef100 ID found in C8_A0A1A9USP6, skipping.\n",
      "No UniRef100 ID found in C8_A0A0K8VH04, skipping.\n",
      "Fetched data for A0A416ECT3: Elongation factor Tu (EF-Tu), Lachnotalea sp. AF33-28\n",
      "Fetched data for A0A1T5CFY0: Elongation factor Tu (EF-Tu), Lachnospiraceae bacterium\n",
      "No data found for UPI000E5D62E5.\n",
      "No data found for UPI00083180EE.\n",
      "Fetched data for A0A9D1C4D7: Elongation factor Tu (EF-Tu), Clostridiales bacterium\n",
      "Fetched data for A0A173ZYQ1: Elongation factor Tu (EF-Tu), Anaerobutyricum hallii\n",
      "Fetched data for A0A2J8B391: Elongation factor Tu (EF-Tu), Lachnospiraceae bacterium\n",
      "Fetched data for G5GHR3: Elongation factor Tu (EF-Tu), Johnsonella ignava ATCC 51276\n",
      "Fetched data for A0A1H9D8W8: Elongation factor Tu (EF-Tu), Lachnospiraceae bacterium NE2001\n",
      "Fetched data for A0A0H5SK50: Elongation factor Tu (EF-Tu), Herbinix hemicellulosilytica\n",
      "Fetched data for A0A3P2AIZ5: Elongation factor Tu (EF-Tu), Clostridiales bacterium COT073_COT-073\n",
      "Fetched data for A0A316R7W5: Elongation factor Tu (EF-Tu), Clostridiales bacterium\n",
      "Fetched data for A0A1I0P5K4: Elongation factor Tu (EF-Tu), [Clostridium] fimetarium\n",
      "Fetched data for A0A1C5SHU8: Elongation factor Tu (EF-Tu), uncultured Ruminococcus sp\n",
      "Fetched data for A0A348ZL69: Elongation factor Tu (EF-Tu), Eubacterium sp\n",
      "No data found for UPI000C14A34B.\n",
      "Fetched data for A0A965YNT0: Elongation factor Tu (EF-Tu), Clostridia bacterium\n",
      "Fetched data for A0A1G4PQ21: Elongation factor Tu (EF-Tu), Lachnospiraceae bacterium C10\n",
      "No data found for UPI001A390A26.\n",
      "No data found for UPI000487E6F7.\n",
      "Fetched data for R9MHP0: Elongation factor Tu (EF-Tu), Lachnospiraceae bacterium MD308\n",
      "Fetched data for A0A1M7J4D3: Elongation factor Tu (EF-Tu), Anaerosporobacter mobilis DSM 15930\n",
      "Fetched data for A0A7U9WYI3: Elongation factor Tu (EF-Tu), Lachnospiraceae bacterium\n",
      "Fetched data for A0A1V5PZ15: Elongation factor Tu (EF-Tu), Firmicutes bacterium ADurb.Bin354\n",
      "Fetched data for X8GTG1: Elongation factor Tu (EF-Tu), Shuttleworthella sp. MSX8B\n",
      "No data found for UPI00047B6FD5.\n",
      "No UniRef100 ID found in C9_MGYP000017821391, skipping.\n",
      "No UniRef100 ID found in C9_MGYP000768469712, skipping.\n",
      "No UniRef100 ID found in C9_EndMetStandDraft_7_1072992.scaffolds.fasta_scaffold6838370_1, skipping.\n",
      "No UniRef100 ID found in C9_MGYP000057911226, skipping.\n",
      "No UniRef100 ID found in C9_uvig_489744_22, skipping.\n",
      "No UniRef100 ID found in C9_uvig_122952_67, skipping.\n",
      "No UniRef100 ID found in C9_MGYP000372260479, skipping.\n",
      "No UniRef100 ID found in C9_uvig_363561_2, skipping.\n",
      "No UniRef100 ID found in C9_MGYP001092748976, skipping.\n",
      "No UniRef100 ID found in C9_MGYP000888659071, skipping.\n",
      "No UniRef100 ID found in C9_MGYP000006696926, skipping.\n",
      "No UniRef100 ID found in C9_MGYP001418009948, skipping.\n",
      "No UniRef100 ID found in C9_MGYP000549267022, skipping.\n",
      "No UniRef100 ID found in C9_MGYP000631784814, skipping.\n",
      "No UniRef100 ID found in C9_MGYP001073322157, skipping.\n",
      "No data found for UPI0003B45AC5.\n",
      "No UniRef100 ID found in C9_MGYP000860544653, skipping.\n",
      "No UniRef100 ID found in C9_MGYP000829072203, skipping.\n",
      "No UniRef100 ID found in C9_MGYP000164080408, skipping.\n",
      "No UniRef100 ID found in C9_MGYP000127458722, skipping.\n",
      "No UniRef100 ID found in C9_MGYP000207204153, skipping.\n",
      "No UniRef100 ID found in C9_MGYP001060800050, skipping.\n",
      "No UniRef100 ID found in C9_MGYP000946897909, skipping.\n",
      "No UniRef100 ID found in C9_MGYP000518451441, skipping.\n",
      "No UniRef100 ID found in C9_MGYP000454807926, skipping.\n",
      "No UniRef100 ID found in C9_MGYP000093159489, skipping.\n",
      "No UniRef100 ID found in C9_MGYP000790466735, skipping.\n",
      "Updated FASTA written to /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX.fa\n"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "from Bio import SeqIO\n",
    "import os\n",
    "import glob\n",
    "\n",
    "# Function to fetch protein and organism information from UniProt, if the ID has\n",
    "# no hit in uniprot, no save in the ourput for downstream analysis. \n",
    "def fetch_uniprot_data(uniref_id):\n",
    "    url = f\"https://rest.uniprot.org/uniprotkb/stream?query={uniref_id}&format=tsv&fields=protein_name,organism_name\"\n",
    "    headers = {\"User-Agent\": \"Python script\"}\n",
    "    response = requests.get(url, headers=headers)\n",
    "    \n",
    "    if response.status_code == 200:\n",
    "        lines = response.text.splitlines()\n",
    "        if len(lines) > 1:\n",
    "            data = lines[1].split(\"\\t\")\n",
    "            if len(data) >= 2:\n",
    "                protein_name = data[0]\n",
    "                organism_name = data[1]\n",
    "                print(f\"Fetched data for {uniref_id}: {protein_name}, {organism_name}\")\n",
    "                return protein_name, organism_name\n",
    "    else:\n",
    "        print(f\"Failed to fetch data for {uniref_id}. Status code: {response.status_code}\")\n",
    "        print(\"Response text:\")\n",
    "        print(response.text)\n",
    "        \n",
    "    print(f\"No data found for {uniref_id}.\")\n",
    "    return None, None\n",
    "\n",
    "# Function to update FASTA labels and exclude entries without UniProt data\n",
    "def update_fasta_labels(input_file):\n",
    "    print(f\"Processing {input_file}...\")\n",
    "    records = list(SeqIO.parse(input_file, \"fasta\"))\n",
    "    updated_records = []\n",
    "    \n",
    "    # Modify the description to replace underscores with dashes and fetch UniProt data\n",
    "    for record in records:\n",
    "        original_description = record.description\n",
    "        \n",
    "        # Look for UniRef100 ID and extract it\n",
    "        if \"UniRef100_\" in original_description:\n",
    "            uniref_id_part = original_description.split(\"UniRef100_\")[-1].split()[0].strip()  # Extract the UniRef ID part\n",
    "            \n",
    "            # Fetch data from UniProt\n",
    "            protein_name, organism_name = fetch_uniprot_data(uniref_id_part)\n",
    "            \n",
    "            # Only add records with valid UniProt data\n",
    "            if protein_name and organism_name:\n",
    "                record.description = f\"{protein_name} [{organism_name}] | {record.description}\"\n",
    "                updated_records.append(record)\n",
    "        else:\n",
    "            print(f\"No UniRef100 ID found in {record.id}, skipping.\")\n",
    "    \n",
    "    # Write the updated records to the output file if there are any\n",
    "    if updated_records:\n",
    "        output_file = input_file.replace('.fasta', '_TAX.fa')\n",
    "        SeqIO.write(updated_records, output_file, \"fasta\")\n",
    "        print(f\"Updated FASTA written to {output_file}\")\n",
    "    else:\n",
    "        print(f\"No valid records with UniProt data found in {input_file}.\")\n",
    "\n",
    "# Function to process all .fasta files in a directory\n",
    "def process_directory(path):\n",
    "    fasta_files = glob.glob(os.path.join(path, '*.fasta'))\n",
    "    for file_path in fasta_files:\n",
    "        update_fasta_labels(file_path)\n",
    "\n",
    "# Set the directory path containing the .fasta files\n",
    "path = '/home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree'\n",
    "\n",
    "# Now process all FASTA files in this directory\n",
    "process_directory(path)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d90884c2-274a-4340-b886-df32cf36ec17",
   "metadata": {
    "jupyter": {
     "source_hidden": true
    }
   },
   "source": [
    "# this cell is marked down. It does not exclude sequences \n",
    "# without uniprot ID. \n",
    "import requests\n",
    "from Bio import SeqIO\n",
    "import os\n",
    "import glob\n",
    "\n",
    "# Function to fetch protein and organism information from UniProt\n",
    "def fetch_uniprot_data(uniref_id):\n",
    "    url = f\"https://rest.uniprot.org/uniprotkb/stream?query={uniref_id}&format=tsv&fields=protein_name,organism_name\"\n",
    "    headers = {\"User-Agent\": \"Python script\"}\n",
    "    response = requests.get(url, headers=headers)\n",
    "    \n",
    "    if response.status_code == 200:\n",
    "        lines = response.text.splitlines()\n",
    "        if len(lines) > 1:\n",
    "            data = lines[1].split(\"\\t\")\n",
    "            if len(data) >= 2:\n",
    "                protein_name = data[0]\n",
    "                organism_name = data[1]\n",
    "                print(f\"Fetched data for {uniref_id}: {protein_name}, {organism_name}\")\n",
    "                return protein_name, organism_name\n",
    "    else:\n",
    "        print(f\"Failed to fetch data for {uniref_id}. Status code: {response.status_code}\")\n",
    "        print(\"Response text:\")\n",
    "        print(response.text)\n",
    "        \n",
    "    print(f\"No data found for {uniref_id}.\")\n",
    "    return None, None\n",
    "\n",
    "# Function to update FASTA labels\n",
    "def update_fasta_labels(input_file):\n",
    "    print(f\"Processing {input_file}...\")\n",
    "    records = list(SeqIO.parse(input_file, \"fasta\"))\n",
    "    updated_records = []\n",
    "    \n",
    "    # Modify the description to replace underscores with dashes and fetch UniProt data\n",
    "    for record in records:\n",
    "        original_description = record.description\n",
    "        \n",
    "        # Replace underscores with dashes\n",
    "        #record.description = original_description.replace('_', '-')\n",
    "        \n",
    "        # Look for UniRef100 ID and extract it\n",
    "        if \"UniRef100_\" in original_description:\n",
    "            uniref_id_part = original_description.split(\"UniRef100_\")[-1].split()[0].strip()  # Extract the UniRef ID part\n",
    "            \n",
    "            # Fetch data from UniProt\n",
    "            protein_name, organism_name = fetch_uniprot_data(uniref_id_part)\n",
    "            \n",
    "            if protein_name and organism_name:\n",
    "                record.description = f\"{protein_name} [{organism_name}] | {record.description}\"\n",
    "                updated_records.append(record)\n",
    "            else:\n",
    "                updated_records.append(record)  # Add even if data is missing\n",
    "        else:\n",
    "            updated_records.append(record)  # Add records without UniRef100 as is\n",
    "\n",
    "    if updated_records:\n",
    "        output_file = input_file.replace('.fasta', '_TAX.fa')\n",
    "        SeqIO.write(updated_records, output_file, \"fasta\")\n",
    "        print(f\"Updated FASTA written to {output_file}\")\n",
    "    else:\n",
    "        print(f\"No valid updates made to {input_file}.\")\n",
    "\n",
    "# Function to process all .fasta files in a directory\n",
    "def process_directory(path):\n",
    "    fasta_files = glob.glob(os.path.join(path, '*.fasta'))\n",
    "    for file_path in fasta_files:\n",
    "        update_fasta_labels(file_path)\n",
    "\n",
    "# Set the directory path containing the .fasta files\n",
    "path = '/home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb'\n",
    "\n",
    "# Now process all FASTA files in this directory\n",
    "process_directory(path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "9f983cd5-cd75-4431-9f85-8f3a529541b3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Renamed sequences saved to /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened.fas\n"
     ]
    }
   ],
   "source": [
    "# Define your conditions for renaming here. \n",
    "# This is to replace the long ID to shorter ones for itol coloring\n",
    "import os\n",
    "import re\n",
    "from Bio import SeqIO\n",
    "rename_conditions = {\n",
    "    \"Drosophila\": \"umt\", \n",
    "    #\"G-like protein ef-g2\": \"EF-G2\",\n",
    "   #\"elongation factor g 1\": \"EF-G1\",\n",
    "    #\"elongation factor g 2\": \"EF-G2\",\n",
    "    #\"tetracycline\": \"Tetra\",\n",
    "    #\"gtp-binding protein\": \"GtpB\",\n",
    "    #\"p-loop containing\": \"Ploop\",\n",
    "    \"tr-type g\": \"TrG\",\n",
    "    #\"elongation factor g\": \"EFG\",\n",
    "    #\"elongation factor ef-g\": \"EFG\",\n",
    "    #\"elongation factor efg\": \"EFG\",\n",
    "    \"EF-Tu\": \"Tu\",\n",
    "    \"elongation factor Tu\": \"Tu\",\n",
    "    #\"EF-2\": \"EF2\",\n",
    "    #\"elongation factor 4\": \"EF4\",\n",
    "    # Add more conditions as needed\n",
    "}\n",
    "\n",
    "# Function to rename sequence ID based on conditions\n",
    "def rename_sequence_id(sequence_id, rename_conditions):\n",
    "    # Extract the original identifier after the last space, tab, or pipe\n",
    "    # Split the ID into parts by space, tab, or pipe\n",
    "    parts = re.split(r'[ \\t|]+', sequence_id)\n",
    "    # Get the last relevant part\n",
    "    original_id = parts[-1]\n",
    "    \n",
    "    # Convert the sequence ID to lowercase for case-insensitive matching\n",
    "    lower_seq_id = sequence_id.lower()\n",
    "    \n",
    "    # Initialize the prefix as empty\n",
    "    prefix = \"\"\n",
    "    \n",
    "    # Check each condition in the rename_conditions dictionary\n",
    "    for term, rename in rename_conditions.items():\n",
    "        if term.lower() in lower_seq_id:\n",
    "            prefix = rename\n",
    "            break  # Stop at the first matching condition\n",
    "    \n",
    "    # Create the new short name\n",
    "    if prefix:\n",
    "        new_id = f\"{prefix}-{original_id}\"\n",
    "    else:\n",
    "        new_id = f\"{original_id}\"\n",
    "    \n",
    "    return new_id\n",
    "    \n",
    "    # execute shorten name \n",
    "# File path for the input .fa file\n",
    "input_fa_file = os.path.join(path, 'Concatenated_All_TAX.fa') # Define the file path to Concatenated_All_TAX.fa using the 'path' variable from the previous cell\n",
    "\n",
    "# Automatically create the output file name by appending \"_shortened.fa\"\n",
    "output_fa_file = os.path.splitext(input_fa_file)[0] + \"_shortened.fas\"\n",
    "\n",
    "# Parse the input .fa file and rename sequence IDs\n",
    "with open(output_fa_file, \"w\") as output_handle:\n",
    "    for record in SeqIO.parse(input_fa_file, \"fasta\"):\n",
    "        original_id = record.description\n",
    "        new_id = rename_sequence_id(original_id, rename_conditions)\n",
    "        record.id = new_id\n",
    "        record.description = \"\"  # Remove the description after ID\n",
    "        SeqIO.write(record, output_handle, \"fasta\")\n",
    "\n",
    "print(f\"Renamed sequences saved to {output_fa_file}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "77000ca6-b5ba-4ac3-9252-fd6636d6024d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CSV file saved as: /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_summary.csv\n"
     ]
    }
   ],
   "source": [
    "# after shorten ID, need to generate scv of three columns: protein, taxon, uniref100 ID\n",
    "import pandas as pd\n",
    "\n",
    "def process_fasta_file(fasta_file):\n",
    "    # Initialize lists to store the extracted information\n",
    "    ids = []\n",
    "    species_names = []\n",
    "    descriptions = []\n",
    "\n",
    "    # Read the fasta file line by line\n",
    "    with open(fasta_file, 'r') as file:\n",
    "        for line in file:\n",
    "            if line.startswith('>'):\n",
    "                # Extract the ID (first part before the first space or tab)\n",
    "                id_part = line.split()[0]\n",
    "                \n",
    "                # Extract the species name (string between [])\n",
    "                if '[' in line and ']' in line:\n",
    "                    species_name = line[line.find('[')+1 : line.find(']')]\n",
    "                else:\n",
    "                    species_name = \"Unknown\"  # Handle cases where species is missing\n",
    "                \n",
    "                # Extract the description (string between the first space/tab and first [)\n",
    "                if ' ' in line and '[' in line:\n",
    "                    description = line.split(' ', 1)[1].split('[')[0].strip()\n",
    "                else:\n",
    "                    description = \"No description\"  # Handle cases where description is missing\n",
    "                \n",
    "                # Append extracted data to lists\n",
    "                ids.append(id_part)\n",
    "                species_names.append(species_name)\n",
    "                descriptions.append(description)\n",
    "\n",
    "    # Create a DataFrame\n",
    "    data = {\n",
    "        'ID': ids,\n",
    "        'Species Name': species_names,\n",
    "        'Description': descriptions\n",
    "    }\n",
    "    df = pd.DataFrame(data)\n",
    "\n",
    "    # Generate the output CSV filename\n",
    "    output_file = fasta_file.replace('.fa', '_summary.csv').replace('.fasta', '_summary.csv')\n",
    "\n",
    "    # Save the DataFrame to a CSV file\n",
    "    df.to_csv(output_file, index=False)\n",
    "    print(f\"CSV file saved as: {output_file}\")\n",
    "\n",
    "# Example usage\n",
    "process_fasta_file(input_fa_file)  # Uses input_fa_file from previous cell.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "4d4fb0b6-c6a8-4a18-b04b-b1fb1bdd7dc4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# make phylo tree with iqtree2\n",
    "def run_phylogenetic_tree(fas_file):\n",
    "    base_name = os.path.splitext(fas_file)[0]\n",
    "    # Command to run IQ-TREE for phylogenetic tree construction\n",
    "    command = [\n",
    "        'iqtree2',\n",
    "        '-s', fas_file,\n",
    "        '-m', 'TEST',\n",
    "        '-bb', '1000',\n",
    "        '-nt', 'AUTO',\n",
    "        '--prefix', base_name\n",
    "    ]\n",
    "    subprocess.run(command)\n",
    "    print(f\"Phylogenetic tree constructed for {fas_file}\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "743655f3-17b1-4b08-8538-96d7f056eacf",
   "metadata": {},
   "outputs": [],
   "source": [
    "# or make ancestral tree with iqtree2\n",
    "def run_phylogenetic_and_ancestral(fas_file):\n",
    "    base_name = os.path.splitext(fas_file)[0]\n",
    "    # Command to run IQ-TREE for phylogenetic tree construction\n",
    "    phylo_command = [\n",
    "        'iqtree2',\n",
    "        '-s', fas_file,\n",
    "        '-m', 'TEST',\n",
    "        '-bb', '1000',\n",
    "        '-nt', 'AUTO',\n",
    "        '--prefix', base_name\n",
    "    ]\n",
    "    subprocess.run(phylo_command)\n",
    "    print(f\"Phylogenetic tree constructed for {fas_file}\")\n",
    "\n",
    "    # Command to run ancestral sequence reconstruction using the generated tree\n",
    "    ancestral_command = [\n",
    "        'iqtree2',\n",
    "        '-s', fas_file,\n",
    "        '-t', f\"{base_name}.treefile\",\n",
    "        '-nt', 'AUTO',\n",
    "        '--ancestral',\n",
    "        '--prefix', f\"{base_name}_anc\"\n",
    "    ]\n",
    "    subprocess.run(ancestral_command)\n",
    "    print(f\"Ancestral sequences reconstructed for {fas_file}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "6d796b77-a5bf-4e45-ad7a-e789913de46e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Enter 'tree' to run only the phylogenetic tree, or 'both' to run both tree and ancestral reconstruction:  both\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IQ-TREE multicore version 2.0.7 for Linux 64-bit built Apr  1 2024\n",
      "Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,\n",
      "Heiko Schmidt, Dominik Schrempf, Michael Woodhams.\n",
      "\n",
      "Host:    BHSC352H6CQ724 (AVX2, FMA3, 31 GB RAM)\n",
      "Command: iqtree2 -s /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened.fas -m TEST -bb 1000 -nt AUTO --prefix /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened\n",
      "Seed:    141879 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Mon Oct 14 22:17:13 2024\n",
      "Kernel:  AVX+FMA - auto-detect threads (32 CPU cores detected)\n",
      "\n",
      "Reading alignment file /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened.fas ... Fasta format detected\n",
      "Alignment most likely contains protein sequences\n",
      "Alignment has 418 sequences with 394 columns, 386 distinct patterns\n",
      "331 parsimony-informative, 36 singleton sites, 27 constant sites\n",
      "                             Gap/Ambiguity  Composition  p-value\n",
      "   1  Tu-C1_UniRef100_A0A8H8S4L3     1.52%    passed     47.19%\n",
      "   2  Tu-C1_UniRef100_A0A3E2HCY8     1.52%    passed     98.47%\n",
      "   3  Tu-C1_UniRef100_A0A8H4RH48     1.52%    passed     98.69%\n",
      "   4  Tu-C1_UniRef100_A0A2A9PMI8     1.27%    passed     99.71%\n",
      "   5  Tu-C1_UniRef100_A0A0D1XB65     1.52%    passed     55.26%\n",
      "   6  Tu-C1_UniRef100_A0A5M9JYY1     1.52%    passed     93.58%\n",
      "   7  Tu-C1_UniRef100_A0A094CMQ6     0.25%    passed     63.78%\n",
      "   8  Tu-C1_UniRef100_A0A1D2JH36     1.52%    passed     97.48%\n",
      "   9  Tu-C1_UniRef100_A0A4Q4W8P2     1.52%    passed     90.73%\n",
      "  10  Tu-C1_UniRef100_A0A093XNZ1     1.52%    passed     98.95%\n",
      "  11  Tu-C1_UniRef100_A0A6A6TY91     1.52%    passed     95.46%\n",
      "  12  Tu-C1_UniRef100_E9CV19         1.52%    passed     37.38%\n",
      "  13  Tu-C1_UniRef100_A0A0F4YMF9     1.52%    passed     97.22%\n",
      "  14  Tu-C1_UniRef100_A0A8H6FBP9     1.52%    passed     70.82%\n",
      "  15  Tu-C1_UniRef100_A0A420Y1Y2     1.27%    passed     89.35%\n",
      "  16  Tu-C1_UniRef100_A0A0G4MSC6     2.03%    passed     86.53%\n",
      "  17  Tu-C1_UniRef100_A0A178B9H0     1.52%    passed     92.85%\n",
      "  18  Tu-C1_UniRef100_A0A1L9SRY3     1.27%    passed     86.67%\n",
      "  19  Tu-C1_UniRef100_A0A0G2GRF6     1.02%    passed     96.85%\n",
      "  20  Tu-C1_UniRef100_A0A100II62     3.30%    passed     59.16%\n",
      "  21  Tu-C1_UniRef100_A0A5N6TSQ9     1.78%    passed     72.53%\n",
      "  22  Tu-C1_UniRef100_A0A5C3NTY7     1.02%    passed     47.85%\n",
      "  23  Tu-C1_UniRef100_A0A0H2S8E1     1.02%    passed     30.99%\n",
      "  24  Tu-C1_UniRef100_A0A4Z0AB69     0.51%    passed     89.35%\n",
      "  25  Tu-C1_UniRef100_A0A2H3IWI7     0.25%    passed     99.46%\n",
      "  26  Tu-C1_UniRef100_A0A8H7FNM9     0.76%    passed     93.94%\n",
      "  27  Tu-C1_UniRef100_A0A409VBL4     1.02%    passed     56.13%\n",
      "  28  Tu-C1_UniRef100_A0A0C3S7H8     1.02%    passed     81.96%\n",
      "  29  Tu-C1_UniRef100_A0A4S8MR95     1.02%    passed     91.06%\n",
      "  30  Tu-C1_UniRef100_A0A067QJI0     1.27%    passed     55.29%\n",
      "  31  Tu-C1_UniRef100_A0A067MI84     1.52%    passed     92.27%\n",
      "  32  Tu-C1_UniRef100_A0A8H8ME73     0.51%    passed     56.75%\n",
      "  33  Tu-C1_UniRef100_A0A286URG5     1.27%    passed     85.49%\n",
      "  34  Tu-C1_UniRef100_K5WPK8         0.51%    passed     23.98%\n",
      "  35  Tu-C1_UniRef100_A0A0L6W793     1.02%    passed     97.60%\n",
      "  36  Tu-C1_UniRef100_A0A0C2T4D2     1.52%    passed     85.32%\n",
      "  37  Tu-C1_UniRef100_A0A369K5C3     1.02%    passed     96.70%\n",
      "  38  Tu-C1_UniRef100_A0A164UBM0     1.27%    passed     86.31%\n",
      "  39  Tu-C1_UniRef100_A8N1J6         1.02%    passed     79.01%\n",
      "  40  Tu-C1_UniRef100_A0A8H5MDS0     1.52%    passed     63.23%\n",
      "  41  Tu-C1_UniRef100_K5WAB1         1.02%    passed     56.19%\n",
      "  42  Tu-C1_UniRef100_A0A0D2LMF0     1.27%    passed     91.35%\n",
      "  43  Tu-C1_UniRef100_A0A8H6W5X9     1.02%    passed     98.21%\n",
      "  44  Tu-C1_UniRef100_A0A5C3QYW4     1.02%    passed     98.92%\n",
      "  45  Tu-C1_UniRef100_A0A401G6V3     0.51%    passed     91.24%\n",
      "  46  Tu-C1_UniRef100_A0A8H6XF81     1.52%    passed     97.54%\n",
      "  47  Tu-C1_UniRef100_A0A8H5FZW9     1.02%    passed     29.19%\n",
      "  48  Tu-C1_UniRef100_A0A0C3BG74     0.76%    passed     97.53%\n",
      "  49  Tu-C1_UniRef100_A0A8H5AZZ4     2.03%    passed     88.92%\n",
      "  50  Tu-C1_UniRef100_A0A8H6TK30     4.31%    passed     87.18%\n",
      "  51  Tu-C1_UniRef100_A0A8H5H174     1.78%    passed     50.43%\n",
      "  52  Tu-C1_UniRef100_A0A4Y9YD42     3.55%    passed     79.36%\n",
      "  53  Tu-C1_UniRef100_A0A1M4SS91     0.25%    passed     99.21%\n",
      "  54  Tu-C1_UniRef100_A0A6I2YU27     0.25%    passed     97.80%\n",
      "  55  Tu-C1_UniRef100_A0A7V9BXY0     0.51%    passed     70.88%\n",
      "  56  Tu-C1_UniRef100_A0A073B1S9     0.25%    passed     68.02%\n",
      "  57  Tu-C1_UniRef100_A0A257RFX4     0.25%    passed     99.38%\n",
      "  58  Tu-C1_UniRef100_A0A931KVY1     0.25%    passed     97.30%\n",
      "  59  Tu-C1_UniRef100_A0A538JUN6     0.25%    passed     98.21%\n",
      "  60  Tu-C1_UniRef100_A0A6I3H0D8     0.25%    passed     98.02%\n",
      "  61  Tu-C1_UniRef100_A0A0D8HL47     0.25%    passed     99.63%\n",
      "  62  Tu-C1_UniRef100_A0A9D8H9Z6     0.51%    passed    100.00%\n",
      "  63  Tu-C1_UniRef100_A0A7W0HE09     0.25%    passed     97.20%\n",
      "  64  Tu-C1_UniRef100_A0A7Y3LMS3     0.25%    passed     97.96%\n",
      "  65  Tu-C1_UniRef100_D3CUS3         0.25%    passed     94.48%\n",
      "  66  Tu-C1_UniRef100_A0A7W1E1A4     0.25%    passed     99.67%\n",
      "  67  Tu-C1_UniRef100_A0A2E5ZGU2     0.25%    passed     99.05%\n",
      "  68  Tu-C1_UniRef100_A0A2U1FDK7     0.25%    passed     97.58%\n",
      "  69  Tu-C1_UniRef100_A0A7J9Z5K4     0.51%    passed     98.65%\n",
      "  70  Tu-C1_UniRef100_A0A966B5E8     0.25%    passed     95.51%\n",
      "  71  Tu-C1_UniRef100_A0A6B1H0D9     0.25%    passed     98.45%\n",
      "  72  Tu-C1_UniRef100_A0A060JN72     0.51%    passed     99.58%\n",
      "  73  Tu-C1_UniRef100_A0A849BFE4     0.51%    passed     99.81%\n",
      "  74  Tu-C1_UniRef100_A0A853BP90     0.25%    passed     83.69%\n",
      "  75  Tu-C1_UniRef100_A0A938J874     0.51%    passed     88.39%\n",
      "  76  Tu-C1_UniRef100_A0A6I2X6Y2     0.25%    passed     94.98%\n",
      "  77  Tu-C1_UniRef100_A0A935AY77     0.25%    passed     99.97%\n",
      "  78  Tu-C1_UniRef100_A0A7W0A023     0.25%    passed     99.47%\n",
      "  79  Tu-C1_UniRef100_A0A7Y0GW13     0.25%    passed     96.79%\n",
      "  80  Tu-C1_UniRef100_A0A538FG31     0.25%    passed     99.52%\n",
      "  81  Tu-C1_UniRef100_A0A970YZR4     0.25%    passed     99.95%\n",
      "  82  Tu-C1_UniRef100_A0A1M6X779     0.25%    passed     92.48%\n",
      "  83  Tu-C1_UniRef100_A0A6N9EZ57     0.25%    passed     98.39%\n",
      "  84  Tu-C1_UniRef100_A0A852VNY4     0.25%    passed     97.64%\n",
      "  85  Tu-C1_UniRef100_A0A0K1JM72     0.25%    passed     98.40%\n",
      "  86  Tu-C1_UniRef100_A0A6I2ZFX3     0.25%    passed     96.70%\n",
      "  87  Tu-C1_UniRef100_A0A2R2W5W5     0.25%    passed     97.65%\n",
      "  88  Tu-C1_UniRef100_A0A6N7I5Y3     0.25%    passed     96.28%\n",
      "  89  Tu-C1_UniRef100_A0A938JSC1     0.51%    passed     99.92%\n",
      "  90  Tu-C1_UniRef100_A0A7J9VS26     0.25%    passed     72.42%\n",
      "  91  Tu-C1_UniRef100_A0A6I6RWI3     0.25%    passed     95.17%\n",
      "  92  Tu-C1_UniRef100_A0A7J9XLA5     0.51%    passed     92.84%\n",
      "  93  Tu-C1_UniRef100_A0A7T3ZWG6     0.25%    passed     97.18%\n",
      "  94  Tu-C1_UniRef100_A0A2W4JPN0     0.25%    passed     94.28%\n",
      "  95  Tu-C1_UniRef100_A0A4R2K0Y4     0.25%    passed     98.78%\n",
      "  96  Tu-C1_UniRef100_A0A938YCV9     0.25%    passed     92.43%\n",
      "  97  Tu-C1_UniRef100_A0A7Y9DK38     0.25%    passed     99.70%\n",
      "  98  Tu-C1_UniRef100_A0A917U0U1     0.25%    passed     99.17%\n",
      "  99  Tu-C1_UniRef100_A0A8J7KNB2     0.25%    passed     99.69%\n",
      " 100  Tu-C1_UniRef100_A0A3Q9UCN8     0.25%    passed     76.37%\n",
      " 101  Tu-C1_UniRef100_D3QB76         0.25%    passed     95.25%\n",
      " 102  Tu-C1_UniRef100_A0A0W8IKR5     0.25%    passed     99.20%\n",
      " 103  Tu-C1_UniRef100_S2WJY6         0.51%    passed     98.90%\n",
      " 104  Tu-C1_UniRef100_A0A1G9Z043     0.25%    passed     99.92%\n",
      " 105  Tu-C1_UniRef100_A0A388NRG8     0.51%    passed     98.41%\n",
      " 106  Tu-C1_UniRef100_A0A810KYL2     0.25%    passed     97.24%\n",
      " 107  Tu-C1_UniRef100_E5XR78         0.51%    passed     83.68%\n",
      " 108  Tu-C1_UniRef100_A0A2G1XI85     0.51%    passed     89.29%\n",
      " 109  Tu-C1_UniRef100_A0A286EN96     0.25%    passed     95.34%\n",
      " 110  Tu-C1_UniRef100_A0A0H5D5X6     0.00%    passed     96.90%\n",
      " 111  Tu-C1_UniRef100_A0A916UM33     0.25%    passed     94.63%\n",
      " 112  Tu-C1_UniRef100_A0A4Q9KFG4     0.25%    passed     99.84%\n",
      " 113  Tu-C1_UniRef100_A0A6P2C7A7     0.51%    passed     98.92%\n",
      " 114  Tu-C1_UniRef100_W5WL60         0.25%    passed     93.64%\n",
      " 115  Tu-C1_UniRef100_A0A1B1MMS2     0.25%    passed     94.99%\n",
      " 116  Tu-C1_UniRef100_A0A383S3W6     0.51%    passed     70.53%\n",
      " 117  Tu-C1_UniRef100_Q8FS84         0.51%    passed     98.71%\n",
      " 118  Tu-C1_UniRef100_A0A9D2ZRP5     0.51%    passed     99.94%\n",
      " 119  Tu-C1_UniRef100_A0A2W5IE43     0.51%    passed     99.31%\n",
      " 120  Tu-C1_UniRef100_A0A9D6XNY2     0.76%    passed     88.55%\n",
      " 121  Tu-C1_UniRef100_A0A563E4C0     0.25%    passed     99.61%\n",
      " 122  Tu-C1_UniRef100_A0A542DJE1     0.51%    passed     49.30%\n",
      " 123  Tu-C1_UniRef100_A0A838A7W8     0.51%    passed     74.69%\n",
      " 124  Tu-C1_UniRef100_A0A1G6HWD1     0.51%    passed     98.50%\n",
      " 125  Tu-C1_UniRef100_A0A7W1R2Z1     0.51%    passed     97.88%\n",
      " 126  Tu-C1_UniRef100_A0A8I1S6W6     0.25%    passed     96.94%\n",
      " 127  Tu-C1_UniRef100_A0A6G8KZR5     0.25%    passed     94.38%\n",
      " 128  Tu-C1_UniRef100_A0A542ZSG3     0.00%    passed     89.84%\n",
      " 129  Tu-C1_UniRef100_I7L9W8         0.76%    passed     95.65%\n",
      " 130  Tu-C1_UniRef100_A0A0T1W7T8     0.51%    passed     86.49%\n",
      " 131  Tu-C1_UniRef100_A0A255EBU4     0.51%    passed     87.29%\n",
      " 132  Tu-C1_UniRef100_A0A7W3LLP6     0.25%    failed      3.33%\n",
      " 133  Tu-C1_UniRef100_A0A543IP01     0.25%    passed      9.44%\n",
      " 134  Tu-C1_UniRef100_A0A6H9YFH4     0.25%    passed     24.61%\n",
      " 135  Tu-C1_UniRef100_A0A4R4M586     0.25%    failed      1.86%\n",
      " 136  Tu-C1_UniRef100_A0A3M2LJI2     0.25%    failed      1.02%\n",
      " 137  Tu-C1_UniRef100_A0A5D0NZ14     0.25%    failed      0.10%\n",
      " 138  Tu-C1_UniRef100_A0A6I4M993     0.25%    failed      4.59%\n",
      " 139  Tu-C1_UniRef100_A0A7K0BVX6     0.51%    failed      2.77%\n",
      " 140  Tu-C1_UniRef100_A0A1H6CUV6     0.25%    passed     10.66%\n",
      " 141  Tu-C1_UniRef100_A0A7W3R6L5     1.02%    failed      1.26%\n",
      " 142  Tu-C1_UniRef100_A0A3D9SNF9     0.25%    failed      4.19%\n",
      " 143  Tu-C1_UniRef100_A0A4R9EXQ9     0.25%    passed      8.77%\n",
      " 144  Tu-C1_UniRef100_A0A1Q4UYS8     1.52%    failed      1.24%\n",
      " 145  Tu-C1_UniRef100_A0A7X3ACK6     1.52%    failed      0.69%\n",
      " 146  Tu-C1_UniRef100_A0A944KRH2     0.51%    failed      2.29%\n",
      " 147  Tu-C1_UniRef100_A0A919DET7     1.27%    failed      3.94%\n",
      " 148  Tu-C1_UniRef100_A0A6G4XFJ0     1.27%    passed     17.18%\n",
      " 149  Tu-C1_UniRef100_A0A5C6JZ56     1.02%    failed      0.10%\n",
      " 150  Tu-C1_UniRef100_A0A2U3C0W4     1.52%    failed      3.34%\n",
      " 151  Tu-C1_UniRef100_A0A0C5GA75     1.52%    failed      0.17%\n",
      " 152  Tu-C1_UniRef100_A0A124ED28     1.27%    failed      0.04%\n",
      " 153  Tu-C1_UniRef100_V6JMA1         1.27%    failed      4.05%\n",
      " 154  Tu-C1_UniRef100_A0A918GNF4     1.52%    failed      0.07%\n",
      " 155  Tu-C1_UniRef100_A0A7K0CB26     1.52%    failed      1.13%\n",
      " 156  Tu-C1_UniRef100_A0A7H0IMG2     1.27%    failed      0.23%\n",
      " 157  Tu-C1_UniRef100_A0A6G3RX88     1.02%    failed      1.53%\n",
      " 158  Tu-C1_UniRef100_A0A2S4XVP4     0.76%    failed      0.05%\n",
      " 159  Tu-C1_UniRef100_A0A6G4TSH5     1.52%    passed      8.90%\n",
      " 160  Tu-C1_UniRef100_A0A8T4IWN8     1.02%    failed      1.95%\n",
      " 161  Tu-C1_UniRef100_A0A365Z5T0     1.52%    failed      0.11%\n",
      " 162  Tu-C1_UniRef100_A0A1I2AXZ0     0.76%    failed      2.56%\n",
      " 163  Tu-C1_UniRef100_A0A505CYL6     1.52%    failed      2.26%\n",
      " 164  Tu-C1_UniRef100_A0A3D0R5B3     1.52%    failed      0.24%\n",
      " 165  Tu-C1_UniRef100_A0A370B0A1     1.27%    failed      2.86%\n",
      " 166  Tu-C1_UniRef100_A0A221NTP9     1.52%    failed      1.73%\n",
      " 167  Tu-C1_UniRef100_A0A2P2GEE7     1.27%    failed      0.12%\n",
      " 168  Tu-C1_UniRef100_A0A1H8NGK5     1.02%    failed      0.86%\n",
      " 169  Tu-C1_UniRef100_A0A6B3HEY9     1.27%    failed      0.11%\n",
      " 170  Tu-C1_UniRef100_A0A2M9KMZ3     0.76%    failed      0.39%\n",
      " 171  Tu-C1_UniRef100_A0A939F4V4     0.51%    failed      3.34%\n",
      " 172  Tu-C1_UniRef100_A0A1Y2NTG1     1.27%    failed      0.07%\n",
      " 173  Tu-C1_UniRef100_A0A918NHC8     1.27%    failed      0.81%\n",
      " 174  Tu-C1_UniRef100_A0A372M4V6     1.52%    failed      2.80%\n",
      " 175  Tu-C1_UniRef100_A0A918U6A5     1.52%    failed      0.04%\n",
      " 176  Tu-C1_UniRef100_A0A1V2RTZ0     1.52%    failed      0.22%\n",
      " 177  Tu-C1_UniRef100_A0A3M0ICR4     1.52%    failed      0.36%\n",
      " 178  Tu-C1_UniRef100_A0A919GYQ9     0.76%    failed      0.20%\n",
      " 179  Tu-C1_UniRef100_S5UYY7         1.27%    failed      1.47%\n",
      " 180  Tu-C1_UniRef100_A0A930ZX77     1.27%    failed      0.12%\n",
      " 181  Tu-C1_UniRef100_A0A8G1ZKR6     1.27%    failed      0.05%\n",
      " 182  Tu-C1_UniRef100_A0A1H0N0L2     1.02%    failed      2.51%\n",
      " 183  Tu-C1_UniRef100_A0A918MFV0     1.27%    failed      0.85%\n",
      " 184  Tu-C1_UniRef100_A0A5P2CC72     1.27%    failed      2.35%\n",
      " 185  Tu-C1_UniRef100_A0A5N8WYA0     1.27%    failed      1.96%\n",
      " 186  Tu-C1_UniRef100_A0A7K3BKF0     1.52%    failed      0.50%\n",
      " 187  Tu-C1_UniRef100_A0A2R4SY40     1.02%    failed      0.38%\n",
      " 188  Tu-C1_UniRef100_A0A345XQ45     1.52%    failed      0.51%\n",
      " 189  Tu-C1_UniRef100_A0A2B8B284     1.52%    failed      0.10%\n",
      " 190  Tu-C1_UniRef100_A0A4R4TEU6     1.52%    failed      0.06%\n",
      " 191  Tu-C1_UniRef100_A0A7M3LV78     1.52%    failed      0.33%\n",
      " 192  Tu-C1_UniRef100_A0A7K2WM35     0.76%    failed      3.87%\n",
      " 193  Tu-C1_UniRef100_A0A101UD17     1.27%    failed      0.06%\n",
      " 194  Tu-C1_UniRef100_A0A7J0C1M1     1.27%    failed      0.13%\n",
      " 195  Tu-C1_UniRef100_D7BY48         1.52%    failed      1.57%\n",
      " 196  Tu-C1_UniRef100_A0A0X3W0C3     1.02%    failed      3.11%\n",
      " 197  Tu-C1_UniRef100_A0A0N1NII9     1.27%    failed      0.62%\n",
      " 198  Tu-C1_UniRef100_A0A1H9BGA1     1.27%    failed      0.75%\n",
      " 199  Tu-C1_UniRef100_A0A1M7Q0J6     0.76%    failed      0.52%\n",
      " 200  Tu-C1_UniRef100_A0A7K3DNB6     0.76%    failed      0.11%\n",
      " 201  Tu-C1_UniRef100_A3KKP5         1.27%    failed      0.01%\n",
      " 202  Tu-C1_UniRef100_A0A0F7G286     1.27%    failed      4.45%\n",
      " 203  Tu-C1_UniRef100_A0A1H6CDW4     1.02%    failed      0.71%\n",
      " 204  Tu-C1_UniRef100_A0A387HNB4     1.02%    failed      2.10%\n",
      " 205  Tu-C1_UniRef100_A0A919BZ14     1.52%    failed      0.02%\n",
      " 206  Tu-C1_UniRef100_A0A6B2RYS8     0.76%    failed      0.40%\n",
      " 207  Tu-C1_UniRef100_A0A7T1TBV1     1.52%    failed      0.55%\n",
      " 208  Tu-C1_UniRef100_A0A5B8IC35     1.27%    failed      0.28%\n",
      " 209  Tu-C1_UniRef100_A0A3R9XQG8     0.76%    failed      0.00%\n",
      " 210  Tu-C1_UniRef100_A0A919L5M4     2.03%    failed      1.06%\n",
      " 211  Tu-C1_UniRef100_A0A344U3D7     0.76%    failed      0.00%\n",
      " 212  Tu-C1_UniRef100_A0A919HD98     1.52%    failed      0.08%\n",
      " 213  Tu-C1_UniRef100_A0A974AYX1     1.02%    failed      0.22%\n",
      " 214  Tu-C1_UniRef100_A0A4Z0HB80     1.02%    failed      0.03%\n",
      " 215  Tu-C1_UniRef100_F3NTJ2         1.78%    failed      0.00%\n",
      " 216  Tu-C1_UniRef100_A0A4R3YAC7     0.25%    passed     99.79%\n",
      " 217  Tu-C1_UniRef100_A0A143WPR6     0.00%    passed     71.61%\n",
      " 218  Tu-C1_UniRef100_A0A209AQB3     0.25%    passed     99.98%\n",
      " 219  Tu-C1_UniRef100_U3TY99         0.00%    passed     99.98%\n",
      " 220  Tu-C1_UniRef100_A0A1N7CGP3     0.25%    passed     96.41%\n",
      " 221  Tu-C1_UniRef100_A0A5P2UM61     0.00%    passed     99.08%\n",
      " 222  Tu-C1_UniRef100_A0A1V8NCP4     0.25%    passed     40.17%\n",
      " 223  Tu-C1_UniRef100_A0A4D6YLG4     0.00%    passed     75.77%\n",
      " 224  C1_UniRef100_A0A420HRM0        0.25%    passed     95.38%\n",
      " 225  Tu-C1_UniRef100_F7XY84         0.00%    passed     99.96%\n",
      " 226  Tu-C1_UniRef100_A0A2C8CVX6     0.00%    passed     97.80%\n",
      " 227  Tu-C1_UniRef100_A0A4D6YN62     0.00%    passed     82.04%\n",
      " 228  Tu-C1_UniRef100_Q1LSY4         0.25%    passed     83.98%\n",
      " 229  Tu-C1_UniRef100_Q7VRP0         0.00%    passed     67.26%\n",
      " 230  Tu-C1_UniRef100_A0A0E3U3M2     0.00%    passed     98.68%\n",
      " 231  Tu-C1_UniRef100_D5BVA9         0.25%    passed     99.87%\n",
      " 232  Tu-C1_UniRef100_A0A1C7AF74     0.25%    passed     98.30%\n",
      " 233  Tu-C1_UniRef100_A0A9C9VER6     0.25%    passed     95.68%\n",
      " 234  Tu-C1_UniRef100_L0DYY6         0.25%    passed     98.21%\n",
      " 235  Tu-C1_UniRef100_A0A1Y0NA39     0.00%    passed     99.45%\n",
      " 236  Tu-C1_UniRef100_A0A1C3L4A1     0.25%    passed     33.76%\n",
      " 237  Tu-C1_UniRef100_A0A3S8T3C9     0.25%    passed     99.88%\n",
      " 238  Tu-C1_UniRef100_A0A450TB23     0.51%    passed     96.18%\n",
      " 239  Tu-C1_UniRef100_A0A6S6UKH2     0.51%    passed     88.49%\n",
      " 240  Tu-C1_UniRef100_A0A949D297     0.25%    passed     96.98%\n",
      " 241  Tu-C1_UniRef100_A0A9D0P1A1     0.25%    passed     89.59%\n",
      " 242  Tu-C1_UniRef100_A0A963D6S8     0.51%    passed     99.19%\n",
      " 243  Tu-C1_UniRef100_A0A2A4UKP8     0.25%    passed     93.84%\n",
      " 244  Tu-C1_UniRef100_A0A2E8KCX4     0.51%    passed     96.47%\n",
      " 245  Tu-C1_UniRef100_A0A1U9JI97     0.25%    passed     99.96%\n",
      " 246  Tu-C1_UniRef100_A0P4F9         0.51%    passed     96.35%\n",
      " 247  Tu-C1_UniRef100_D9SCC2         0.25%    passed     93.47%\n",
      " 248  Tu-C1_UniRef100_A0A1G0K9T7     0.51%    passed     98.61%\n",
      " 249  Tu-C1_UniRef100_A0A086WEB3     0.00%    passed     99.85%\n",
      " 250  Tu-C1_UniRef100_A0A7X1P490     0.25%    passed     99.73%\n",
      " 251  Tu-C1_UniRef100_A0A2A2EQH0     0.25%    passed     99.52%\n",
      " 252  Tu-C1_UniRef100_A0A9D7KDJ5     0.25%    passed     97.29%\n",
      " 253  Tu-C1_UniRef100_A0A7D4SXK4     0.25%    passed     99.12%\n",
      " 254  Tu-C1_UniRef100_A0A661EVB5     0.25%    passed     98.96%\n",
      " 255  Tu-C1_UniRef100_A0A536YJ92     0.25%    passed     93.82%\n",
      " 256  Tu-C1_UniRef100_A0A521XAZ9     0.25%    passed     99.80%\n",
      " 257  Tu-C1_UniRef100_A0A212KMK6     0.25%    passed     99.71%\n",
      " 258  Tu-C1_UniRef100_A0A836S6B4     0.25%    passed     83.04%\n",
      " 259  Tu-C1_UniRef100_A0A6N6VIM4     0.25%    passed     99.86%\n",
      " 260  Tu-C1_UniRef100_A0A9D8CB88     0.25%    passed     96.27%\n",
      " 261  Tu-C1_UniRef100_A0A2E4GJ30     0.00%    passed     99.62%\n",
      " 262  Tu-C1_UniRef100_A0A1H5XFN9     1.52%    passed     98.22%\n",
      " 263  Tu-C1_UniRef100_Q9P9Q9         0.25%    passed     99.97%\n",
      " 264  Tu-C1_UniRef100_A0A1Q6UCA4     0.25%    passed     89.84%\n",
      " 265  Tu-C1_UniRef100_A0A061QEV0     0.00%    passed     98.68%\n",
      " 266  Tu-C1_UniRef100_A0A7W5ELG9     0.25%    passed     89.13%\n",
      " 267  Tu-C1_UniRef100_A0A7R6W0L4     0.25%    passed     60.09%\n",
      " 268  Tu-C1_UniRef100_A0A1N7K9D5     0.00%    passed     99.89%\n",
      " 269  Tu-C1_UniRef100_A0A3B0YSK9     0.25%    passed     97.90%\n",
      " 270  Tu-C1_UniRef100_U2E2F6         0.51%    passed     99.87%\n",
      " 271  Tu-C1_UniRef100_A0A7V8AKP9     0.25%    passed     99.12%\n",
      " 272  Tu-C1_UniRef100_A0A2E7K4G9     0.25%    passed     99.67%\n",
      " 273  Tu-C1_UniRef100_A0A1Z8KXW1     0.51%    passed     99.87%\n",
      " 274  Tu-C1_UniRef100_A0A2D7AAY0     0.25%    passed     99.08%\n",
      " 275  Tu-C1_UniRef100_A0A0Q4NEQ5     0.25%    passed     94.08%\n",
      " 276  Tu-C1_UniRef100_M4S4E3         1.27%    passed    100.00%\n",
      " 277  Tu-C1_UniRef100_A0A3A1Y0W9     3.30%    passed     99.82%\n",
      " 278  Tu-C1_UniRef100_A0A2N6N749     0.25%    passed     99.45%\n",
      " 279  Tu-C1_UniRef100_A0A6L5G3T8     0.51%    passed     99.55%\n",
      " 280  Tu-C1_UniRef100_A0A523FG19     0.25%    passed     99.25%\n",
      " 281  Tu-C1_UniRef100_A0A6L7S3G9     1.52%    passed     99.22%\n",
      " 282  Tu-C1_UniRef100_A0A523FVQ2     0.25%    passed     98.24%\n",
      " 283  Tu-C1_UniRef100_A0A2S6T5Q0     0.25%    passed     85.92%\n",
      " 284  Tu-C1_UniRef100_A0A9E1RA41     0.00%    passed     93.43%\n",
      " 285  Tu-C1_UniRef100_A0A9D8GMY6     0.25%    passed     98.82%\n",
      " 286  Tu-C1_UniRef100_A0A2E5DE46     0.25%    passed     89.62%\n",
      " 287  Tu-C1_UniRef100_A0A0Q6BEW9     2.54%    passed     99.80%\n",
      " 288  Tu-C1_UniRef100_A0A2E5DWF1     1.78%    passed     99.95%\n",
      " 289  Tu-C1_UniRef100_A0A5C7SRI2     0.25%    passed     99.84%\n",
      " 290  Tu-C1_UniRef100_A0A6B0XYM7     0.51%    passed     99.83%\n",
      " 291  Tu-C1_UniRef100_A0A2E4GZN0     0.51%    passed     93.59%\n",
      " 292  Tu-C1_UniRef100_M5E919         0.00%    passed     99.85%\n",
      " 293  Tu-C1_UniRef100_A0A1R4HHK1     0.25%    passed     96.64%\n",
      " 294  Tu-C1_UniRef100_A0A6L8C4X0     1.52%    passed     97.84%\n",
      " 295  Tu-C1_UniRef100_A0A4R5Q2H1     0.76%    passed     99.81%\n",
      " 296  Tu-C1_UniRef100_A0A6N6JNP2     1.52%    passed     95.33%\n",
      " 297  Tu-C1_UniRef100_A0A0S8BXZ4     0.00%    passed     93.79%\n",
      " 298  Tu-C1_UniRef100_I3TNZ7         3.81%    passed     99.87%\n",
      " 299  Tu-C1_UniRef100_A0A0Q4QUV2     0.51%    passed     97.23%\n",
      " 300  Tu-C1_UniRef100_A0A7V8W6U5     0.00%    passed     93.32%\n",
      " 301  Tu-C1_UniRef100_A0A6S6SHZ6     6.35%    passed     88.16%\n",
      " 302  Tu-C1_UniRef100_A0A0Q5DC00     1.52%    passed     99.36%\n",
      " 303  Tu-C2_UniRef100_E9UZA8         0.51%    passed     46.70%\n",
      " 304  Tu-C2_UniRef100_A0A810MQ54     0.25%    failed      0.55%\n",
      " 305  Tu-C2_UniRef100_A0A7Y9ZI65     0.51%    passed      6.88%\n",
      " 306  Tu-C2_UniRef100_A0A853BXE2     0.51%    failed      3.00%\n",
      " 307  Tu-C2_UniRef100_A0A8J4DQ16     0.25%    passed     30.80%\n",
      " 308  Tu-C2_UniRef100_A0A4V6PAK6     0.51%    passed     30.33%\n",
      " 309  Tu-C2_UniRef100_A0A561BKB4     0.25%    passed     42.52%\n",
      " 310  Tu-C2_UniRef100_A0A849DU62     0.51%    failed      1.36%\n",
      " 311  TrG-C3_UniRef100_G7E513        2.54%    failed      0.13%\n",
      " 312  Tu-C4_UniRef100_A0A3B3XB95     0.25%    passed     53.07%\n",
      " 313  Tu-C4_UniRef100_A0A9D3NN79     0.51%    passed     77.84%\n",
      " 314  Tu-C4_UniRef100_A0A8C9VMI8     0.51%    passed     93.78%\n",
      " 315  Tu-C4_UniRef100_A0A8C4G2G9     0.25%    passed     93.86%\n",
      " 316  Tu-C4_UniRef100_A0A8C5DN64     0.25%    passed     94.33%\n",
      " 317  Tu-C4_UniRef100_A0A0F8CCZ2     0.25%    passed     93.24%\n",
      " 318  Tu-C4_UniRef100_A0A8C5ABC5     0.25%    passed     94.80%\n",
      " 319  Tu-C4_UniRef100_A0A3Q3KPB3     0.25%    passed     62.99%\n",
      " 320  Tu-C4_UniRef100_A0A8C6UL35     0.51%    passed     91.65%\n",
      " 321  Tu-C4_UniRef100_A0A3P8W7T3     0.25%    passed     82.74%\n",
      " 322  Tu-C4_UniRef100_A0A671VQ45     0.25%    passed     87.27%\n",
      " 323  Tu-C4_UniRef100_Q5BJ17         0.25%    passed     71.18%\n",
      " 324  Tu-C4_UniRef100_H3AI71         0.51%    passed     55.42%\n",
      " 325  Tu-C4_UniRef100_A0A6P3VPT1     0.25%    passed     99.28%\n",
      " 326  Tu-C4_UniRef100_A0A8C6KEK9     0.25%    passed     85.45%\n",
      " 327  Tu-C4_UniRef100_A0A3Q3R8Z5     0.51%    passed     75.71%\n",
      " 328  Tu-C4_UniRef100_A0A3Q3A9B3     0.25%    passed     57.28%\n",
      " 329  Tu-C4_UniRef100_H9GA61         0.51%    passed     81.26%\n",
      " 330  Tu-C4_UniRef100_A0A3Q3GE42     0.51%    passed     91.45%\n",
      " 331  Tu-C4_UniRef100_A0A9D3X2F3     0.51%    passed     85.19%\n",
      " 332  Tu-C4_UniRef100_A0A8C5R5N1     0.51%    passed     53.45%\n",
      " 333  Tu-C4_UniRef100_A0A3P8Y9G5     0.51%    passed     94.11%\n",
      " 334  Tu-C4_UniRef100_A0A6I9MXI0     0.51%    passed     75.53%\n",
      " 335  Tu-C4_UniRef100_A0A6P7YKF5     0.51%    passed     17.48%\n",
      " 336  Tu-C4_UniRef100_A0A6J0TJB1     0.51%    passed     85.85%\n",
      " 337  Tu-C4_UniRef100_A0A8J6EB46     0.51%    passed     38.10%\n",
      " 338  Tu-C4_UniRef100_A0A8C8VFQ3     0.51%    passed     89.25%\n",
      " 339  Tu-C4_UniRef100_V9KW97         0.51%    passed     90.33%\n",
      " 340  Tu-C4_UniRef100_A0A6P8REA3     0.51%    passed     63.55%\n",
      " 341  Tu-C4_UniRef100_A0A8D2KR75     0.51%    passed     67.66%\n",
      " 342  Tu-C4_UniRef100_A0A8D0BAB1     0.51%    passed     96.75%\n",
      " 343  Tu-C4_UniRef100_A0A674N0P1     0.25%    passed     78.60%\n",
      " 344  Tu-C4_UniRef100_A0A8B9QN43     0.51%    passed     69.52%\n",
      " 345  Tu-C4_UniRef100_A0A5A9N9Y8     0.25%    passed     99.71%\n",
      " 346  Tu-C4_UniRef100_A0A6J2Q291     0.76%    passed     54.36%\n",
      " 347  Tu-C4_UniRef100_A0A2K5W259     0.25%    passed     88.74%\n",
      " 348  Tu-C4_UniRef100_A0A674N8H6     0.25%    passed     33.26%\n",
      " 349  Tu-C4_UniRef100_A0A3Q2TPY7     4.82%    passed     81.00%\n",
      " 350  Tu-C4_UniRef100_A0A8D0TAV6     3.81%    passed     57.36%\n",
      " 351  Tu-C6_UniRef100_A0A653BRQ4     1.27%    passed     53.99%\n",
      " 352  Tu-C6_UniRef100_A0A7R8UBP5     1.02%    passed     53.14%\n",
      " 353  Tu-C6_UniRef100_A0A1A9ZPI0     1.27%    passed     29.08%\n",
      " 354  Tu-C6_UniRef100_A0A8J6LQI3     1.27%    passed     48.39%\n",
      " 355  Tu-C6_UniRef100_A0A182Q8D0     1.02%    passed     51.75%\n",
      " 356  Tu-C6_UniRef100_A0A2J7QJE0     1.52%    passed     73.95%\n",
      " 357  Tu-C6_UniRef100_A0A0K8TSH2     1.02%    passed     49.11%\n",
      " 358  Tu-C6_UniRef100_A0A1W4WQN0     1.27%    passed     51.40%\n",
      " 359  Tu-C6_UniRef100_D6X0G1         1.27%    passed     35.70%\n",
      " 360  Tu-C6_UniRef100_J3JU35         1.27%    passed     61.43%\n",
      " 361  Tu-C6_UniRef100_A0A6B2E7A0     1.27%    passed     87.20%\n",
      " 362  Tu-C6_UniRef100_A0A0L0C7B8     1.27%    passed     43.63%\n",
      " 363  Tu-C6_UniRef100_A0A067R0M6     1.52%    passed     82.07%\n",
      " 364  Tu-C6_UniRef100_U5ESU5         1.27%    passed     15.88%\n",
      " 365  Tu-C6_UniRef100_A0A1Y1JRP4     1.27%    passed     32.14%\n",
      " 366  Tu-C6_UniRef100_A0A6P7GQG5     1.27%    passed     74.14%\n",
      " 367  Tu-C6_UniRef100_A0A1L8DRI0     1.27%    passed     68.11%\n",
      " 368  Tu-C6_UniRef100_A0A336LPD7     1.27%    passed     13.36%\n",
      " 369  Tu-C6_UniRef100_A0A7R9AWA3     1.27%    passed     53.68%\n",
      " 370  Tu-C6_UniRef100_A0A6M2DP98     1.27%    passed     52.29%\n",
      " 371  Tu-C6_UniRef100_A0A0T6B4E2     1.02%    passed     27.43%\n",
      " 372  Tu-C6_UniRef100_A0A8K0DJL1     1.27%    passed     30.20%\n",
      " 373  Tu-C6_UniRef100_A0A1J1HQB0     1.02%    passed     73.98%\n",
      " 374  Tu-C6_UniRef100_A0A6J2YUP2     1.27%    passed     46.84%\n",
      " 375  Tu-C6_UniRef100_A0A834MHC9     5.58%    passed     36.99%\n",
      " 376  Tu-C6_UniRef100_A0A1B0DQV9     1.27%    passed     97.04%\n",
      " 377  C7_UniRef100_A0A8S4R2T2        2.54%    failed      2.15%\n",
      " 378  TrG-C7_UniRef100_A0A8J9Y142    2.28%    failed      4.02%\n",
      " 379  C7_UniRef100_A0A8S3XLH0        2.54%    failed      1.47%\n",
      " 380  C7_UniRef100_A0A8S4F4S8        1.27%    passed     11.48%\n",
      " 381  C7_UniRef100_A0A6J1WDU5        2.54%    failed      3.33%\n",
      " 382  C7_UniRef100_A0A6J2KHG4        2.28%    passed     28.95%\n",
      " 383  C7_UniRef100_A0A7E5W3U6        2.28%    failed      1.29%\n",
      " 384  TrG-C7_UniRef100_A0A821XFK6    2.54%    failed      3.38%\n",
      " 385  TrG-C7_UniRef100_A0A8S0Z0F4    2.28%    failed      2.47%\n",
      " 386  Tu-C7_UniRef100_A0A0N1IQ97     2.28%    passed     15.95%\n",
      " 387  TrG-C7_UniRef100_A0A922SEV8    2.28%    passed     11.14%\n",
      " 388  C8_UniRef100_A0A1I8M4A3        2.03%    passed     24.47%\n",
      " 389  umt-C8_UniRef100_A0A6J1MLA1    2.03%    failed      3.00%\n",
      " 390  umt-C8_UniRef100_B4LLC6        1.52%    failed      3.21%\n",
      " 391  TrG-C8_UniRef100_A0A1I8P1X1    2.03%    failed      1.33%\n",
      " 392  umt-C8_UniRef100_B4J9Z8        2.54%    passed      8.48%\n",
      " 393  umt-C8_UniRef100_A0A3B0JQR7    2.79%    failed      3.71%\n",
      " 394  umt-C8_UniRef100_A0A6J2UIW1    3.05%    failed      1.27%\n",
      " 395  umt-C8_UniRef100_B4MNZ3        1.27%    failed      3.77%\n",
      " 396  Tu-C8_UniRef100_A0A034W4T8     2.54%    failed      2.83%\n",
      " 397  Tu-C8_UniRef100_W8BJV6         2.79%    failed      0.37%\n",
      " 398  C8_UniRef100_A0A1A9USP6        2.54%    passed     24.17%\n",
      " 399  Tu-C9_UniRef100_A0A416ECT3     0.25%    passed     82.42%\n",
      " 400  Tu-C9_UniRef100_A0A1T5CFY0     0.25%    passed     98.67%\n",
      " 401  Tu-C9_UniRef100_A0A9D1C4D7     0.51%    passed     90.37%\n",
      " 402  Tu-C9_UniRef100_A0A173ZYQ1     0.25%    passed     88.74%\n",
      " 403  Tu-C9_UniRef100_A0A2J8B391     0.25%    passed     57.51%\n",
      " 404  Tu-C9_UniRef100_G5GHR3         0.25%    passed     78.47%\n",
      " 405  Tu-C9_UniRef100_A0A1H9D8W8     0.76%    passed     95.66%\n",
      " 406  Tu-C9_UniRef100_A0A0H5SK50     0.25%    passed     96.56%\n",
      " 407  Tu-C9_UniRef100_A0A3P2AIZ5     0.25%    passed     87.82%\n",
      " 408  Tu-C9_UniRef100_A0A316R7W5     0.76%    passed     44.06%\n",
      " 409  Tu-C9_UniRef100_A0A1I0P5K4     0.00%    passed     76.36%\n",
      " 410  Tu-C9_UniRef100_A0A1C5SHU8     0.25%    passed     51.38%\n",
      " 411  Tu-C9_UniRef100_A0A348ZL69     0.76%    passed     68.18%\n",
      " 412  Tu-C9_UniRef100_A0A965YNT0     0.76%    passed     95.46%\n",
      " 413  Tu-C9_UniRef100_A0A1G4PQ21     0.76%    passed     94.90%\n",
      " 414  Tu-C9_UniRef100_R9MHP0         0.76%    passed     66.07%\n",
      " 415  Tu-C9_UniRef100_A0A1M7J4D3     0.76%    passed     85.22%\n",
      " 416  Tu-C9_UniRef100_A0A7U9WYI3     0.76%    passed     40.54%\n",
      " 417  Tu-C9_UniRef100_A0A1V5PZ15     0.76%    passed     69.90%\n",
      " 418  Tu-C9_UniRef100_X8GTG1         0.76%    passed     62.72%\n",
      "****  TOTAL                          0.87%  97 sequences failed composition chi2 test (p-value<5%; df=19)\n",
      "\n",
      "\n",
      "Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.089 seconds\n",
      "Measuring multi-threading efficiency up to 32 CPU cores\n",
      "Increase to 10 rounds for branch lengths\n",
      "14 trees examined\n",
      "Threads: 1 / Time: 34.233 sec / Speedup: 1.000 / Efficiency: 100% / LogL: -170194\n",
      "Threads: 2 / Time: 19.579 sec / Speedup: 1.748 / Efficiency: 87% / LogL: -170194\n",
      "Threads: 3 / Time: 14.927 sec / Speedup: 2.293 / Efficiency: 76% / LogL: -170194\n",
      "Threads: 4 / Time: 12.572 sec / Speedup: 2.723 / Efficiency: 68% / LogL: -170194\n",
      "Threads: 5 / Time: 11.350 sec / Speedup: 3.016 / Efficiency: 60% / LogL: -170194\n",
      "Threads: 6 / Time: 10.814 sec / Speedup: 3.166 / Efficiency: 53% / LogL: -170194\n",
      "Threads: 7 / Time: 10.366 sec / Speedup: 3.303 / Efficiency: 47% / LogL: -170194\n",
      "BEST NUMBER OF THREADS: 5\n",
      "\n",
      "Perform fast likelihood tree search using LG+I+G model...\n",
      "Estimate model parameters (epsilon = 5.000)\n",
      "Perform nearest neighbor interchange...\n",
      "Estimate model parameters (epsilon = 1.000)\n",
      "1. Initial log-likelihood: -80553.902\n",
      "Optimal log-likelihood: -80553.408\n",
      "Proportion of invariable sites: 0.065\n",
      "Gamma shape alpha: 0.799\n",
      "Parameters optimization took 1 rounds (0.270 sec)\n",
      "Time for fast ML tree search: 2.822 seconds\n",
      "\n",
      "NOTE: ModelFinder requires 106 MB RAM!\n",
      "ModelFinder will test up to 168 protein models (sample size: 394) ...\n",
      " No. Model         -LnL         df  AIC          AICc         BIC\n",
      "  1  LG            88351.809    833 178369.618   1567813.618  181681.918\n",
      "  2  LG+I          87511.236    834 176690.472   1569470.472  180006.748\n",
      "  3  LG+G4         80620.361    834 162908.721   1555688.721  166224.998\n",
      "  4  LG+I+G4       80553.350    835 162776.701   1558896.701  166096.954\n",
      "  8  LG+F+I+G4     80703.715    854 163115.430   1623455.430  166511.233\n",
      " 12  WAG+I+G4      81050.070    835 163770.140   1559890.140  167090.393\n",
      " 16  WAG+F+I+G4    81277.702    854 164263.404   1624603.404  167659.208\n",
      " 20  JTT+I+G4      81757.653    835 165185.307   1561305.307  168505.560\n",
      " 24  JTT+F+I+G4    82314.488    854 166336.977   1626676.977  169732.781\n",
      " 28  JTTDCMut+I+G4 81714.326    835 165098.652   1561218.652  168418.905\n",
      " 32  JTTDCMut+F+I+G4 82249.744    854 166207.487   1626547.487  169603.291\n",
      " 36  DCMut+I+G4    81952.652    835 165575.304   1561695.304  168895.557\n",
      " 40  DCMut+F+I+G4  82183.075    854 166074.150   1626414.150  169469.953\n",
      " 44  VT+I+G4       81595.657    835 164861.314   1560981.314  168181.567\n",
      " 48  VT+F+I+G4     81887.790    854 165483.579   1625823.579  168879.383\n",
      " 52  PMB+I+G4      82343.710    835 166357.420   1562477.420  169677.673\n",
      " 56  PMB+F+I+G4    82388.150    854 166484.299   1626824.299  169880.103\n",
      " 60  Blosum62+I+G4 82219.731    835 166109.462   1562229.462  169429.715\n",
      " 64  Blosum62+F+I+G4 82299.800    854 166307.599   1626647.599  169703.403\n",
      " 68  Dayhoff+I+G4  81942.960    835 165555.920   1561675.920  168876.173\n",
      " 72  Dayhoff+F+I+G4 82176.031    854 166060.061   1626400.061  169455.865\n",
      " 76  mtREV+I+G4    86662.164    835 174994.328   1571114.328  178314.581\n",
      " 80  mtREV+F+I+G4  83785.808    854 169279.616   1629619.616  172675.419\n",
      " 84  mtART+I+G4    87084.852    835 175839.705   1571959.705  179159.958\n",
      " 88  mtART+F+I+G4  84533.145    854 170774.290   1631114.290  174170.094\n",
      " 92  mtZOA+I+G4    84516.593    835 170703.187   1566823.187  174023.440\n",
      " 96  mtZOA+F+I+G4  82872.310    854 167452.620   1627792.620  170848.423\n",
      "100  mtMet+I+G4    86449.697    835 174569.395   1570689.395  177889.648\n",
      "104  mtMet+F+I+G4  84071.930    854 169851.859   1630191.859  173247.663\n",
      "108  mtVer+I+G4    87685.892    835 177041.783   1573161.783  180362.036\n",
      "112  mtVer+F+I+G4  85935.704    854 173579.409   1633919.409  176975.212\n",
      "116  mtInv+I+G4    86155.411    835 173980.822   1570100.822  177301.075\n",
      "120  mtInv+F+I+G4  82727.030    854 167162.060   1627502.060  170557.863\n",
      "124  mtMAM+I+G4    89871.311    835 181412.623   1577532.623  184732.876\n",
      "128  mtMAM+F+I+G4  87380.964    854 176469.928   1636809.928  179865.732\n",
      "132  HIVb+I+G4     85801.317    835 173272.633   1569392.633  176592.886\n",
      "136  HIVb+F+I+G4   86307.982    854 174323.965   1634663.965  177719.768\n",
      "140  HIVw+I+G4     90296.446    835 182262.892   1578382.892  185583.145\n",
      "144  HIVw+F+I+G4   90260.216    854 182228.431   1642568.431  185624.235\n",
      "148  FLU+I+G4      84298.614    835 170267.227   1566387.227  173587.480\n",
      "152  FLU+F+I+G4    84799.168    854 171306.337   1631646.337  174702.140\n",
      "156  rtREV+I+G4    81317.070    835 164304.140   1560424.140  167624.393\n",
      "160  rtREV+F+I+G4  81022.047    854 163752.094   1624092.094  167147.898\n",
      "164  cpREV+I+G4    82129.139    835 165928.279   1562048.279  169248.532\n",
      "168  cpREV+F+I+G4  81815.608    854 165339.216   1625679.216  168735.020\n",
      "Akaike Information Criterion:           LG+I+G4\n",
      "Corrected Akaike Information Criterion: LG+G4\n",
      "Bayesian Information Criterion:         LG+I+G4\n",
      "Best-fit model: LG+I+G4 chosen according to BIC\n",
      "\n",
      "All model information printed to /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened.model.gz\n",
      "CPU time for ModelFinder: 631.191 seconds (0h:10m:31s)\n",
      "Wall-clock time for ModelFinder: 163.771 seconds (0h:2m:43s)\n",
      "Generating 1000 samples for ultrafast bootstrap (seed: 141879)...\n",
      "\n",
      "NOTE: 106 MB RAM (0 GB) is required!\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "Thoroughly optimizing +I+G parameters from 10 start values...\n",
      "Init pinv, alpha: 0.000, 0.799 / Estimate: 0.064, 0.799 / LogL: -80553.351\n",
      "Init pinv, alpha: 0.008, 0.799 / Estimate: 0.064, 0.799 / LogL: -80553.350\n",
      "Init pinv, alpha: 0.015, 0.799 / Estimate: 0.064, 0.799 / LogL: -80553.350\n",
      "Init pinv, alpha: 0.023, 0.799 / Estimate: 0.064, 0.799 / LogL: -80553.350\n",
      "Init pinv, alpha: 0.030, 0.799 / Estimate: 0.064, 0.799 / LogL: -80553.350\n",
      "Init pinv, alpha: 0.038, 0.799 / Estimate: 0.064, 0.799 / LogL: -80553.350\n",
      "Init pinv, alpha: 0.046, 0.799 / Estimate: 0.064, 0.799 / LogL: -80553.350\n",
      "Init pinv, alpha: 0.053, 0.799 / Estimate: 0.064, 0.799 / LogL: -80553.350\n",
      "Init pinv, alpha: 0.061, 0.799 / Estimate: 0.064, 0.798 / LogL: -80553.351\n",
      "Init pinv, alpha: 0.069, 0.799 / Estimate: 0.065, 0.799 / LogL: -80553.350\n",
      "Optimal pinv,alpha: 0.064, 0.799 / LogL: -80553.350\n",
      "\n",
      "Parameters optimization took 6.808 sec\n",
      "Computing ML distances based on estimated model parameters... 8.277 sec\n",
      "Computing BIONJ tree...\n",
      "0.047 seconds\n",
      "Log-likelihood of BIONJ tree: -81505.032\n",
      "--------------------------------------------------------------------\n",
      "|             INITIALIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Generating 98 parsimony trees... 9.130 second\n",
      "Computing log-likelihood of 98 initial trees ... 8.410 seconds\n",
      "Current best score: -80553.350\n",
      "\n",
      "Do NNI search on 20 best initial trees\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 1: -80541.660\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 2: -80454.532\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 3: -80432.919\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 7: -80421.989\n",
      "Iteration 10 / LogL: -80491.223 / Time: 0h:0m:44s\n",
      "Iteration 20 / LogL: -80517.948 / Time: 0h:1m:0s\n",
      "Finish initializing candidate tree set (20)\n",
      "Current best tree score: -80421.989 / CPU time: 51.641\n",
      "Number of iterations: 20\n",
      "--------------------------------------------------------------------\n",
      "|               OPTIMIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 21: -80409.003\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 24: -80399.031\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 25: -80385.383\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 27: -80377.177\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 30: -80358.133\n",
      "Iteration 30 / LogL: -80358.133 / Time: 0h:1m:21s (0h:4m:42s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 36: -80341.042\n",
      "Iteration 40 / LogL: -80357.439 / Time: 0h:1m:41s (0h:4m:9s left)\n",
      "Iteration 50 / LogL: -80407.365 / Time: 0h:2m:0s (0h:3m:31s left)\n",
      "Log-likelihood cutoff on original alignment: -80631.950\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 51: -80330.601\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 53: -80319.786\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 59: -80306.713\n",
      "Iteration 60 / LogL: -80317.228 / Time: 0h:2m:21s (0h:3m:57s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 70: -80293.852\n",
      "Iteration 70 / LogL: -80293.852 / Time: 0h:2m:41s (0h:3m:54s left)\n",
      "Iteration 80 / LogL: -80315.572 / Time: 0h:2m:59s (0h:3m:24s left)\n",
      "Iteration 90 / LogL: -80319.392 / Time: 0h:3m:19s (0h:2m:59s left)\n",
      "Iteration 100 / LogL: -80317.895 / Time: 0h:3m:37s (0h:2m:33s left)\n",
      "Log-likelihood cutoff on original alignment: -80551.389\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.952\n",
      "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 104: -80289.105\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 105: -80273.881\n",
      "Iteration 110 / LogL: -80309.177 / Time: 0h:3m:56s (0h:3m:26s left)\n",
      "Iteration 120 / LogL: -80291.860 / Time: 0h:4m:15s (0h:3m:2s left)\n",
      "Iteration 130 / LogL: -80294.771 / Time: 0h:4m:34s (0h:2m:39s left)\n",
      "Iteration 140 / LogL: -80369.282 / Time: 0h:4m:52s (0h:2m:16s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 146: -80273.265\n",
      "Iteration 150 / LogL: -80294.170 / Time: 0h:5m:11s (0h:3m:20s left)\n",
      "Log-likelihood cutoff on original alignment: -80551.389\n",
      "Iteration 160 / LogL: -80288.509 / Time: 0h:5m:30s (0h:2m:58s left)\n",
      "Iteration 170 / LogL: -80295.383 / Time: 0h:5m:48s (0h:2m:36s left)\n",
      "Iteration 180 / LogL: -80296.210 / Time: 0h:6m:6s (0h:2m:14s left)\n",
      "Iteration 190 / LogL: -80382.487 / Time: 0h:6m:24s (0h:1m:53s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 195: -80258.405\n",
      "Iteration 200 / LogL: -80318.381 / Time: 0h:6m:42s (0h:3m:11s left)\n",
      "Log-likelihood cutoff on original alignment: -80551.389\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.952\n",
      "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
      "Iteration 210 / LogL: -80278.957 / Time: 0h:7m:1s (0h:3m:1s left)\n",
      "Iteration 220 / LogL: -80337.934 / Time: 0h:7m:20s (0h:2m:40s left)\n",
      "Iteration 230 / LogL: -80297.092 / Time: 0h:7m:39s (0h:2m:20s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 231: -80257.558\n",
      "Iteration 240 / LogL: -80293.391 / Time: 0h:7m:58s (0h:3m:2s left)\n",
      "Iteration 250 / LogL: -80289.914 / Time: 0h:8m:17s (0h:2m:41s left)\n",
      "Log-likelihood cutoff on original alignment: -80522.929\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 252: -80254.209\n",
      "Iteration 260 / LogL: -80266.028 / Time: 0h:8m:37s (0h:3m:3s left)\n",
      "Iteration 270 / LogL: -80275.270 / Time: 0h:8m:56s (0h:2m:43s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 274: -80239.198\n",
      "Iteration 280 / LogL: -80339.818 / Time: 0h:9m:14s (0h:3m:6s left)\n",
      "Iteration 290 / LogL: -80263.361 / Time: 0h:9m:33s (0h:2m:46s left)\n",
      "Iteration 300 / LogL: -80319.859 / Time: 0h:9m:51s (0h:2m:26s left)\n",
      "Log-likelihood cutoff on original alignment: -80519.595\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.977\n",
      "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
      "Iteration 310 / LogL: -80249.551 / Time: 0h:10m:11s (0h:2m:58s left)\n",
      "Iteration 320 / LogL: -80376.686 / Time: 0h:10m:31s (0h:2m:38s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 329: -80238.507\n",
      "Iteration 330 / LogL: -80271.470 / Time: 0h:10m:50s (0h:3m:15s left)\n",
      "Iteration 340 / LogL: -80249.287 / Time: 0h:11m:8s (0h:2m:55s left)\n",
      "Iteration 350 / LogL: -80293.069 / Time: 0h:11m:26s (0h:2m:35s left)\n",
      "Log-likelihood cutoff on original alignment: -80519.595\n",
      "Iteration 360 / LogL: -80252.316 / Time: 0h:11m:46s (0h:2m:15s left)\n",
      "Iteration 370 / LogL: -80275.583 / Time: 0h:12m:4s (0h:1m:55s left)\n",
      "Iteration 380 / LogL: -80267.618 / Time: 0h:12m:22s (0h:1m:36s left)\n",
      "Iteration 390 / LogL: -80268.133 / Time: 0h:12m:41s (0h:1m:16s left)\n",
      "Iteration 400 / LogL: -80245.729 / Time: 0h:13m:0s (0h:0m:56s left)\n",
      "Log-likelihood cutoff on original alignment: -80519.595\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.973\n",
      "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 404: -80235.808\n",
      "Iteration 410 / LogL: -80255.972 / Time: 0h:13m:19s (0h:3m:3s left)\n",
      "Iteration 420 / LogL: -80511.085 / Time: 0h:13m:37s (0h:2m:43s left)\n",
      "Iteration 430 / LogL: -80260.660 / Time: 0h:13m:56s (0h:2m:24s left)\n",
      "Iteration 440 / LogL: -80265.188 / Time: 0h:14m:14s (0h:2m:4s left)\n",
      "Iteration 450 / LogL: -80306.596 / Time: 0h:14m:34s (0h:1m:45s left)\n",
      "Log-likelihood cutoff on original alignment: -80519.595\n",
      "Iteration 460 / LogL: -80287.997 / Time: 0h:14m:54s (0h:1m:25s left)\n",
      "Iteration 470 / LogL: -80273.902 / Time: 0h:15m:13s (0h:1m:6s left)\n",
      "Iteration 480 / LogL: -80239.440 / Time: 0h:15m:32s (0h:0m:46s left)\n",
      "Iteration 490 / LogL: -80269.615 / Time: 0h:15m:51s (0h:0m:27s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 493: -80231.114\n",
      "Iteration 500 / LogL: -80258.191 / Time: 0h:16m:9s (0h:3m:0s left)\n",
      "Log-likelihood cutoff on original alignment: -80519.595\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.978\n",
      "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
      "Iteration 510 / LogL: -80244.017 / Time: 0h:16m:29s (0h:2m:55s left)\n",
      "Iteration 520 / LogL: -80265.197 / Time: 0h:16m:49s (0h:2m:35s left)\n",
      "Iteration 530 / LogL: -80247.161 / Time: 0h:17m:10s (0h:2m:16s left)\n",
      "Iteration 540 / LogL: -80263.241 / Time: 0h:17m:27s (0h:1m:56s left)\n",
      "Iteration 550 / LogL: -80255.090 / Time: 0h:17m:47s (0h:1m:37s left)\n",
      "Log-likelihood cutoff on original alignment: -80519.595\n",
      "Iteration 560 / LogL: -80382.945 / Time: 0h:18m:7s (0h:1m:17s left)\n",
      "Iteration 570 / LogL: -80314.172 / Time: 0h:18m:25s (0h:0m:58s left)\n",
      "Iteration 580 / LogL: -80258.639 / Time: 0h:18m:43s (0h:0m:38s left)\n",
      "Iteration 590 / LogL: -80255.100 / Time: 0h:19m:2s (0h:0m:19s left)\n",
      "Iteration 600 / LogL: -80388.800 / Time: 0h:19m:20s (0h:0m:0s left)\n",
      "Log-likelihood cutoff on original alignment: -80519.595\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.987\n",
      "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
      "Iteration 610 / LogL: -80309.262 / Time: 0h:19m:40s (0h:2m:54s left)\n",
      "Iteration 620 / LogL: -80293.477 / Time: 0h:19m:59s (0h:2m:35s left)\n",
      "Iteration 630 / LogL: -80306.679 / Time: 0h:20m:18s (0h:2m:15s left)\n",
      "Iteration 640 / LogL: -80295.751 / Time: 0h:20m:36s (0h:1m:56s left)\n",
      "Iteration 650 / LogL: -80245.032 / Time: 0h:20m:53s (0h:1m:36s left)\n",
      "Log-likelihood cutoff on original alignment: -80519.595\n",
      "Iteration 660 / LogL: -80285.666 / Time: 0h:21m:13s (0h:1m:17s left)\n",
      "Iteration 670 / LogL: -80263.227 / Time: 0h:21m:31s (0h:0m:57s left)\n",
      "Iteration 680 / LogL: -80296.057 / Time: 0h:21m:50s (0h:0m:38s left)\n",
      "Iteration 690 / LogL: -80246.204 / Time: 0h:22m:10s (0h:0m:19s left)\n",
      "Iteration 700 / LogL: -80265.124 / Time: 0h:22m:29s (0h:0m:0s left)\n",
      "Log-likelihood cutoff on original alignment: -80519.595\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.990\n",
      "TREE SEARCH COMPLETED AFTER 700 ITERATIONS / Time: 0h:22m:29s\n",
      "\n",
      "--------------------------------------------------------------------\n",
      "|                    FINALIZING TREE SEARCH                        |\n",
      "--------------------------------------------------------------------\n",
      "Performs final model parameters optimization\n",
      "Estimate model parameters (epsilon = 0.010)\n",
      "1. Initial log-likelihood: -80231.114\n",
      "Optimal log-likelihood: -80231.114\n",
      "Proportion of invariable sites: 0.065\n",
      "Gamma shape alpha: 0.796\n",
      "Parameters optimization took 1 rounds (0.230 sec)\n",
      "BEST SCORE FOUND : -80231.114\n",
      "Creating bootstrap support values...\n",
      "Split supports printed to NEXUS file /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened.splits.nex\n",
      "Total tree length: 64.835\n",
      "\n",
      "Total number of iterations: 700\n",
      "CPU time used for tree search: 6783.731 sec (1h:53m:3s)\n",
      "Wall-clock time used for tree search: 1340.741 sec (0h:22m:20s)\n",
      "Total CPU time used: 6832.716 sec (1h:53m:52s)\n",
      "Total wall-clock time used: 1350.805 sec (0h:22m:30s)\n",
      "\n",
      "Computing bootstrap consensus tree...\n",
      "Reading input file /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened.splits.nex...\n",
      "418 taxa and 3569 splits.\n",
      "Consensus tree written to /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened.contree\n",
      "Reading input trees file /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened.contree\n",
      "Log-likelihood of consensus tree: -80227.546\n",
      "\n",
      "Analysis results written to: \n",
      "  IQ-TREE report:                /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened.iqtree\n",
      "  Maximum-likelihood tree:       /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened.treefile\n",
      "  Likelihood distances:          /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened.mldist\n",
      "\n",
      "Ultrafast bootstrap approximation results written to:\n",
      "  Split support values:          /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened.splits.nex\n",
      "  Consensus tree:                /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened.contree\n",
      "  Screen log file:               /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened.log\n",
      "\n",
      "Date and Time: Mon Oct 14 22:43:22 2024\n",
      "Phylogenetic tree constructed for /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened.fas\n",
      "IQ-TREE multicore version 2.0.7 for Linux 64-bit built Apr  1 2024\n",
      "Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,\n",
      "Heiko Schmidt, Dominik Schrempf, Michael Woodhams.\n",
      "\n",
      "Host:    BHSC352H6CQ724 (AVX2, FMA3, 31 GB RAM)\n",
      "Command: iqtree2 -s /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened.fas -t /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened.treefile -nt AUTO --ancestral --prefix /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened_anc\n",
      "Seed:    813295 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Mon Oct 14 22:43:22 2024\n",
      "Kernel:  AVX+FMA - auto-detect threads (32 CPU cores detected)\n",
      "\n",
      "Reading alignment file /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened.fas ... Fasta format detected\n",
      "Alignment most likely contains protein sequences\n",
      "Alignment has 418 sequences with 394 columns, 386 distinct patterns\n",
      "331 parsimony-informative, 36 singleton sites, 27 constant sites\n",
      "                             Gap/Ambiguity  Composition  p-value\n",
      "   1  Tu-C1_UniRef100_A0A8H8S4L3     1.52%    passed     47.19%\n",
      "   2  Tu-C1_UniRef100_A0A3E2HCY8     1.52%    passed     98.47%\n",
      "   3  Tu-C1_UniRef100_A0A8H4RH48     1.52%    passed     98.69%\n",
      "   4  Tu-C1_UniRef100_A0A2A9PMI8     1.27%    passed     99.71%\n",
      "   5  Tu-C1_UniRef100_A0A0D1XB65     1.52%    passed     55.26%\n",
      "   6  Tu-C1_UniRef100_A0A5M9JYY1     1.52%    passed     93.58%\n",
      "   7  Tu-C1_UniRef100_A0A094CMQ6     0.25%    passed     63.78%\n",
      "   8  Tu-C1_UniRef100_A0A1D2JH36     1.52%    passed     97.48%\n",
      "   9  Tu-C1_UniRef100_A0A4Q4W8P2     1.52%    passed     90.73%\n",
      "  10  Tu-C1_UniRef100_A0A093XNZ1     1.52%    passed     98.95%\n",
      "  11  Tu-C1_UniRef100_A0A6A6TY91     1.52%    passed     95.46%\n",
      "  12  Tu-C1_UniRef100_E9CV19         1.52%    passed     37.38%\n",
      "  13  Tu-C1_UniRef100_A0A0F4YMF9     1.52%    passed     97.22%\n",
      "  14  Tu-C1_UniRef100_A0A8H6FBP9     1.52%    passed     70.82%\n",
      "  15  Tu-C1_UniRef100_A0A420Y1Y2     1.27%    passed     89.35%\n",
      "  16  Tu-C1_UniRef100_A0A0G4MSC6     2.03%    passed     86.53%\n",
      "  17  Tu-C1_UniRef100_A0A178B9H0     1.52%    passed     92.85%\n",
      "  18  Tu-C1_UniRef100_A0A1L9SRY3     1.27%    passed     86.67%\n",
      "  19  Tu-C1_UniRef100_A0A0G2GRF6     1.02%    passed     96.85%\n",
      "  20  Tu-C1_UniRef100_A0A100II62     3.30%    passed     59.16%\n",
      "  21  Tu-C1_UniRef100_A0A5N6TSQ9     1.78%    passed     72.53%\n",
      "  22  Tu-C1_UniRef100_A0A5C3NTY7     1.02%    passed     47.85%\n",
      "  23  Tu-C1_UniRef100_A0A0H2S8E1     1.02%    passed     30.99%\n",
      "  24  Tu-C1_UniRef100_A0A4Z0AB69     0.51%    passed     89.35%\n",
      "  25  Tu-C1_UniRef100_A0A2H3IWI7     0.25%    passed     99.46%\n",
      "  26  Tu-C1_UniRef100_A0A8H7FNM9     0.76%    passed     93.94%\n",
      "  27  Tu-C1_UniRef100_A0A409VBL4     1.02%    passed     56.13%\n",
      "  28  Tu-C1_UniRef100_A0A0C3S7H8     1.02%    passed     81.96%\n",
      "  29  Tu-C1_UniRef100_A0A4S8MR95     1.02%    passed     91.06%\n",
      "  30  Tu-C1_UniRef100_A0A067QJI0     1.27%    passed     55.29%\n",
      "  31  Tu-C1_UniRef100_A0A067MI84     1.52%    passed     92.27%\n",
      "  32  Tu-C1_UniRef100_A0A8H8ME73     0.51%    passed     56.75%\n",
      "  33  Tu-C1_UniRef100_A0A286URG5     1.27%    passed     85.49%\n",
      "  34  Tu-C1_UniRef100_K5WPK8         0.51%    passed     23.98%\n",
      "  35  Tu-C1_UniRef100_A0A0L6W793     1.02%    passed     97.60%\n",
      "  36  Tu-C1_UniRef100_A0A0C2T4D2     1.52%    passed     85.32%\n",
      "  37  Tu-C1_UniRef100_A0A369K5C3     1.02%    passed     96.70%\n",
      "  38  Tu-C1_UniRef100_A0A164UBM0     1.27%    passed     86.31%\n",
      "  39  Tu-C1_UniRef100_A8N1J6         1.02%    passed     79.01%\n",
      "  40  Tu-C1_UniRef100_A0A8H5MDS0     1.52%    passed     63.23%\n",
      "  41  Tu-C1_UniRef100_K5WAB1         1.02%    passed     56.19%\n",
      "  42  Tu-C1_UniRef100_A0A0D2LMF0     1.27%    passed     91.35%\n",
      "  43  Tu-C1_UniRef100_A0A8H6W5X9     1.02%    passed     98.21%\n",
      "  44  Tu-C1_UniRef100_A0A5C3QYW4     1.02%    passed     98.92%\n",
      "  45  Tu-C1_UniRef100_A0A401G6V3     0.51%    passed     91.24%\n",
      "  46  Tu-C1_UniRef100_A0A8H6XF81     1.52%    passed     97.54%\n",
      "  47  Tu-C1_UniRef100_A0A8H5FZW9     1.02%    passed     29.19%\n",
      "  48  Tu-C1_UniRef100_A0A0C3BG74     0.76%    passed     97.53%\n",
      "  49  Tu-C1_UniRef100_A0A8H5AZZ4     2.03%    passed     88.92%\n",
      "  50  Tu-C1_UniRef100_A0A8H6TK30     4.31%    passed     87.18%\n",
      "  51  Tu-C1_UniRef100_A0A8H5H174     1.78%    passed     50.43%\n",
      "  52  Tu-C1_UniRef100_A0A4Y9YD42     3.55%    passed     79.36%\n",
      "  53  Tu-C1_UniRef100_A0A1M4SS91     0.25%    passed     99.21%\n",
      "  54  Tu-C1_UniRef100_A0A6I2YU27     0.25%    passed     97.80%\n",
      "  55  Tu-C1_UniRef100_A0A7V9BXY0     0.51%    passed     70.88%\n",
      "  56  Tu-C1_UniRef100_A0A073B1S9     0.25%    passed     68.02%\n",
      "  57  Tu-C1_UniRef100_A0A257RFX4     0.25%    passed     99.38%\n",
      "  58  Tu-C1_UniRef100_A0A931KVY1     0.25%    passed     97.30%\n",
      "  59  Tu-C1_UniRef100_A0A538JUN6     0.25%    passed     98.21%\n",
      "  60  Tu-C1_UniRef100_A0A6I3H0D8     0.25%    passed     98.02%\n",
      "  61  Tu-C1_UniRef100_A0A0D8HL47     0.25%    passed     99.63%\n",
      "  62  Tu-C1_UniRef100_A0A9D8H9Z6     0.51%    passed    100.00%\n",
      "  63  Tu-C1_UniRef100_A0A7W0HE09     0.25%    passed     97.20%\n",
      "  64  Tu-C1_UniRef100_A0A7Y3LMS3     0.25%    passed     97.96%\n",
      "  65  Tu-C1_UniRef100_D3CUS3         0.25%    passed     94.48%\n",
      "  66  Tu-C1_UniRef100_A0A7W1E1A4     0.25%    passed     99.67%\n",
      "  67  Tu-C1_UniRef100_A0A2E5ZGU2     0.25%    passed     99.05%\n",
      "  68  Tu-C1_UniRef100_A0A2U1FDK7     0.25%    passed     97.58%\n",
      "  69  Tu-C1_UniRef100_A0A7J9Z5K4     0.51%    passed     98.65%\n",
      "  70  Tu-C1_UniRef100_A0A966B5E8     0.25%    passed     95.51%\n",
      "  71  Tu-C1_UniRef100_A0A6B1H0D9     0.25%    passed     98.45%\n",
      "  72  Tu-C1_UniRef100_A0A060JN72     0.51%    passed     99.58%\n",
      "  73  Tu-C1_UniRef100_A0A849BFE4     0.51%    passed     99.81%\n",
      "  74  Tu-C1_UniRef100_A0A853BP90     0.25%    passed     83.69%\n",
      "  75  Tu-C1_UniRef100_A0A938J874     0.51%    passed     88.39%\n",
      "  76  Tu-C1_UniRef100_A0A6I2X6Y2     0.25%    passed     94.98%\n",
      "  77  Tu-C1_UniRef100_A0A935AY77     0.25%    passed     99.97%\n",
      "  78  Tu-C1_UniRef100_A0A7W0A023     0.25%    passed     99.47%\n",
      "  79  Tu-C1_UniRef100_A0A7Y0GW13     0.25%    passed     96.79%\n",
      "  80  Tu-C1_UniRef100_A0A538FG31     0.25%    passed     99.52%\n",
      "  81  Tu-C1_UniRef100_A0A970YZR4     0.25%    passed     99.95%\n",
      "  82  Tu-C1_UniRef100_A0A1M6X779     0.25%    passed     92.48%\n",
      "  83  Tu-C1_UniRef100_A0A6N9EZ57     0.25%    passed     98.39%\n",
      "  84  Tu-C1_UniRef100_A0A852VNY4     0.25%    passed     97.64%\n",
      "  85  Tu-C1_UniRef100_A0A0K1JM72     0.25%    passed     98.40%\n",
      "  86  Tu-C1_UniRef100_A0A6I2ZFX3     0.25%    passed     96.70%\n",
      "  87  Tu-C1_UniRef100_A0A2R2W5W5     0.25%    passed     97.65%\n",
      "  88  Tu-C1_UniRef100_A0A6N7I5Y3     0.25%    passed     96.28%\n",
      "  89  Tu-C1_UniRef100_A0A938JSC1     0.51%    passed     99.92%\n",
      "  90  Tu-C1_UniRef100_A0A7J9VS26     0.25%    passed     72.42%\n",
      "  91  Tu-C1_UniRef100_A0A6I6RWI3     0.25%    passed     95.17%\n",
      "  92  Tu-C1_UniRef100_A0A7J9XLA5     0.51%    passed     92.84%\n",
      "  93  Tu-C1_UniRef100_A0A7T3ZWG6     0.25%    passed     97.18%\n",
      "  94  Tu-C1_UniRef100_A0A2W4JPN0     0.25%    passed     94.28%\n",
      "  95  Tu-C1_UniRef100_A0A4R2K0Y4     0.25%    passed     98.78%\n",
      "  96  Tu-C1_UniRef100_A0A938YCV9     0.25%    passed     92.43%\n",
      "  97  Tu-C1_UniRef100_A0A7Y9DK38     0.25%    passed     99.70%\n",
      "  98  Tu-C1_UniRef100_A0A917U0U1     0.25%    passed     99.17%\n",
      "  99  Tu-C1_UniRef100_A0A8J7KNB2     0.25%    passed     99.69%\n",
      " 100  Tu-C1_UniRef100_A0A3Q9UCN8     0.25%    passed     76.37%\n",
      " 101  Tu-C1_UniRef100_D3QB76         0.25%    passed     95.25%\n",
      " 102  Tu-C1_UniRef100_A0A0W8IKR5     0.25%    passed     99.20%\n",
      " 103  Tu-C1_UniRef100_S2WJY6         0.51%    passed     98.90%\n",
      " 104  Tu-C1_UniRef100_A0A1G9Z043     0.25%    passed     99.92%\n",
      " 105  Tu-C1_UniRef100_A0A388NRG8     0.51%    passed     98.41%\n",
      " 106  Tu-C1_UniRef100_A0A810KYL2     0.25%    passed     97.24%\n",
      " 107  Tu-C1_UniRef100_E5XR78         0.51%    passed     83.68%\n",
      " 108  Tu-C1_UniRef100_A0A2G1XI85     0.51%    passed     89.29%\n",
      " 109  Tu-C1_UniRef100_A0A286EN96     0.25%    passed     95.34%\n",
      " 110  Tu-C1_UniRef100_A0A0H5D5X6     0.00%    passed     96.90%\n",
      " 111  Tu-C1_UniRef100_A0A916UM33     0.25%    passed     94.63%\n",
      " 112  Tu-C1_UniRef100_A0A4Q9KFG4     0.25%    passed     99.84%\n",
      " 113  Tu-C1_UniRef100_A0A6P2C7A7     0.51%    passed     98.92%\n",
      " 114  Tu-C1_UniRef100_W5WL60         0.25%    passed     93.64%\n",
      " 115  Tu-C1_UniRef100_A0A1B1MMS2     0.25%    passed     94.99%\n",
      " 116  Tu-C1_UniRef100_A0A383S3W6     0.51%    passed     70.53%\n",
      " 117  Tu-C1_UniRef100_Q8FS84         0.51%    passed     98.71%\n",
      " 118  Tu-C1_UniRef100_A0A9D2ZRP5     0.51%    passed     99.94%\n",
      " 119  Tu-C1_UniRef100_A0A2W5IE43     0.51%    passed     99.31%\n",
      " 120  Tu-C1_UniRef100_A0A9D6XNY2     0.76%    passed     88.55%\n",
      " 121  Tu-C1_UniRef100_A0A563E4C0     0.25%    passed     99.61%\n",
      " 122  Tu-C1_UniRef100_A0A542DJE1     0.51%    passed     49.30%\n",
      " 123  Tu-C1_UniRef100_A0A838A7W8     0.51%    passed     74.69%\n",
      " 124  Tu-C1_UniRef100_A0A1G6HWD1     0.51%    passed     98.50%\n",
      " 125  Tu-C1_UniRef100_A0A7W1R2Z1     0.51%    passed     97.88%\n",
      " 126  Tu-C1_UniRef100_A0A8I1S6W6     0.25%    passed     96.94%\n",
      " 127  Tu-C1_UniRef100_A0A6G8KZR5     0.25%    passed     94.38%\n",
      " 128  Tu-C1_UniRef100_A0A542ZSG3     0.00%    passed     89.84%\n",
      " 129  Tu-C1_UniRef100_I7L9W8         0.76%    passed     95.65%\n",
      " 130  Tu-C1_UniRef100_A0A0T1W7T8     0.51%    passed     86.49%\n",
      " 131  Tu-C1_UniRef100_A0A255EBU4     0.51%    passed     87.29%\n",
      " 132  Tu-C1_UniRef100_A0A7W3LLP6     0.25%    failed      3.33%\n",
      " 133  Tu-C1_UniRef100_A0A543IP01     0.25%    passed      9.44%\n",
      " 134  Tu-C1_UniRef100_A0A6H9YFH4     0.25%    passed     24.61%\n",
      " 135  Tu-C1_UniRef100_A0A4R4M586     0.25%    failed      1.86%\n",
      " 136  Tu-C1_UniRef100_A0A3M2LJI2     0.25%    failed      1.02%\n",
      " 137  Tu-C1_UniRef100_A0A5D0NZ14     0.25%    failed      0.10%\n",
      " 138  Tu-C1_UniRef100_A0A6I4M993     0.25%    failed      4.59%\n",
      " 139  Tu-C1_UniRef100_A0A7K0BVX6     0.51%    failed      2.77%\n",
      " 140  Tu-C1_UniRef100_A0A1H6CUV6     0.25%    passed     10.66%\n",
      " 141  Tu-C1_UniRef100_A0A7W3R6L5     1.02%    failed      1.26%\n",
      " 142  Tu-C1_UniRef100_A0A3D9SNF9     0.25%    failed      4.19%\n",
      " 143  Tu-C1_UniRef100_A0A4R9EXQ9     0.25%    passed      8.77%\n",
      " 144  Tu-C1_UniRef100_A0A1Q4UYS8     1.52%    failed      1.24%\n",
      " 145  Tu-C1_UniRef100_A0A7X3ACK6     1.52%    failed      0.69%\n",
      " 146  Tu-C1_UniRef100_A0A944KRH2     0.51%    failed      2.29%\n",
      " 147  Tu-C1_UniRef100_A0A919DET7     1.27%    failed      3.94%\n",
      " 148  Tu-C1_UniRef100_A0A6G4XFJ0     1.27%    passed     17.18%\n",
      " 149  Tu-C1_UniRef100_A0A5C6JZ56     1.02%    failed      0.10%\n",
      " 150  Tu-C1_UniRef100_A0A2U3C0W4     1.52%    failed      3.34%\n",
      " 151  Tu-C1_UniRef100_A0A0C5GA75     1.52%    failed      0.17%\n",
      " 152  Tu-C1_UniRef100_A0A124ED28     1.27%    failed      0.04%\n",
      " 153  Tu-C1_UniRef100_V6JMA1         1.27%    failed      4.05%\n",
      " 154  Tu-C1_UniRef100_A0A918GNF4     1.52%    failed      0.07%\n",
      " 155  Tu-C1_UniRef100_A0A7K0CB26     1.52%    failed      1.13%\n",
      " 156  Tu-C1_UniRef100_A0A7H0IMG2     1.27%    failed      0.23%\n",
      " 157  Tu-C1_UniRef100_A0A6G3RX88     1.02%    failed      1.53%\n",
      " 158  Tu-C1_UniRef100_A0A2S4XVP4     0.76%    failed      0.05%\n",
      " 159  Tu-C1_UniRef100_A0A6G4TSH5     1.52%    passed      8.90%\n",
      " 160  Tu-C1_UniRef100_A0A8T4IWN8     1.02%    failed      1.95%\n",
      " 161  Tu-C1_UniRef100_A0A365Z5T0     1.52%    failed      0.11%\n",
      " 162  Tu-C1_UniRef100_A0A1I2AXZ0     0.76%    failed      2.56%\n",
      " 163  Tu-C1_UniRef100_A0A505CYL6     1.52%    failed      2.26%\n",
      " 164  Tu-C1_UniRef100_A0A3D0R5B3     1.52%    failed      0.24%\n",
      " 165  Tu-C1_UniRef100_A0A370B0A1     1.27%    failed      2.86%\n",
      " 166  Tu-C1_UniRef100_A0A221NTP9     1.52%    failed      1.73%\n",
      " 167  Tu-C1_UniRef100_A0A2P2GEE7     1.27%    failed      0.12%\n",
      " 168  Tu-C1_UniRef100_A0A1H8NGK5     1.02%    failed      0.86%\n",
      " 169  Tu-C1_UniRef100_A0A6B3HEY9     1.27%    failed      0.11%\n",
      " 170  Tu-C1_UniRef100_A0A2M9KMZ3     0.76%    failed      0.39%\n",
      " 171  Tu-C1_UniRef100_A0A939F4V4     0.51%    failed      3.34%\n",
      " 172  Tu-C1_UniRef100_A0A1Y2NTG1     1.27%    failed      0.07%\n",
      " 173  Tu-C1_UniRef100_A0A918NHC8     1.27%    failed      0.81%\n",
      " 174  Tu-C1_UniRef100_A0A372M4V6     1.52%    failed      2.80%\n",
      " 175  Tu-C1_UniRef100_A0A918U6A5     1.52%    failed      0.04%\n",
      " 176  Tu-C1_UniRef100_A0A1V2RTZ0     1.52%    failed      0.22%\n",
      " 177  Tu-C1_UniRef100_A0A3M0ICR4     1.52%    failed      0.36%\n",
      " 178  Tu-C1_UniRef100_A0A919GYQ9     0.76%    failed      0.20%\n",
      " 179  Tu-C1_UniRef100_S5UYY7         1.27%    failed      1.47%\n",
      " 180  Tu-C1_UniRef100_A0A930ZX77     1.27%    failed      0.12%\n",
      " 181  Tu-C1_UniRef100_A0A8G1ZKR6     1.27%    failed      0.05%\n",
      " 182  Tu-C1_UniRef100_A0A1H0N0L2     1.02%    failed      2.51%\n",
      " 183  Tu-C1_UniRef100_A0A918MFV0     1.27%    failed      0.85%\n",
      " 184  Tu-C1_UniRef100_A0A5P2CC72     1.27%    failed      2.35%\n",
      " 185  Tu-C1_UniRef100_A0A5N8WYA0     1.27%    failed      1.96%\n",
      " 186  Tu-C1_UniRef100_A0A7K3BKF0     1.52%    failed      0.50%\n",
      " 187  Tu-C1_UniRef100_A0A2R4SY40     1.02%    failed      0.38%\n",
      " 188  Tu-C1_UniRef100_A0A345XQ45     1.52%    failed      0.51%\n",
      " 189  Tu-C1_UniRef100_A0A2B8B284     1.52%    failed      0.10%\n",
      " 190  Tu-C1_UniRef100_A0A4R4TEU6     1.52%    failed      0.06%\n",
      " 191  Tu-C1_UniRef100_A0A7M3LV78     1.52%    failed      0.33%\n",
      " 192  Tu-C1_UniRef100_A0A7K2WM35     0.76%    failed      3.87%\n",
      " 193  Tu-C1_UniRef100_A0A101UD17     1.27%    failed      0.06%\n",
      " 194  Tu-C1_UniRef100_A0A7J0C1M1     1.27%    failed      0.13%\n",
      " 195  Tu-C1_UniRef100_D7BY48         1.52%    failed      1.57%\n",
      " 196  Tu-C1_UniRef100_A0A0X3W0C3     1.02%    failed      3.11%\n",
      " 197  Tu-C1_UniRef100_A0A0N1NII9     1.27%    failed      0.62%\n",
      " 198  Tu-C1_UniRef100_A0A1H9BGA1     1.27%    failed      0.75%\n",
      " 199  Tu-C1_UniRef100_A0A1M7Q0J6     0.76%    failed      0.52%\n",
      " 200  Tu-C1_UniRef100_A0A7K3DNB6     0.76%    failed      0.11%\n",
      " 201  Tu-C1_UniRef100_A3KKP5         1.27%    failed      0.01%\n",
      " 202  Tu-C1_UniRef100_A0A0F7G286     1.27%    failed      4.45%\n",
      " 203  Tu-C1_UniRef100_A0A1H6CDW4     1.02%    failed      0.71%\n",
      " 204  Tu-C1_UniRef100_A0A387HNB4     1.02%    failed      2.10%\n",
      " 205  Tu-C1_UniRef100_A0A919BZ14     1.52%    failed      0.02%\n",
      " 206  Tu-C1_UniRef100_A0A6B2RYS8     0.76%    failed      0.40%\n",
      " 207  Tu-C1_UniRef100_A0A7T1TBV1     1.52%    failed      0.55%\n",
      " 208  Tu-C1_UniRef100_A0A5B8IC35     1.27%    failed      0.28%\n",
      " 209  Tu-C1_UniRef100_A0A3R9XQG8     0.76%    failed      0.00%\n",
      " 210  Tu-C1_UniRef100_A0A919L5M4     2.03%    failed      1.06%\n",
      " 211  Tu-C1_UniRef100_A0A344U3D7     0.76%    failed      0.00%\n",
      " 212  Tu-C1_UniRef100_A0A919HD98     1.52%    failed      0.08%\n",
      " 213  Tu-C1_UniRef100_A0A974AYX1     1.02%    failed      0.22%\n",
      " 214  Tu-C1_UniRef100_A0A4Z0HB80     1.02%    failed      0.03%\n",
      " 215  Tu-C1_UniRef100_F3NTJ2         1.78%    failed      0.00%\n",
      " 216  Tu-C1_UniRef100_A0A4R3YAC7     0.25%    passed     99.79%\n",
      " 217  Tu-C1_UniRef100_A0A143WPR6     0.00%    passed     71.61%\n",
      " 218  Tu-C1_UniRef100_A0A209AQB3     0.25%    passed     99.98%\n",
      " 219  Tu-C1_UniRef100_U3TY99         0.00%    passed     99.98%\n",
      " 220  Tu-C1_UniRef100_A0A1N7CGP3     0.25%    passed     96.41%\n",
      " 221  Tu-C1_UniRef100_A0A5P2UM61     0.00%    passed     99.08%\n",
      " 222  Tu-C1_UniRef100_A0A1V8NCP4     0.25%    passed     40.17%\n",
      " 223  Tu-C1_UniRef100_A0A4D6YLG4     0.00%    passed     75.77%\n",
      " 224  C1_UniRef100_A0A420HRM0        0.25%    passed     95.38%\n",
      " 225  Tu-C1_UniRef100_F7XY84         0.00%    passed     99.96%\n",
      " 226  Tu-C1_UniRef100_A0A2C8CVX6     0.00%    passed     97.80%\n",
      " 227  Tu-C1_UniRef100_A0A4D6YN62     0.00%    passed     82.04%\n",
      " 228  Tu-C1_UniRef100_Q1LSY4         0.25%    passed     83.98%\n",
      " 229  Tu-C1_UniRef100_Q7VRP0         0.00%    passed     67.26%\n",
      " 230  Tu-C1_UniRef100_A0A0E3U3M2     0.00%    passed     98.68%\n",
      " 231  Tu-C1_UniRef100_D5BVA9         0.25%    passed     99.87%\n",
      " 232  Tu-C1_UniRef100_A0A1C7AF74     0.25%    passed     98.30%\n",
      " 233  Tu-C1_UniRef100_A0A9C9VER6     0.25%    passed     95.68%\n",
      " 234  Tu-C1_UniRef100_L0DYY6         0.25%    passed     98.21%\n",
      " 235  Tu-C1_UniRef100_A0A1Y0NA39     0.00%    passed     99.45%\n",
      " 236  Tu-C1_UniRef100_A0A1C3L4A1     0.25%    passed     33.76%\n",
      " 237  Tu-C1_UniRef100_A0A3S8T3C9     0.25%    passed     99.88%\n",
      " 238  Tu-C1_UniRef100_A0A450TB23     0.51%    passed     96.18%\n",
      " 239  Tu-C1_UniRef100_A0A6S6UKH2     0.51%    passed     88.49%\n",
      " 240  Tu-C1_UniRef100_A0A949D297     0.25%    passed     96.98%\n",
      " 241  Tu-C1_UniRef100_A0A9D0P1A1     0.25%    passed     89.59%\n",
      " 242  Tu-C1_UniRef100_A0A963D6S8     0.51%    passed     99.19%\n",
      " 243  Tu-C1_UniRef100_A0A2A4UKP8     0.25%    passed     93.84%\n",
      " 244  Tu-C1_UniRef100_A0A2E8KCX4     0.51%    passed     96.47%\n",
      " 245  Tu-C1_UniRef100_A0A1U9JI97     0.25%    passed     99.96%\n",
      " 246  Tu-C1_UniRef100_A0P4F9         0.51%    passed     96.35%\n",
      " 247  Tu-C1_UniRef100_D9SCC2         0.25%    passed     93.47%\n",
      " 248  Tu-C1_UniRef100_A0A1G0K9T7     0.51%    passed     98.61%\n",
      " 249  Tu-C1_UniRef100_A0A086WEB3     0.00%    passed     99.85%\n",
      " 250  Tu-C1_UniRef100_A0A7X1P490     0.25%    passed     99.73%\n",
      " 251  Tu-C1_UniRef100_A0A2A2EQH0     0.25%    passed     99.52%\n",
      " 252  Tu-C1_UniRef100_A0A9D7KDJ5     0.25%    passed     97.29%\n",
      " 253  Tu-C1_UniRef100_A0A7D4SXK4     0.25%    passed     99.12%\n",
      " 254  Tu-C1_UniRef100_A0A661EVB5     0.25%    passed     98.96%\n",
      " 255  Tu-C1_UniRef100_A0A536YJ92     0.25%    passed     93.82%\n",
      " 256  Tu-C1_UniRef100_A0A521XAZ9     0.25%    passed     99.80%\n",
      " 257  Tu-C1_UniRef100_A0A212KMK6     0.25%    passed     99.71%\n",
      " 258  Tu-C1_UniRef100_A0A836S6B4     0.25%    passed     83.04%\n",
      " 259  Tu-C1_UniRef100_A0A6N6VIM4     0.25%    passed     99.86%\n",
      " 260  Tu-C1_UniRef100_A0A9D8CB88     0.25%    passed     96.27%\n",
      " 261  Tu-C1_UniRef100_A0A2E4GJ30     0.00%    passed     99.62%\n",
      " 262  Tu-C1_UniRef100_A0A1H5XFN9     1.52%    passed     98.22%\n",
      " 263  Tu-C1_UniRef100_Q9P9Q9         0.25%    passed     99.97%\n",
      " 264  Tu-C1_UniRef100_A0A1Q6UCA4     0.25%    passed     89.84%\n",
      " 265  Tu-C1_UniRef100_A0A061QEV0     0.00%    passed     98.68%\n",
      " 266  Tu-C1_UniRef100_A0A7W5ELG9     0.25%    passed     89.13%\n",
      " 267  Tu-C1_UniRef100_A0A7R6W0L4     0.25%    passed     60.09%\n",
      " 268  Tu-C1_UniRef100_A0A1N7K9D5     0.00%    passed     99.89%\n",
      " 269  Tu-C1_UniRef100_A0A3B0YSK9     0.25%    passed     97.90%\n",
      " 270  Tu-C1_UniRef100_U2E2F6         0.51%    passed     99.87%\n",
      " 271  Tu-C1_UniRef100_A0A7V8AKP9     0.25%    passed     99.12%\n",
      " 272  Tu-C1_UniRef100_A0A2E7K4G9     0.25%    passed     99.67%\n",
      " 273  Tu-C1_UniRef100_A0A1Z8KXW1     0.51%    passed     99.87%\n",
      " 274  Tu-C1_UniRef100_A0A2D7AAY0     0.25%    passed     99.08%\n",
      " 275  Tu-C1_UniRef100_A0A0Q4NEQ5     0.25%    passed     94.08%\n",
      " 276  Tu-C1_UniRef100_M4S4E3         1.27%    passed    100.00%\n",
      " 277  Tu-C1_UniRef100_A0A3A1Y0W9     3.30%    passed     99.82%\n",
      " 278  Tu-C1_UniRef100_A0A2N6N749     0.25%    passed     99.45%\n",
      " 279  Tu-C1_UniRef100_A0A6L5G3T8     0.51%    passed     99.55%\n",
      " 280  Tu-C1_UniRef100_A0A523FG19     0.25%    passed     99.25%\n",
      " 281  Tu-C1_UniRef100_A0A6L7S3G9     1.52%    passed     99.22%\n",
      " 282  Tu-C1_UniRef100_A0A523FVQ2     0.25%    passed     98.24%\n",
      " 283  Tu-C1_UniRef100_A0A2S6T5Q0     0.25%    passed     85.92%\n",
      " 284  Tu-C1_UniRef100_A0A9E1RA41     0.00%    passed     93.43%\n",
      " 285  Tu-C1_UniRef100_A0A9D8GMY6     0.25%    passed     98.82%\n",
      " 286  Tu-C1_UniRef100_A0A2E5DE46     0.25%    passed     89.62%\n",
      " 287  Tu-C1_UniRef100_A0A0Q6BEW9     2.54%    passed     99.80%\n",
      " 288  Tu-C1_UniRef100_A0A2E5DWF1     1.78%    passed     99.95%\n",
      " 289  Tu-C1_UniRef100_A0A5C7SRI2     0.25%    passed     99.84%\n",
      " 290  Tu-C1_UniRef100_A0A6B0XYM7     0.51%    passed     99.83%\n",
      " 291  Tu-C1_UniRef100_A0A2E4GZN0     0.51%    passed     93.59%\n",
      " 292  Tu-C1_UniRef100_M5E919         0.00%    passed     99.85%\n",
      " 293  Tu-C1_UniRef100_A0A1R4HHK1     0.25%    passed     96.64%\n",
      " 294  Tu-C1_UniRef100_A0A6L8C4X0     1.52%    passed     97.84%\n",
      " 295  Tu-C1_UniRef100_A0A4R5Q2H1     0.76%    passed     99.81%\n",
      " 296  Tu-C1_UniRef100_A0A6N6JNP2     1.52%    passed     95.33%\n",
      " 297  Tu-C1_UniRef100_A0A0S8BXZ4     0.00%    passed     93.79%\n",
      " 298  Tu-C1_UniRef100_I3TNZ7         3.81%    passed     99.87%\n",
      " 299  Tu-C1_UniRef100_A0A0Q4QUV2     0.51%    passed     97.23%\n",
      " 300  Tu-C1_UniRef100_A0A7V8W6U5     0.00%    passed     93.32%\n",
      " 301  Tu-C1_UniRef100_A0A6S6SHZ6     6.35%    passed     88.16%\n",
      " 302  Tu-C1_UniRef100_A0A0Q5DC00     1.52%    passed     99.36%\n",
      " 303  Tu-C2_UniRef100_E9UZA8         0.51%    passed     46.70%\n",
      " 304  Tu-C2_UniRef100_A0A810MQ54     0.25%    failed      0.55%\n",
      " 305  Tu-C2_UniRef100_A0A7Y9ZI65     0.51%    passed      6.88%\n",
      " 306  Tu-C2_UniRef100_A0A853BXE2     0.51%    failed      3.00%\n",
      " 307  Tu-C2_UniRef100_A0A8J4DQ16     0.25%    passed     30.80%\n",
      " 308  Tu-C2_UniRef100_A0A4V6PAK6     0.51%    passed     30.33%\n",
      " 309  Tu-C2_UniRef100_A0A561BKB4     0.25%    passed     42.52%\n",
      " 310  Tu-C2_UniRef100_A0A849DU62     0.51%    failed      1.36%\n",
      " 311  TrG-C3_UniRef100_G7E513        2.54%    failed      0.13%\n",
      " 312  Tu-C4_UniRef100_A0A3B3XB95     0.25%    passed     53.07%\n",
      " 313  Tu-C4_UniRef100_A0A9D3NN79     0.51%    passed     77.84%\n",
      " 314  Tu-C4_UniRef100_A0A8C9VMI8     0.51%    passed     93.78%\n",
      " 315  Tu-C4_UniRef100_A0A8C4G2G9     0.25%    passed     93.86%\n",
      " 316  Tu-C4_UniRef100_A0A8C5DN64     0.25%    passed     94.33%\n",
      " 317  Tu-C4_UniRef100_A0A0F8CCZ2     0.25%    passed     93.24%\n",
      " 318  Tu-C4_UniRef100_A0A8C5ABC5     0.25%    passed     94.80%\n",
      " 319  Tu-C4_UniRef100_A0A3Q3KPB3     0.25%    passed     62.99%\n",
      " 320  Tu-C4_UniRef100_A0A8C6UL35     0.51%    passed     91.65%\n",
      " 321  Tu-C4_UniRef100_A0A3P8W7T3     0.25%    passed     82.74%\n",
      " 322  Tu-C4_UniRef100_A0A671VQ45     0.25%    passed     87.27%\n",
      " 323  Tu-C4_UniRef100_Q5BJ17         0.25%    passed     71.18%\n",
      " 324  Tu-C4_UniRef100_H3AI71         0.51%    passed     55.42%\n",
      " 325  Tu-C4_UniRef100_A0A6P3VPT1     0.25%    passed     99.28%\n",
      " 326  Tu-C4_UniRef100_A0A8C6KEK9     0.25%    passed     85.45%\n",
      " 327  Tu-C4_UniRef100_A0A3Q3R8Z5     0.51%    passed     75.71%\n",
      " 328  Tu-C4_UniRef100_A0A3Q3A9B3     0.25%    passed     57.28%\n",
      " 329  Tu-C4_UniRef100_H9GA61         0.51%    passed     81.26%\n",
      " 330  Tu-C4_UniRef100_A0A3Q3GE42     0.51%    passed     91.45%\n",
      " 331  Tu-C4_UniRef100_A0A9D3X2F3     0.51%    passed     85.19%\n",
      " 332  Tu-C4_UniRef100_A0A8C5R5N1     0.51%    passed     53.45%\n",
      " 333  Tu-C4_UniRef100_A0A3P8Y9G5     0.51%    passed     94.11%\n",
      " 334  Tu-C4_UniRef100_A0A6I9MXI0     0.51%    passed     75.53%\n",
      " 335  Tu-C4_UniRef100_A0A6P7YKF5     0.51%    passed     17.48%\n",
      " 336  Tu-C4_UniRef100_A0A6J0TJB1     0.51%    passed     85.85%\n",
      " 337  Tu-C4_UniRef100_A0A8J6EB46     0.51%    passed     38.10%\n",
      " 338  Tu-C4_UniRef100_A0A8C8VFQ3     0.51%    passed     89.25%\n",
      " 339  Tu-C4_UniRef100_V9KW97         0.51%    passed     90.33%\n",
      " 340  Tu-C4_UniRef100_A0A6P8REA3     0.51%    passed     63.55%\n",
      " 341  Tu-C4_UniRef100_A0A8D2KR75     0.51%    passed     67.66%\n",
      " 342  Tu-C4_UniRef100_A0A8D0BAB1     0.51%    passed     96.75%\n",
      " 343  Tu-C4_UniRef100_A0A674N0P1     0.25%    passed     78.60%\n",
      " 344  Tu-C4_UniRef100_A0A8B9QN43     0.51%    passed     69.52%\n",
      " 345  Tu-C4_UniRef100_A0A5A9N9Y8     0.25%    passed     99.71%\n",
      " 346  Tu-C4_UniRef100_A0A6J2Q291     0.76%    passed     54.36%\n",
      " 347  Tu-C4_UniRef100_A0A2K5W259     0.25%    passed     88.74%\n",
      " 348  Tu-C4_UniRef100_A0A674N8H6     0.25%    passed     33.26%\n",
      " 349  Tu-C4_UniRef100_A0A3Q2TPY7     4.82%    passed     81.00%\n",
      " 350  Tu-C4_UniRef100_A0A8D0TAV6     3.81%    passed     57.36%\n",
      " 351  Tu-C6_UniRef100_A0A653BRQ4     1.27%    passed     53.99%\n",
      " 352  Tu-C6_UniRef100_A0A7R8UBP5     1.02%    passed     53.14%\n",
      " 353  Tu-C6_UniRef100_A0A1A9ZPI0     1.27%    passed     29.08%\n",
      " 354  Tu-C6_UniRef100_A0A8J6LQI3     1.27%    passed     48.39%\n",
      " 355  Tu-C6_UniRef100_A0A182Q8D0     1.02%    passed     51.75%\n",
      " 356  Tu-C6_UniRef100_A0A2J7QJE0     1.52%    passed     73.95%\n",
      " 357  Tu-C6_UniRef100_A0A0K8TSH2     1.02%    passed     49.11%\n",
      " 358  Tu-C6_UniRef100_A0A1W4WQN0     1.27%    passed     51.40%\n",
      " 359  Tu-C6_UniRef100_D6X0G1         1.27%    passed     35.70%\n",
      " 360  Tu-C6_UniRef100_J3JU35         1.27%    passed     61.43%\n",
      " 361  Tu-C6_UniRef100_A0A6B2E7A0     1.27%    passed     87.20%\n",
      " 362  Tu-C6_UniRef100_A0A0L0C7B8     1.27%    passed     43.63%\n",
      " 363  Tu-C6_UniRef100_A0A067R0M6     1.52%    passed     82.07%\n",
      " 364  Tu-C6_UniRef100_U5ESU5         1.27%    passed     15.88%\n",
      " 365  Tu-C6_UniRef100_A0A1Y1JRP4     1.27%    passed     32.14%\n",
      " 366  Tu-C6_UniRef100_A0A6P7GQG5     1.27%    passed     74.14%\n",
      " 367  Tu-C6_UniRef100_A0A1L8DRI0     1.27%    passed     68.11%\n",
      " 368  Tu-C6_UniRef100_A0A336LPD7     1.27%    passed     13.36%\n",
      " 369  Tu-C6_UniRef100_A0A7R9AWA3     1.27%    passed     53.68%\n",
      " 370  Tu-C6_UniRef100_A0A6M2DP98     1.27%    passed     52.29%\n",
      " 371  Tu-C6_UniRef100_A0A0T6B4E2     1.02%    passed     27.43%\n",
      " 372  Tu-C6_UniRef100_A0A8K0DJL1     1.27%    passed     30.20%\n",
      " 373  Tu-C6_UniRef100_A0A1J1HQB0     1.02%    passed     73.98%\n",
      " 374  Tu-C6_UniRef100_A0A6J2YUP2     1.27%    passed     46.84%\n",
      " 375  Tu-C6_UniRef100_A0A834MHC9     5.58%    passed     36.99%\n",
      " 376  Tu-C6_UniRef100_A0A1B0DQV9     1.27%    passed     97.04%\n",
      " 377  C7_UniRef100_A0A8S4R2T2        2.54%    failed      2.15%\n",
      " 378  TrG-C7_UniRef100_A0A8J9Y142    2.28%    failed      4.02%\n",
      " 379  C7_UniRef100_A0A8S3XLH0        2.54%    failed      1.47%\n",
      " 380  C7_UniRef100_A0A8S4F4S8        1.27%    passed     11.48%\n",
      " 381  C7_UniRef100_A0A6J1WDU5        2.54%    failed      3.33%\n",
      " 382  C7_UniRef100_A0A6J2KHG4        2.28%    passed     28.95%\n",
      " 383  C7_UniRef100_A0A7E5W3U6        2.28%    failed      1.29%\n",
      " 384  TrG-C7_UniRef100_A0A821XFK6    2.54%    failed      3.38%\n",
      " 385  TrG-C7_UniRef100_A0A8S0Z0F4    2.28%    failed      2.47%\n",
      " 386  Tu-C7_UniRef100_A0A0N1IQ97     2.28%    passed     15.95%\n",
      " 387  TrG-C7_UniRef100_A0A922SEV8    2.28%    passed     11.14%\n",
      " 388  C8_UniRef100_A0A1I8M4A3        2.03%    passed     24.47%\n",
      " 389  umt-C8_UniRef100_A0A6J1MLA1    2.03%    failed      3.00%\n",
      " 390  umt-C8_UniRef100_B4LLC6        1.52%    failed      3.21%\n",
      " 391  TrG-C8_UniRef100_A0A1I8P1X1    2.03%    failed      1.33%\n",
      " 392  umt-C8_UniRef100_B4J9Z8        2.54%    passed      8.48%\n",
      " 393  umt-C8_UniRef100_A0A3B0JQR7    2.79%    failed      3.71%\n",
      " 394  umt-C8_UniRef100_A0A6J2UIW1    3.05%    failed      1.27%\n",
      " 395  umt-C8_UniRef100_B4MNZ3        1.27%    failed      3.77%\n",
      " 396  Tu-C8_UniRef100_A0A034W4T8     2.54%    failed      2.83%\n",
      " 397  Tu-C8_UniRef100_W8BJV6         2.79%    failed      0.37%\n",
      " 398  C8_UniRef100_A0A1A9USP6        2.54%    passed     24.17%\n",
      " 399  Tu-C9_UniRef100_A0A416ECT3     0.25%    passed     82.42%\n",
      " 400  Tu-C9_UniRef100_A0A1T5CFY0     0.25%    passed     98.67%\n",
      " 401  Tu-C9_UniRef100_A0A9D1C4D7     0.51%    passed     90.37%\n",
      " 402  Tu-C9_UniRef100_A0A173ZYQ1     0.25%    passed     88.74%\n",
      " 403  Tu-C9_UniRef100_A0A2J8B391     0.25%    passed     57.51%\n",
      " 404  Tu-C9_UniRef100_G5GHR3         0.25%    passed     78.47%\n",
      " 405  Tu-C9_UniRef100_A0A1H9D8W8     0.76%    passed     95.66%\n",
      " 406  Tu-C9_UniRef100_A0A0H5SK50     0.25%    passed     96.56%\n",
      " 407  Tu-C9_UniRef100_A0A3P2AIZ5     0.25%    passed     87.82%\n",
      " 408  Tu-C9_UniRef100_A0A316R7W5     0.76%    passed     44.06%\n",
      " 409  Tu-C9_UniRef100_A0A1I0P5K4     0.00%    passed     76.36%\n",
      " 410  Tu-C9_UniRef100_A0A1C5SHU8     0.25%    passed     51.38%\n",
      " 411  Tu-C9_UniRef100_A0A348ZL69     0.76%    passed     68.18%\n",
      " 412  Tu-C9_UniRef100_A0A965YNT0     0.76%    passed     95.46%\n",
      " 413  Tu-C9_UniRef100_A0A1G4PQ21     0.76%    passed     94.90%\n",
      " 414  Tu-C9_UniRef100_R9MHP0         0.76%    passed     66.07%\n",
      " 415  Tu-C9_UniRef100_A0A1M7J4D3     0.76%    passed     85.22%\n",
      " 416  Tu-C9_UniRef100_A0A7U9WYI3     0.76%    passed     40.54%\n",
      " 417  Tu-C9_UniRef100_A0A1V5PZ15     0.76%    passed     69.90%\n",
      " 418  Tu-C9_UniRef100_X8GTG1         0.76%    passed     62.72%\n",
      "****  TOTAL                          0.87%  97 sequences failed composition chi2 test (p-value<5%; df=19)\n",
      "\n",
      "Reading input tree file /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened.treefile ...\n",
      "Measuring multi-threading efficiency up to 32 CPU cores\n",
      "Increase to 10 rounds for branch lengths\n",
      "14 trees examined\n",
      "Threads: 1 / Time: 33.626 sec / Speedup: 1.000 / Efficiency: 100% / LogL: -169370\n",
      "Threads: 2 / Time: 19.386 sec / Speedup: 1.735 / Efficiency: 87% / LogL: -169370\n",
      "Threads: 3 / Time: 14.145 sec / Speedup: 2.377 / Efficiency: 79% / LogL: -169370\n",
      "Threads: 4 / Time: 12.090 sec / Speedup: 2.781 / Efficiency: 70% / LogL: -169370\n",
      "Threads: 5 / Time: 11.164 sec / Speedup: 3.012 / Efficiency: 60% / LogL: -169370\n",
      "Threads: 6 / Time: 10.847 sec / Speedup: 3.100 / Efficiency: 52% / LogL: -169370\n",
      "Threads: 7 / Time: 10.735 sec / Speedup: 3.132 / Efficiency: 45% / LogL: -169370\n",
      "BEST NUMBER OF THREADS: 5\n",
      "\n",
      "Perform fast likelihood tree search using LG+I+G model...\n",
      "Estimate model parameters (epsilon = 5.000)\n",
      "Perform nearest neighbor interchange...\n",
      "Estimate model parameters (epsilon = 1.000)\n",
      "1. Initial log-likelihood: -80230.295\n",
      "Optimal log-likelihood: -80230.293\n",
      "Proportion of invariable sites: 0.065\n",
      "Gamma shape alpha: 0.796\n",
      "Parameters optimization took 1 rounds (0.270 sec)\n",
      "Time for fast ML tree search: 1.560 seconds\n",
      "\n",
      "NOTE: ModelFinder requires 263 MB RAM!\n",
      "ModelFinder will test up to 546 protein models (sample size: 394) ...\n",
      " No. Model         -LnL         df  AIC          AICc         BIC\n",
      "  1  LG            88029.864    833 177725.728   1567169.728  181038.029\n",
      "  2  LG+I          87189.561    834 176047.121   1568827.121  179363.398\n",
      "  3  LG+G4         80296.412    834 162260.825   1555040.825  165577.101\n",
      "  4  LG+I+G4       80230.294    835 162130.589   1558250.589  165450.842\n",
      "  5  LG+R2         82105.789    835 165881.577   1562001.577  169201.830\n",
      "  6  LG+R3         80586.914    837 162847.829   1565659.829  166176.034\n",
      "  7  LG+R4         80206.793    839 162091.587   1571611.587  165427.745\n",
      "  8  LG+R5         80075.875    841 161833.751   1578077.751  165177.862\n",
      "  9  LG+R6         79976.743    843 161639.487   1584623.487  164991.551\n",
      " 10  LG+R7         79924.304    845 161538.607   1591278.607  164898.624\n",
      " 11  LG+R8         79901.924    847 161497.847   1598009.847  164865.816\n",
      " 12  LG+R9         79894.539    849 161487.078   1604787.078  164863.000\n",
      " 13  LG+R10        79889.097    851 161480.194   1611584.194  164864.068\n",
      " 24  LG+F+R8       80059.988    866 161851.976   1663495.976  165295.496\n",
      " 25  LG+F+R9       80055.290    868 161846.581   1670430.581  165298.053\n",
      " 37  WAG+R8        80445.820    847 162585.641   1599097.641  165953.610\n",
      " 38  WAG+R9        80438.921    849 162575.842   1605875.842  165951.764\n",
      " 39  WAG+R10       80438.892    851 162579.784   1612683.784  165963.659\n",
      " 50  WAG+F+R8      80691.896    866 163115.791   1664759.791  166559.311\n",
      " 51  WAG+F+R9      80687.610    868 163111.220   1671695.220  166562.693\n",
      " 63  JTT+R8        81097.363    847 163888.726   1600400.726  167256.695\n",
      " 64  JTT+R9        81090.373    849 163878.746   1607178.746  167254.667\n",
      " 65  JTT+R10       81086.952    851 163875.904   1613979.904  167259.779\n",
      " 76  JTT+F+R8      81688.627    866 165109.254   1666753.254  168552.774\n",
      " 77  JTT+F+R9      81676.237    868 165088.473   1673672.473  168539.946\n",
      " 78  JTT+F+R10     81674.732    870 165089.464   1680629.464  168548.889\n",
      " 89  JTTDCMut+R8   81053.619    847 163801.239   1600313.239  167169.208\n",
      " 90  JTTDCMut+R9   81046.820    849 163791.640   1607091.640  167167.562\n",
      " 91  JTTDCMut+R10  81043.421    851 163788.842   1613892.842  167172.716\n",
      "102  JTTDCMut+F+R8 81623.412    866 164978.824   1666622.824  168422.344\n",
      "103  JTTDCMut+F+R9 81610.972    868 164957.944   1673541.944  168409.417\n",
      "104  JTTDCMut+F+R10 81609.369    870 164958.738   1680498.738  168418.163\n",
      "115  DCMut+R8      81305.455    847 164304.910   1600816.910  167672.879\n",
      "116  DCMut+R9      81300.322    849 164298.644   1607598.644  167674.566\n",
      "128  DCMut+F+R8    81565.755    866 164863.511   1666507.511  168307.031\n",
      "129  DCMut+F+R9    81561.503    868 164859.006   1673443.006  168310.479\n",
      "141  VT+R8         80986.343    847 163666.685   1600178.685  167034.655\n",
      "142  VT+R9         80981.676    849 163661.351   1606961.351  167037.273\n",
      "154  VT+F+R8       81302.087    866 164336.173   1665980.173  167779.693\n",
      "155  VT+F+R9       81298.239    868 164332.478   1672916.478  167783.950\n",
      "167  PMB+R8        81739.619    847 165173.238   1601685.238  168541.207\n",
      "168  PMB+R9        81735.666    849 165169.332   1608469.332  168545.254\n",
      "180  PMB+F+R8      81806.011    866 165344.021   1666988.021  168787.541\n",
      "181  PMB+F+R9      81803.660    868 165343.321   1673927.321  168794.793\n",
      "193  Blosum62+R8   81619.140    847 164932.280   1601444.280  168300.249\n",
      "194  Blosum62+R9   81615.408    849 164928.815   1608228.815  168304.737\n",
      "206  Blosum62+F+R8 81716.416    866 165164.831   1666808.831  168608.351\n",
      "207  Blosum62+F+R9 81713.052    868 165162.105   1673746.105  168613.578\n",
      "219  Dayhoff+R8    81294.145    847 164282.289   1600794.289  167650.259\n",
      "220  Dayhoff+R9    81288.967    849 164275.935   1607575.935  167651.857\n",
      "232  Dayhoff+F+R8  81556.846    866 164845.693   1666489.693  168289.212\n",
      "233  Dayhoff+F+R9  81552.710    868 164841.419   1673425.419  168292.892\n",
      "245  mtREV+R8      85863.029    847 173420.059   1609932.059  176788.028\n",
      "246  mtREV+R9      85849.064    849 173396.128   1616696.128  176772.050\n",
      "247  mtREV+R10     85847.218    851 173396.437   1623500.437  176780.311\n",
      "258  mtREV+F+R8    83039.455    866 167810.909   1669454.909  171254.429\n",
      "259  mtREV+F+R9    83024.132    868 167784.264   1676368.264  171235.736\n",
      "260  mtREV+F+R10   82997.837    870 167735.674   1683275.674  171195.099\n",
      "271  mtART+R8      86189.261    847 174072.521   1610584.521  177440.490\n",
      "272  mtART+R9      86155.652    849 174009.304   1617309.304  177385.226\n",
      "273  mtART+R10     86153.349    851 174008.698   1624112.698  177392.573\n",
      "284  mtART+F+R8    83736.940    866 169205.880   1670849.880  172649.400\n",
      "285  mtART+F+R9    83723.359    868 169182.717   1677766.717  172634.190\n",
      "286  mtART+F+R10   83702.352    870 169144.704   1684684.704  172604.130\n",
      "297  mtZOA+R8      83698.417    847 169090.835   1605602.835  172458.804\n",
      "298  mtZOA+R9      83675.340    849 169048.680   1612348.680  172424.602\n",
      "299  mtZOA+R10     83644.364    851 168990.727   1619094.727  172374.602\n",
      "310  mtZOA+F+R8    82130.410    866 165992.821   1667636.821  169436.341\n",
      "311  mtZOA+F+R9    82118.040    868 165972.080   1674556.080  169423.553\n",
      "312  mtZOA+F+R10   82095.467    870 165930.935   1681470.935  169390.360\n",
      "323  mtMet+R8      85631.169    847 172956.337   1609468.337  176324.307\n",
      "324  mtMet+R9      85615.926    849 172929.852   1616229.852  176305.774\n",
      "325  mtMet+R10     85613.943    851 172929.885   1623033.885  176313.760\n",
      "336  mtMet+F+R8    83334.099    866 168400.199   1670044.199  171843.719\n",
      "337  mtMet+F+R9    83320.603    868 168377.207   1676961.207  171828.680\n",
      "338  mtMet+F+R10   83296.865    870 168333.729   1683873.729  171793.154\n",
      "349  mtVer+R8      86855.330    847 175404.660   1611916.660  178772.629\n",
      "350  mtVer+R9      86834.610    849 175367.219   1618667.219  178743.141\n",
      "351  mtVer+R10     86802.421    851 175306.843   1625410.843  178690.717\n",
      "362  mtVer+F+R8    85146.780    866 172025.559   1673669.559  175469.079\n",
      "363  mtVer+F+R9    85132.876    868 172001.752   1680585.752  175453.225\n",
      "364  mtVer+F+R10   85117.127    870 171974.254   1687514.254  175433.679\n",
      "375  mtInv+R8      85381.398    847 172456.796   1608968.796  175824.765\n",
      "376  mtInv+R9      85358.682    849 172415.364   1615715.364  175791.286\n",
      "377  mtInv+R10     85351.403    851 172404.806   1622508.806  175788.680\n",
      "388  mtInv+F+R8    82036.098    866 165804.195   1667448.195  169247.715\n",
      "389  mtInv+F+R9    82018.269    868 165772.538   1674356.538  169224.010\n",
      "390  mtInv+F+R10   82018.046    870 165776.092   1681316.092  169235.517\n",
      "401  mtMAM+R8      88883.759    847 179461.518   1615973.518  182829.487\n",
      "402  mtMAM+R9      88841.483    849 179380.967   1622680.967  182756.889\n",
      "403  mtMAM+R10     88831.453    851 179364.906   1629468.906  182748.781\n",
      "414  mtMAM+F+R8    86476.065    866 174684.130   1676328.130  178127.650\n",
      "415  mtMAM+F+R9    86457.028    868 174650.056   1683234.056  178101.528\n",
      "416  mtMAM+F+R10   86453.884    870 174647.768   1690187.768  178107.193\n",
      "427  HIVb+R8       85073.893    847 171841.785   1608353.785  175209.755\n",
      "428  HIVb+R9       85058.147    849 171814.294   1615114.294  175190.216\n",
      "429  HIVb+R10      85026.296    851 171754.591   1621858.591  175138.466\n",
      "440  HIVb+F+R8     85587.150    866 172906.301   1674550.301  176349.820\n",
      "441  HIVb+F+R9     85571.100    868 172878.200   1681462.200  176329.672\n",
      "442  HIVb+F+R10    85555.229    870 172850.457   1688390.457  176309.883\n",
      "453  HIVw+R8       89560.934    847 180815.869   1617327.869  184183.838\n",
      "454  HIVw+R9       89550.380    849 180798.759   1624098.759  184174.681\n",
      "455  HIVw+R10      89527.488    851 180756.977   1630860.977  184140.852\n",
      "466  HIVw+F+R8     89538.030    866 180808.060   1682452.060  184251.580\n",
      "467  HIVw+F+R9     89528.301    868 180792.602   1689376.602  184244.075\n",
      "468  HIVw+F+R10    89518.056    870 180776.112   1696316.112  184235.537\n",
      "479  FLU+R8        83613.992    847 168921.985   1605433.985  172289.954\n",
      "480  FLU+R9        83605.954    849 168909.907   1612209.907  172285.829\n",
      "481  FLU+R10       83582.608    851 168867.216   1618971.216  172251.091\n",
      "492  FLU+F+R8      84106.452    866 169944.904   1671588.904  173388.424\n",
      "493  FLU+F+R9      84093.358    868 169922.715   1678506.715  173374.188\n",
      "494  FLU+F+R10     84075.016    870 169890.032   1685430.032  173349.458\n",
      "505  rtREV+R8      80646.647    847 162987.294   1599499.294  166355.263\n",
      "506  rtREV+R9      80642.716    849 162983.431   1606283.431  166359.353\n",
      "518  rtREV+F+R8    80376.649    866 162485.298   1664129.298  165928.818\n",
      "519  rtREV+F+R9    80372.868    868 162481.736   1671065.736  165933.208\n",
      "531  cpREV+R8      81479.062    847 164652.123   1601164.123  168020.093\n",
      "532  cpREV+R9      81475.749    849 164649.498   1607949.498  168025.420\n",
      "544  cpREV+F+R8    81182.266    866 164096.532   1665740.532  167540.052\n",
      "545  cpREV+F+R9    81179.945    868 164095.891   1672679.891  167547.363\n",
      "Akaike Information Criterion:           LG+R10\n",
      "Corrected Akaike Information Criterion: LG+G4\n",
      "Bayesian Information Criterion:         LG+R9\n",
      "Best-fit model: LG+R9 chosen according to BIC\n",
      "\n",
      "All model information printed to /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened_anc.model.gz\n",
      "CPU time for ModelFinder: 27570.458 seconds (7h:39m:30s)\n",
      "Wall-clock time for ModelFinder: 5360.412 seconds (1h:29m:20s)\n",
      "\n",
      "NOTE: 237 MB RAM (0 GB) is required!\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "1. Initial log-likelihood: -79894.539\n",
      "2. Current log-likelihood: -79893.649\n",
      "3. Current log-likelihood: -79893.255\n",
      "4. Current log-likelihood: -79893.044\n",
      "Optimal log-likelihood: -79892.936\n",
      "Site proportion and rates:  (0.246,0.055) (0.116,0.174) (0.055,0.300) (0.076,0.464) (0.142,0.766) (0.154,1.278) (0.086,2.010) (0.083,2.974) (0.041,4.577)\n",
      "Parameters optimization took 4 rounds (12.269 sec)\n",
      "\n",
      "Do NNI search on 1 best initial trees\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 1: -79888.625\n",
      "Finish initializing candidate tree set (2)\n",
      "Current best tree score: -79888.625 / CPU time: 13.426\n",
      "Number of iterations: 1\n",
      "--------------------------------------------------------------------\n",
      "|               OPTIMIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Iteration 10 / LogL: -79907.995 / Time: 0h:1m:1s\n",
      "Iteration 20 / LogL: -79911.147 / Time: 0h:1m:45s\n",
      "Iteration 30 / LogL: -79928.097 / Time: 0h:2m:28s (0h:6m:3s left)\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 38: -79886.151\n",
      "Iteration 40 / LogL: -79937.407 / Time: 0h:3m:22s (0h:8m:29s left)\n",
      "Iteration 50 / LogL: -79911.697 / Time: 0h:4m:3s (0h:7m:17s left)\n",
      "Iteration 60 / LogL: -79974.332 / Time: 0h:4m:48s (0h:6m:21s left)\n",
      "Iteration 70 / LogL: -79890.657 / Time: 0h:5m:32s (0h:5m:27s left)\n",
      "Iteration 80 / LogL: -79912.248 / Time: 0h:6m:13s (0h:4m:34s left)\n",
      "Iteration 90 / LogL: -79926.666 / Time: 0h:6m:55s (0h:3m:44s left)\n",
      "Iteration 100 / LogL: -79933.248 / Time: 0h:7m:36s (0h:2m:55s left)\n",
      "Iteration 110 / LogL: -79910.947 / Time: 0h:8m:15s (0h:2m:7s left)\n",
      "Iteration 120 / LogL: -79900.460 / Time: 0h:8m:57s (0h:1m:21s left)\n",
      "Iteration 130 / LogL: -80003.001 / Time: 0h:9m:36s (0h:0m:35s left)\n",
      "TREE SEARCH COMPLETED AFTER 139 ITERATIONS / Time: 0h:10m:12s\n",
      "\n",
      "--------------------------------------------------------------------\n",
      "|                    FINALIZING TREE SEARCH                        |\n",
      "--------------------------------------------------------------------\n",
      "Performs final model parameters optimization\n",
      "Estimate model parameters (epsilon = 0.010)\n",
      "1. Initial log-likelihood: -79886.151\n",
      "2. Current log-likelihood: -79885.578\n",
      "3. Current log-likelihood: -79885.366\n",
      "4. Current log-likelihood: -79885.262\n",
      "5. Current log-likelihood: -79885.224\n",
      "6. Current log-likelihood: -79885.207\n",
      "7. Current log-likelihood: -79885.195\n",
      "Optimal log-likelihood: -79885.183\n",
      "Site proportion and rates:  (0.241,0.052) (0.123,0.172) (0.049,0.306) (0.078,0.452) (0.143,0.761) (0.157,1.282) (0.088,2.044) (0.080,2.995) (0.041,4.557)\n",
      "Parameters optimization took 7 rounds (15.676 sec)\n",
      "BEST SCORE FOUND : -79885.183\n",
      "Ancestral state probabilities printed to /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened_anc.state\n",
      "Total tree length: 50.094\n",
      "\n",
      "Total number of iterations: 139\n",
      "CPU time used for tree search: 3080.192 sec (0h:51m:20s)\n",
      "Wall-clock time used for tree search: 600.672 sec (0h:10m:0s)\n",
      "Total CPU time used: 3226.094 sec (0h:53m:46s)\n",
      "Total wall-clock time used: 629.272 sec (0h:10m:29s)\n",
      "\n",
      "Analysis results written to: \n",
      "  IQ-TREE report:                /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened_anc.iqtree\n",
      "  Maximum-likelihood tree:       /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened_anc.treefile\n",
      "  Ancestral state:               /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened_anc.state\n",
      "  Screen log file:               /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened_anc.log\n",
      "\n",
      "Date and Time: Tue Oct 15 00:26:48 2024\n",
      "Ancestral sequences reconstructed for /home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/Concatenated_All_TAX_shortened.fas\n",
      "Requested analyses completed.\n"
     ]
    }
   ],
   "source": [
    "# Decide which process to run\n",
    "from glob import glob\n",
    "import subprocess\n",
    "\n",
    "analysis_type = input(\"Enter 'tree' to run only the phylogenetic tree, or 'both' to run both tree and ancestral reconstruction: \")\n",
    "\n",
    "# List all _aligned.fasta files in the directory\n",
    "#output_dir = '/home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/' from earlier cells\n",
    "directory = '/home/yuhong/localcolabfold/SEQ/Cluster/sel_pdb/uniprot_tree/'  \n",
    "fasta_files = glob(os.path.join(directory, '*.fas'))\n",
    "# Process each file according to the selected analysis type\n",
    "for fasta_file in fasta_files:\n",
    "    if analysis_type.lower() == 'both':\n",
    "        run_phylogenetic_and_ancestral(fasta_file)\n",
    "    elif analysis_type.lower() == 'tree':\n",
    "        run_phylogenetic_tree(fasta_file)\n",
    "    else:\n",
    "        print(\"Invalid input. Please enter 'tree' or 'both'.\")\n",
    "\n",
    "print(\"Requested analyses completed.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fb9b4cd0-ac58-42be-be35-32c12b8d35d0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define three different category_color mappings for proteins\n",
    "from Bio import Phylo\n",
    "category_colors_A = {\n",
    "    \"EF2\": \"#ff0000\",  # red\n",
    "    \"EFG\": \"#c0c0c0\",  # silver grey\n",
    "}\n",
    "\n",
    "category_colors_B = {\n",
    "    \"EF-G2\": \"#00ffff\",  # cyan\n",
    "    \"Tu\": \"#9B30FF\", # vibrant purple\n",
    "    \"TrG\": \"#ff69b4\",  # hot pink\n",
    "    \"Tetra\": \"#0000ff\",  # blue\n",
    "    \"EF-G1\":\"#ffd700\", #sunshine yellow\n",
    "    \"umt\": \"#B22222\", # brick brown\n",
    "}\n",
    "\n",
    "category_colors_C = {\n",
    "    \"Ploop\": \"#800080\",  # purple\n",
    "    \"GtpB\": \"#00ff00\",  # green\n",
    "    \"RF2\": \"#ffa500\",  # orange\n",
    "    #\"TrG\": \"#ff69b4\",  # hot pink\n",
    "}\n",
    "\n",
    "def process_tree_file(tree_file, category_colors, dataset_label, output_suffix):\n",
    "    tree = Phylo.read(tree_file, 'newick')\n",
    "\n",
    "    # Note: Changed SEPARATOR to SPACE. Revert to TAB if SPACE is not supported by iTOL.\n",
    "    dataset_content = (f'DATASET_COLORSTRIP\\n'\n",
    "                       f'SEPARATOR SPACE\\n'\n",
    "                       f'DATASET_LABEL {dataset_label}\\n'\n",
    "                       f'COLOR #ff0000\\n'\n",
    "                       f'DATA\\n')\n",
    "\n",
    "    for label in tree.get_terminals():\n",
    "        label_name = label.name if label.name else \"\"\n",
    "        color = '#ffffff'  # Default color\n",
    "        for category, category_color in category_colors.items():\n",
    "            if category in label_name:\n",
    "                color = category_color\n",
    "                break\n",
    "        dataset_content += f'{label.name} {color}\\n'\n",
    "\n",
    "    # Define the output file path\n",
    "    output_file_path = os.path.splitext(tree_file)[0] + output_suffix + '.txt'\n",
    "    with open(output_file_path, 'w') as out_file:\n",
    "        out_file.write(dataset_content)\n",
    "    print(f\"iTOL dataset file created: {output_file_path}\")\n",
    "\n",
    "def process_all_treefiles(directory):\n",
    "    for filename in os.listdir(directory):\n",
    "        if filename.endswith('.treefile'):\n",
    "            tree_file_path = os.path.join(directory, filename)\n",
    "            print(f\"Processing {tree_file_path}...\")\n",
    "\n",
    "            # Process for each category color mapping\n",
    "           # process_tree_file(tree_file_path, category_colors_A, \"labelProA\", \"_itol_datasetA\")\n",
    "            process_tree_file(tree_file_path, category_colors_B, \"labelProB\", \"_itol_datasetB\")\n",
    "            #process_tree_file(tree_file_path, category_colors_C, \"labelProC\", \"_itol_datasetC\")\n",
    "\n",
    "# Path to the directory containing .treefile files\n",
    "directory_path = output_dir  # Assigning the same value\n",
    "\n",
    "# Ensure the directory exists\n",
    "if os.path.isdir(directory_path):\n",
    "    process_all_treefiles(directory_path)\n",
    "else:\n",
    "    print(\"Invalid path or directory does not exist.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3279d5b1-78ed-4180-a5b9-4e6df8f0050a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define conditions for label background colors\n",
    "branch_colors_conditions = {\n",
    "\t\"C1\":\t\"#a52a2a\",\t# brown\n",
    "\t\"C2\":\t\"#000000\",\t# black\n",
    "\t\"C3\":\t\"#c0c0c0\",\t# silver\n",
    "\t\"C4\":\t\"#008080\",\t# teal\n",
    "\t\"C5\":\t\"#ffd700\",\t# yellow\n",
    "\t\"C6\":\t\"#ff69b4\",\t# hot pink\n",
    "\t\"C7\":\t\"#ffa500\",\t# orange\n",
    "\t\"C8\":\t\"#800080\",\t# purple\n",
    "\t\"C9\":\t\"#0000ff\",\t# blue\n",
    "\t#\"000450\":\t\"#00ff00\",\t# green\n",
    "\t#\"003050\":\t\"#00ffff\",\t# cyan\n",
    "\t#\"000040\":\t\"#ff0000\",\t# red\n",
    "    # Add more conditions as needed\n",
    "}\n",
    "\n",
    "def generate_itol_branch_colors(tree_file):\n",
    "    \"\"\"Generate an iTOL TREE_COLORS dataset for branch colors.\"\"\"\n",
    "    tree = Phylo.read(tree_file, 'newick')\n",
    "    output_filename = os.path.splitext(tree_file)[0] + '_branch_colors.txt'\n",
    "    \n",
    "    with open(output_filename, 'w') as out_file:\n",
    "        out_file.write(\"TREE_COLORS\\n\")\n",
    "        out_file.write(\"#lines starting with a hash are comments and ignored during parsing\\n\")\n",
    "        out_file.write(\"SEPARATOR SPACE\\n\")\n",
    "        out_file.write(\"DATA\\n\")\n",
    "        \n",
    "        for clade in tree.find_clades():\n",
    "            if clade.name:\n",
    "                for keyword, color in branch_colors_conditions.items():\n",
    "                    if keyword in clade.name:\n",
    "                        # Write the branch color information, including style and width\n",
    "                        # Format: NODE_ID branch COLOR normal 1\n",
    "                        out_file.write(f\"{clade.name} branch {color} normal 1\\n\")\n",
    "                        break\n",
    "\n",
    "    print(f\"iTOL branch colors dataset generated: {output_filename}\")\n",
    "\n",
    "import os\n",
    "import glob  # Import the glob module to use glob.glob()\n",
    "\n",
    "def process_directory(directory):\n",
    "    \"\"\"Process all .treefile files in the specified directory to generate iTOL branch colors datasets.\"\"\"\n",
    "    # Ensure that you use glob.glob() here to search for all .treefile files\n",
    "    for tree_file in glob.glob(os.path.join(directory, '*.treefile')):\n",
    "        print(f\"Processing {tree_file}...\")\n",
    "        generate_itol_branch_colors(tree_file)  # Ensure this function is defined elsewhere\n",
    "\n",
    "# Set the directory path where your .treefile files are located\n",
    "directory_path = output_dir  # Assigning the same value\n",
    "\n",
    "# Ensure the directory exists and process the files\n",
    "if os.path.isdir(directory_path):\n",
    "    process_directory(directory_path)\n",
    "else:\n",
    "    print(\"Invalid path or directory does not exist.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6f19ebc7-4925-4feb-ad59-3562a58a40f5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# test of ID extraction\n",
    "def extract_ids(nwk_file):\n",
    "    with open(nwk_file, 'r') as file:\n",
    "        newick_content = file.read()\n",
    "\n",
    "    # Regular expression to match taxon names (excluding numbers and entries starting with numbers)\n",
    "    taxon_pattern = re.compile(r'([^\\s,:();]+)')\n",
    "\n",
    "    # Find all matches\n",
    "    taxon_ids = re.findall(taxon_pattern, newick_content)\n",
    "\n",
    "    # Remove duplicates and filter out entries that are purely numeric or start with a digit\n",
    "    filtered_ids = {taxon_id for taxon_id in set(taxon_ids) if not re.match(r'^\\d', taxon_id)}\n",
    "\n",
    "    # Extract only the last part of the ID after the last hyphen\n",
    "    final_ids = {taxon_id.split('-')[-1] for taxon_id in filtered_ids}\n",
    "\n",
    "    return final_ids\n",
    "\n",
    "# Specify the path to your Newick (.nwk or .treefile) file\n",
    "nwk_file_path = \"./list1list2_TAX_shortened_aligned.treefile\"\n",
    "\n",
    "# Extract IDs from the file\n",
    "ids = extract_ids(nwk_file_path)\n",
    "\n",
    "# Generate the output filename based on the input filename\n",
    "input_filename = os.path.basename(nwk_file_path)\n",
    "output_filename = f\"{os.path.splitext(input_filename)[0]}_IDs.csv\"\n",
    "output_filepath = os.path.join(os.path.dirname(nwk_file_path), output_filename)\n",
    "\n",
    "# Write the IDs to a CSV file\n",
    "with open(output_filepath, 'w', newline='') as csvfile:\n",
    "    writer = csv.writer(csvfile)\n",
    "    writer.writerow(['Taxon_ID'])  # Write the header\n",
    "    for taxon_id in sorted(ids):\n",
    "        writer.writerow([taxon_id])\n",
    "\n",
    "print(f\"Extracted IDs have been written to {output_filepath}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4ec94c55-a22f-416c-9974-34ba7625db3d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# add another color to compare to trees for common IDs\n",
    "# Define the color mapping based on common categories between two trees\n",
    "# Updated category colors for common branches\n",
    "\n",
    "def extract_id(label):\n",
    "    \"\"\"\n",
    "    Extracts the last part of the label after the last hyphen up to the colon,\n",
    "    which indicates the start of the branch length.\n",
    "    Example: 'EFG-list2-UniRef100-A0A7Y4R8I1:0.4184184849' returns 'A0A7Y4R8I1'.\n",
    "    \"\"\"\n",
    "    if ':' in label:\n",
    "        label = label.split(':')[0]\n",
    "    if '-' in label:\n",
    "        label = label.split('-')[-1]\n",
    "    return label\n",
    "\n",
    "def compare_trees_and_color_branches(tree_file1, tree_file2, output_suffix):\n",
    "    # Read both trees\n",
    "    tree1 = Phylo.read(tree_file1, 'newick')\n",
    "    tree2 = Phylo.read(tree_file2, 'newick')\n",
    "\n",
    "    # Extract and process labels from both trees using the updated extract_id\n",
    "    labels1 = {clade.name: extract_id(clade.name) for clade in tree1.get_terminals() if clade.name}\n",
    "    labels2 = {extract_id(clade.name) for clade in tree2.get_terminals() if clade.name}\n",
    "\n",
    "    # Find common labels based on the extracted IDs\n",
    "    common_labels = set(labels1.values()).intersection(labels2)\n",
    "\n",
    "    # Generate iTOL dataset for common labels\n",
    "    dataset_content = 'DATASET_COLORSTRIP\\nSEPARATOR SPACE\\nDATASET_LABEL common_labels\\nCOLOR #ff4500\\nDATA\\n'\n",
    "    for label, shortened_id in labels1.items():\n",
    "        if shortened_id in common_labels:\n",
    "            dataset_content += f'{label} #4c1130\\n'  # Color common labels from tree_file1 red\n",
    "    # Define the output file path for the iTOL dataset\n",
    "    output_file_path = f\"{directory_path}common_branches_{output_suffix}.txt\"\n",
    "    with open(output_file_path, 'w') as out_file:\n",
    "        out_file.write(dataset_content)\n",
    "    print(f\"iTOL dataset file created: {output_file_path}\")\n",
    "\n",
    "    # Optionally save the full IDs from tree_file1 that are common to both trees\n",
    "    common_ids_path = f'common_ids_from_{os.path.basename(tree_file1)}.txt'\n",
    "    with open(common_ids_path, 'w') as id_file:\n",
    "        for label, shortened_id in labels1.items():\n",
    "            if shortened_id in common_labels:\n",
    "                id_file.write(label + '\\n')\n",
    "    print(f\"Common IDs from {tree_file1} saved to {common_ids_path}\")\n",
    "\n",
    "# Example usage (Update these paths to match your files)\n",
    "tree_file1 = f\"{directory_path}/WT_All_aligned_anc.treefile\"\n",
    "tree_file2 = f\"{directory_path}/Node10list1list2_TAX_shortened_aligned.treefile\"\n",
    "compare_trees_and_color_branches(tree_file1, tree_file2, \"dataset\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "040c9ef3-cd47-4524-837d-e939b5a6a769",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (myenv)",
   "language": "python",
   "name": "myenv"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
